<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_201) on Fri Oct 04 10:35:33 EDT 2019 -->
<title>AbstractAlignmentMerger (picard 2.21.1-SNAPSHOT API)</title>
<meta name="date" content="2019-10-04">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="AbstractAlignmentMerger (picard 2.21.1-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":9,"i1":10,"i2":9,"i3":9,"i4":9,"i5":9,"i6":10,"i7":10,"i8":6,"i9":10,"i10":10,"i11":6,"i12":10,"i13":10,"i14":10,"i15":10,"i16":10,"i17":10,"i18":10,"i19":10,"i20":10,"i21":10,"i22":10,"i23":10,"i24":10,"i25":10,"i26":10,"i27":10,"i28":10};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],4:["t3","Abstract Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev&nbsp;Class</li>
<li><a href="../../picard/sam/AbstractAlignmentMerger.UnmappingReadStrategy.html" title="enum in picard.sam"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/sam/AbstractAlignmentMerger.html" target="_top">Frames</a></li>
<li><a href="AbstractAlignmentMerger.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">picard.sam</div>
<h2 title="Class AbstractAlignmentMerger" class="title">Class AbstractAlignmentMerger</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>picard.sam.AbstractAlignmentMerger</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>Direct Known Subclasses:</dt>
<dd><a href="../../picard/sam/SamAlignmentMerger.html" title="class in picard.sam">SamAlignmentMerger</a></dd>
</dl>
<hr>
<br>
<pre>public abstract class <span class="typeNameLabel">AbstractAlignmentMerger</span>
extends java.lang.Object</pre>
<div class="block">Abstract class that coordinates the general task of taking in a set of alignment information,
 possibly in SAM format, possibly in other formats, and merging that with the set of all reads
 for which alignment was attempted, stored in an unmapped SAM file.
 <p/>
 The order of processing is as follows:
 <p/>
 1.  Get records from the unmapped bam and the alignment data
 2.  Merge the alignment information and public tags ONLY from the aligned SAMRecords
 3.  Do additional modifications -- handle clipping, trimming, etc.
 4.  Fix up mate information on paired reads
 5.  Do a final calculation of the NM and UQ tags (coordinate sorted only)
 6.  Write the records to the output file.
 <p/>
 Concrete subclasses which extend AbstractAlignmentMerger should implement getQueryNameSortedAlignedRecords.
 If these records are not in queryname order, mergeAlignment will throw an IllegalStateException.
 <p/>
 Subclasses may optionally implement ignoreAlignment(), which can be used to skip over certain alignments.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.UnmappingReadStrategy.html" title="enum in picard.sam">AbstractAlignmentMerger.UnmappingReadStrategy</a></span></code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#MAX_RECORDS_IN_RAM">MAX_RECORDS_IN_RAM</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected java.io.File</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#referenceFasta">referenceFasta</a></span></code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#AbstractAlignmentMerger-java.io.File-java.io.File-java.io.File-boolean-boolean-boolean-htsjdk.samtools.SAMProgramRecord-java.util.List-java.util.List-java.lang.Integer-java.lang.Integer-java.util.List-htsjdk.samtools.SAMFileHeader.SortOrder-picard.sam.PrimaryAlignmentSelectionStrategy-boolean-boolean-">AbstractAlignmentMerger</a></span>(java.io.File&nbsp;unmappedBamFile,
                       java.io.File&nbsp;targetBamFile,
                       java.io.File&nbsp;referenceFasta,
                       boolean&nbsp;clipAdapters,
                       boolean&nbsp;bisulfiteSequence,
                       boolean&nbsp;alignedReadsOnly,
                       htsjdk.samtools.SAMProgramRecord&nbsp;programRecord,
                       java.util.List&lt;java.lang.String&gt;&nbsp;attributesToRetain,
                       java.util.List&lt;java.lang.String&gt;&nbsp;attributesToRemove,
                       java.lang.Integer&nbsp;read1BasesTrimmed,
                       java.lang.Integer&nbsp;read2BasesTrimmed,
                       java.util.List&lt;htsjdk.samtools.SamPairUtil.PairOrientation&gt;&nbsp;expectedOrientations,
                       htsjdk.samtools.SAMFileHeader.SortOrder&nbsp;sortOrder,
                       <a href="../../picard/sam/PrimaryAlignmentSelectionStrategy.html" title="interface in picard.sam">PrimaryAlignmentSelectionStrategy</a>&nbsp;primaryAlignmentSelectionStrategy,
                       boolean&nbsp;addMateCigar,
                       boolean&nbsp;unmapContaminantReads)</code>
<div class="block">constructor with a default setting for unmappingReadsStrategy.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#AbstractAlignmentMerger-java.io.File-java.io.File-java.io.File-boolean-boolean-boolean-htsjdk.samtools.SAMProgramRecord-java.util.List-java.util.List-java.lang.Integer-java.lang.Integer-java.util.List-htsjdk.samtools.SAMFileHeader.SortOrder-picard.sam.PrimaryAlignmentSelectionStrategy-boolean-boolean-picard.sam.AbstractAlignmentMerger.UnmappingReadStrategy-">AbstractAlignmentMerger</a></span>(java.io.File&nbsp;unmappedBamFile,
                       java.io.File&nbsp;targetBamFile,
                       java.io.File&nbsp;referenceFasta,
                       boolean&nbsp;clipAdapters,
                       boolean&nbsp;bisulfiteSequence,
                       boolean&nbsp;alignedReadsOnly,
                       htsjdk.samtools.SAMProgramRecord&nbsp;programRecord,
                       java.util.List&lt;java.lang.String&gt;&nbsp;attributesToRetain,
                       java.util.List&lt;java.lang.String&gt;&nbsp;attributesToRemove,
                       java.lang.Integer&nbsp;read1BasesTrimmed,
                       java.lang.Integer&nbsp;read2BasesTrimmed,
                       java.util.List&lt;htsjdk.samtools.SamPairUtil.PairOrientation&gt;&nbsp;expectedOrientations,
                       htsjdk.samtools.SAMFileHeader.SortOrder&nbsp;sortOrder,
                       <a href="../../picard/sam/PrimaryAlignmentSelectionStrategy.html" title="interface in picard.sam">PrimaryAlignmentSelectionStrategy</a>&nbsp;primaryAlignmentSelectionStrategy,
                       boolean&nbsp;addMateCigar,
                       boolean&nbsp;unmapContaminantReads,
                       <a href="../../picard/sam/AbstractAlignmentMerger.UnmappingReadStrategy.html" title="enum in picard.sam">AbstractAlignmentMerger.UnmappingReadStrategy</a>&nbsp;unmappingReadsStrategy)</code>
<div class="block">Constructor</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t3" class="tableTab"><span><a href="javascript:show(4);">Abstract Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>protected static void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#clipForOverlappingReads-htsjdk.samtools.SAMRecord-htsjdk.samtools.SAMRecord-">clipForOverlappingReads</a></span>(htsjdk.samtools.SAMRecord&nbsp;read1,
                       htsjdk.samtools.SAMRecord&nbsp;read2)</code>
<div class="block">Checks to see whether the ends of the reads overlap and soft clips reads
 them if necessary.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#close--">close</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#createNewCigarsIfMapsOffEndOfReference-htsjdk.samtools.SAMRecord-">createNewCigarsIfMapsOffEndOfReference</a></span>(htsjdk.samtools.SAMRecord&nbsp;rec)</code>
<div class="block">Soft-clip an alignment that hangs off the end of its reference sequence.</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#encodeMappingInformation-htsjdk.samtools.SAMRecord-">encodeMappingInformation</a></span>(htsjdk.samtools.SAMRecord&nbsp;rec)</code>
<div class="block">Encodes mapping information from a record into a string according to the format sepcified in the
 Sam-Spec under the SA tag.</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#fixNmMdAndUq-htsjdk.samtools.SAMRecord-htsjdk.samtools.reference.ReferenceSequenceFileWalker-boolean-">fixNmMdAndUq</a></span>(htsjdk.samtools.SAMRecord&nbsp;record,
            htsjdk.samtools.reference.ReferenceSequenceFileWalker&nbsp;refSeqWalker,
            boolean&nbsp;isBisulfiteSequence)</code>
<div class="block">Calculates and sets the NM, MD, and and UQ tags from the record and the reference</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#fixUq-htsjdk.samtools.SAMRecord-htsjdk.samtools.reference.ReferenceSequenceFileWalker-boolean-">fixUq</a></span>(htsjdk.samtools.SAMRecord&nbsp;record,
     htsjdk.samtools.reference.ReferenceSequenceFileWalker&nbsp;refSeqWalker,
     boolean&nbsp;isBisulfiteSequence)</code>
<div class="block">Calculates and sets UQ tag from the record and the reference</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>java.util.Set&lt;java.lang.String&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#getAttributesToReverse--">getAttributesToReverse</a></span>()</code>
<div class="block">Gets the set of attributes to be reversed on reads marked as negative strand.</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>java.util.Set&lt;java.lang.String&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#getAttributesToReverseComplement--">getAttributesToReverseComplement</a></span>()</code>
<div class="block">Gets the set of attributes to be reverse complemented on reads marked as negative strand.</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>protected abstract htsjdk.samtools.SAMSequenceDictionary</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#getDictionaryForMergedBam--">getDictionaryForMergedBam</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>protected htsjdk.samtools.SAMFileHeader</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#getHeader--">getHeader</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>protected htsjdk.samtools.SAMProgramRecord</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#getProgramRecord--">getProgramRecord</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>protected abstract htsjdk.samtools.util.CloseableIterator&lt;htsjdk.samtools.SAMRecord&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#getQuerynameSortedAlignedRecords--">getQuerynameSortedAlignedRecords</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>protected boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#ignoreAlignment-htsjdk.samtools.SAMRecord-">ignoreAlignment</a></span>(htsjdk.samtools.SAMRecord&nbsp;sam)</code>&nbsp;</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#isClipOverlappingReads--">isClipOverlappingReads</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>protected boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#isContaminant-picard.sam.HitsForInsert-">isContaminant</a></span>(picard.sam.HitsForInsert&nbsp;hits)</code>&nbsp;</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#isKeepAlignerProperPairFlags--">isKeepAlignerProperPairFlags</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>protected boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#isReservedTag-java.lang.String-">isReservedTag</a></span>(java.lang.String&nbsp;tag)</code>&nbsp;</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#mergeAlignment-java.io.File-">mergeAlignment</a></span>(java.io.File&nbsp;referenceFasta)</code>
<div class="block">Merges the alignment data with the non-aligned records from the source BAM file.</div>
</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#resetRefSeqFileWalker--">resetRefSeqFileWalker</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#setAddPGTagToReads-boolean-">setAddPGTagToReads</a></span>(boolean&nbsp;addPGTagToReads)</code>
<div class="block">Set addPGTagToReads.</div>
</td>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#setAttributesToReverse-java.util.Set-">setAttributesToReverse</a></span>(java.util.Set&lt;java.lang.String&gt;&nbsp;attributesToReverse)</code>
<div class="block">Sets the set of attributes to be reversed on reads marked as negative strand.</div>
</td>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#setAttributesToReverseComplement-java.util.Set-">setAttributesToReverseComplement</a></span>(java.util.Set&lt;java.lang.String&gt;&nbsp;attributesToReverseComplement)</code>
<div class="block">Sets the set of attributes to be reverse complemented on reads marked as negative strand.</div>
</td>
</tr>
<tr id="i22" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#setClipOverlappingReads-boolean-">setClipOverlappingReads</a></span>(boolean&nbsp;clipOverlappingReads)</code>&nbsp;</td>
</tr>
<tr id="i23" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#setIncludeSecondaryAlignments-boolean-">setIncludeSecondaryAlignments</a></span>(boolean&nbsp;includeSecondaryAlignments)</code>&nbsp;</td>
</tr>
<tr id="i24" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#setKeepAlignerProperPairFlags-boolean-">setKeepAlignerProperPairFlags</a></span>(boolean&nbsp;keepAlignerProperPairFlags)</code>
<div class="block">If true, keep the aligner's idea of proper pairs rather than letting alignment merger decide.</div>
</td>
</tr>
<tr id="i25" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#setMaxRecordsInRam-int-">setMaxRecordsInRam</a></span>(int&nbsp;maxRecordsInRam)</code>
<div class="block">Allows the caller to override the maximum records in RAM.</div>
</td>
</tr>
<tr id="i26" class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#setProgramRecord-htsjdk.samtools.SAMProgramRecord-">setProgramRecord</a></span>(htsjdk.samtools.SAMProgramRecord&nbsp;pg)</code>&nbsp;</td>
</tr>
<tr id="i27" class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#setValuesFromAlignment-htsjdk.samtools.SAMRecord-htsjdk.samtools.SAMRecord-boolean-">setValuesFromAlignment</a></span>(htsjdk.samtools.SAMRecord&nbsp;rec,
                      htsjdk.samtools.SAMRecord&nbsp;alignment,
                      boolean&nbsp;needsSafeReverseComplement)</code>
<div class="block">Sets the values from the alignment record on the unaligned BAM record.</div>
</td>
</tr>
<tr id="i28" class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/AbstractAlignmentMerger.html#updateCigarForTrimmedOrClippedBases-htsjdk.samtools.SAMRecord-htsjdk.samtools.SAMRecord-">updateCigarForTrimmedOrClippedBases</a></span>(htsjdk.samtools.SAMRecord&nbsp;rec,
                                   htsjdk.samtools.SAMRecord&nbsp;alignment)</code>&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="MAX_RECORDS_IN_RAM">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>MAX_RECORDS_IN_RAM</h4>
<pre>public static final&nbsp;int MAX_RECORDS_IN_RAM</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../constant-values.html#picard.sam.AbstractAlignmentMerger.MAX_RECORDS_IN_RAM">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a name="referenceFasta">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>referenceFasta</h4>
<pre>protected final&nbsp;java.io.File referenceFasta</pre>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="AbstractAlignmentMerger-java.io.File-java.io.File-java.io.File-boolean-boolean-boolean-htsjdk.samtools.SAMProgramRecord-java.util.List-java.util.List-java.lang.Integer-java.lang.Integer-java.util.List-htsjdk.samtools.SAMFileHeader.SortOrder-picard.sam.PrimaryAlignmentSelectionStrategy-boolean-boolean-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>AbstractAlignmentMerger</h4>
<pre>public&nbsp;AbstractAlignmentMerger(java.io.File&nbsp;unmappedBamFile,
                               java.io.File&nbsp;targetBamFile,
                               java.io.File&nbsp;referenceFasta,
                               boolean&nbsp;clipAdapters,
                               boolean&nbsp;bisulfiteSequence,
                               boolean&nbsp;alignedReadsOnly,
                               htsjdk.samtools.SAMProgramRecord&nbsp;programRecord,
                               java.util.List&lt;java.lang.String&gt;&nbsp;attributesToRetain,
                               java.util.List&lt;java.lang.String&gt;&nbsp;attributesToRemove,
                               java.lang.Integer&nbsp;read1BasesTrimmed,
                               java.lang.Integer&nbsp;read2BasesTrimmed,
                               java.util.List&lt;htsjdk.samtools.SamPairUtil.PairOrientation&gt;&nbsp;expectedOrientations,
                               htsjdk.samtools.SAMFileHeader.SortOrder&nbsp;sortOrder,
                               <a href="../../picard/sam/PrimaryAlignmentSelectionStrategy.html" title="interface in picard.sam">PrimaryAlignmentSelectionStrategy</a>&nbsp;primaryAlignmentSelectionStrategy,
                               boolean&nbsp;addMateCigar,
                               boolean&nbsp;unmapContaminantReads)</pre>
<div class="block">constructor with a default setting for unmappingReadsStrategy.

 see full constructor for parameters</div>
</li>
</ul>
<a name="AbstractAlignmentMerger-java.io.File-java.io.File-java.io.File-boolean-boolean-boolean-htsjdk.samtools.SAMProgramRecord-java.util.List-java.util.List-java.lang.Integer-java.lang.Integer-java.util.List-htsjdk.samtools.SAMFileHeader.SortOrder-picard.sam.PrimaryAlignmentSelectionStrategy-boolean-boolean-picard.sam.AbstractAlignmentMerger.UnmappingReadStrategy-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>AbstractAlignmentMerger</h4>
<pre>public&nbsp;AbstractAlignmentMerger(java.io.File&nbsp;unmappedBamFile,
                               java.io.File&nbsp;targetBamFile,
                               java.io.File&nbsp;referenceFasta,
                               boolean&nbsp;clipAdapters,
                               boolean&nbsp;bisulfiteSequence,
                               boolean&nbsp;alignedReadsOnly,
                               htsjdk.samtools.SAMProgramRecord&nbsp;programRecord,
                               java.util.List&lt;java.lang.String&gt;&nbsp;attributesToRetain,
                               java.util.List&lt;java.lang.String&gt;&nbsp;attributesToRemove,
                               java.lang.Integer&nbsp;read1BasesTrimmed,
                               java.lang.Integer&nbsp;read2BasesTrimmed,
                               java.util.List&lt;htsjdk.samtools.SamPairUtil.PairOrientation&gt;&nbsp;expectedOrientations,
                               htsjdk.samtools.SAMFileHeader.SortOrder&nbsp;sortOrder,
                               <a href="../../picard/sam/PrimaryAlignmentSelectionStrategy.html" title="interface in picard.sam">PrimaryAlignmentSelectionStrategy</a>&nbsp;primaryAlignmentSelectionStrategy,
                               boolean&nbsp;addMateCigar,
                               boolean&nbsp;unmapContaminantReads,
                               <a href="../../picard/sam/AbstractAlignmentMerger.UnmappingReadStrategy.html" title="enum in picard.sam">AbstractAlignmentMerger.UnmappingReadStrategy</a>&nbsp;unmappingReadsStrategy)</pre>
<div class="block">Constructor</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>unmappedBamFile</code> - The BAM file that was used as the input to the aligner, which will
                                          include info on all the reads that did not map.  Required.</dd>
<dd><code>targetBamFile</code> - The file to which to write the merged SAM records. Required.</dd>
<dd><code>referenceFasta</code> - The reference sequence for the map files. Required.</dd>
<dd><code>clipAdapters</code> - Whether adapters marked in unmapped BAM file should be marked as
                                          soft clipped in the merged bam. Required.</dd>
<dd><code>bisulfiteSequence</code> - Whether the reads are bisulfite sequence (used when calculating the
                                          NM and UQ tags). Required.</dd>
<dd><code>alignedReadsOnly</code> - Whether to output only those reads that have alignment data</dd>
<dd><code>programRecord</code> - Program record for target file SAMRecords created.</dd>
<dd><code>attributesToRetain</code> - private attributes from the alignment record that should be
                                          included when merging.  This overrides the exclusion of
                                          attributes whose tags start with the reserved characters
                                          of X, Y, and Z</dd>
<dd><code>attributesToRemove</code> - attributes from the alignment record that should be
                                          removed when merging.  This overrides attributesToRetain if they share
                                          common tags.</dd>
<dd><code>read1BasesTrimmed</code> - The number of bases trimmed from start of read 1 prior to alignment.  Optional.</dd>
<dd><code>read2BasesTrimmed</code> - The number of bases trimmed from start of read 2 prior to alignment.  Optional.</dd>
<dd><code>expectedOrientations</code> - A List of SamPairUtil.PairOrientations that are expected for
                                          aligned pairs.  Used to determine the properPair flag.</dd>
<dd><code>sortOrder</code> - The order in which the merged records should be output.  If null,
                                          output will be coordinate-sorted</dd>
<dd><code>primaryAlignmentSelectionStrategy</code> - What to do when there are multiple primary alignments, or multiple
                                          alignments but none primary, for a read or read pair.</dd>
<dd><code>addMateCigar</code> - True if we are to add or maintain the mate CIGAR (MC) tag, false if we are to remove or not include.</dd>
<dd><code>unmapContaminantReads</code> - If true, identify reads having the signature of cross-species contamination (i.e. mostly clipped bases),
                                          and mark them as unmapped.</dd>
<dd><code>unmappingReadsStrategy</code> - An enum describing how to deal with reads whose mapping information are being removed (currently this happens due to cross-species
                                          contamination). Ignored unless unmapContaminantReads is true.</dd>
</dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getDictionaryForMergedBam--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDictionaryForMergedBam</h4>
<pre>protected abstract&nbsp;htsjdk.samtools.SAMSequenceDictionary&nbsp;getDictionaryForMergedBam()</pre>
</li>
</ul>
<a name="getQuerynameSortedAlignedRecords--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getQuerynameSortedAlignedRecords</h4>
<pre>protected abstract&nbsp;htsjdk.samtools.util.CloseableIterator&lt;htsjdk.samtools.SAMRecord&gt;&nbsp;getQuerynameSortedAlignedRecords()</pre>
</li>
</ul>
<a name="ignoreAlignment-htsjdk.samtools.SAMRecord-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>ignoreAlignment</h4>
<pre>protected&nbsp;boolean&nbsp;ignoreAlignment(htsjdk.samtools.SAMRecord&nbsp;sam)</pre>
</li>
</ul>
<a name="isContaminant-picard.sam.HitsForInsert-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isContaminant</h4>
<pre>protected&nbsp;boolean&nbsp;isContaminant(picard.sam.HitsForInsert&nbsp;hits)</pre>
</li>
</ul>
<a name="getAttributesToReverse--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAttributesToReverse</h4>
<pre>public&nbsp;java.util.Set&lt;java.lang.String&gt;&nbsp;getAttributesToReverse()</pre>
<div class="block">Gets the set of attributes to be reversed on reads marked as negative strand.</div>
</li>
</ul>
<a name="setAttributesToReverse-java.util.Set-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAttributesToReverse</h4>
<pre>public&nbsp;void&nbsp;setAttributesToReverse(java.util.Set&lt;java.lang.String&gt;&nbsp;attributesToReverse)</pre>
<div class="block">Sets the set of attributes to be reversed on reads marked as negative strand.</div>
</li>
</ul>
<a name="getAttributesToReverseComplement--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAttributesToReverseComplement</h4>
<pre>public&nbsp;java.util.Set&lt;java.lang.String&gt;&nbsp;getAttributesToReverseComplement()</pre>
<div class="block">Gets the set of attributes to be reverse complemented on reads marked as negative strand.</div>
</li>
</ul>
<a name="setAttributesToReverseComplement-java.util.Set-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAttributesToReverseComplement</h4>
<pre>public&nbsp;void&nbsp;setAttributesToReverseComplement(java.util.Set&lt;java.lang.String&gt;&nbsp;attributesToReverseComplement)</pre>
<div class="block">Sets the set of attributes to be reverse complemented on reads marked as negative strand.</div>
</li>
</ul>
<a name="setMaxRecordsInRam-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMaxRecordsInRam</h4>
<pre>public&nbsp;void&nbsp;setMaxRecordsInRam(int&nbsp;maxRecordsInRam)</pre>
<div class="block">Allows the caller to override the maximum records in RAM.</div>
</li>
</ul>
<a name="setAddPGTagToReads-boolean-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAddPGTagToReads</h4>
<pre>public&nbsp;void&nbsp;setAddPGTagToReads(boolean&nbsp;addPGTagToReads)</pre>
<div class="block">Set addPGTagToReads. If true, the PG will be added to reads when applicable. If false, the PG tag will not be added.
 Default is true</div>
</li>
</ul>
<a name="mergeAlignment-java.io.File-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>mergeAlignment</h4>
<pre>public&nbsp;void&nbsp;mergeAlignment(java.io.File&nbsp;referenceFasta)</pre>
<div class="block">Merges the alignment data with the non-aligned records from the source BAM file.</div>
</li>
</ul>
<a name="fixNmMdAndUq-htsjdk.samtools.SAMRecord-htsjdk.samtools.reference.ReferenceSequenceFileWalker-boolean-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fixNmMdAndUq</h4>
<pre>public static&nbsp;void&nbsp;fixNmMdAndUq(htsjdk.samtools.SAMRecord&nbsp;record,
                                htsjdk.samtools.reference.ReferenceSequenceFileWalker&nbsp;refSeqWalker,
                                boolean&nbsp;isBisulfiteSequence)</pre>
<div class="block">Calculates and sets the NM, MD, and and UQ tags from the record and the reference</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>record</code> - the record to be fixed</dd>
<dd><code>refSeqWalker</code> - a ReferenceSequenceWalker that will be used to traverse the reference</dd>
<dd><code>isBisulfiteSequence</code> - a flag indicating whether the sequence came from bisulfite-sequencing which would imply a different
 calculation of the NM tag.

 No return value, modifies the provided record.</dd>
</dl>
</li>
</ul>
<a name="fixUq-htsjdk.samtools.SAMRecord-htsjdk.samtools.reference.ReferenceSequenceFileWalker-boolean-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fixUq</h4>
<pre>public static&nbsp;void&nbsp;fixUq(htsjdk.samtools.SAMRecord&nbsp;record,
                         htsjdk.samtools.reference.ReferenceSequenceFileWalker&nbsp;refSeqWalker,
                         boolean&nbsp;isBisulfiteSequence)</pre>
<div class="block">Calculates and sets UQ tag from the record and the reference</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>record</code> - the record to be fixed</dd>
<dd><code>refSeqWalker</code> - a ReferenceSequenceWalker that will be used to traverse the reference</dd>
<dd><code>isBisulfiteSequence</code> - a flag indicating whether the sequence came from bisulfite-sequencing.

 No return value, modifies the provided record.</dd>
</dl>
</li>
</ul>
<a name="encodeMappingInformation-htsjdk.samtools.SAMRecord-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>encodeMappingInformation</h4>
<pre>public static&nbsp;java.lang.String&nbsp;encodeMappingInformation(htsjdk.samtools.SAMRecord&nbsp;rec)</pre>
<div class="block">Encodes mapping information from a record into a string according to the format sepcified in the
 Sam-Spec under the SA tag. No protection against missing values (for cigar, and NM tag).
 (Might make sense to move this to htsJDK.)</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>rec</code> - SAMRecord whose alignment information will be encoded</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>String encoding rec's alignment information according to SA tag in the SAM spec</dd>
</dl>
</li>
</ul>
<a name="clipForOverlappingReads-htsjdk.samtools.SAMRecord-htsjdk.samtools.SAMRecord-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>clipForOverlappingReads</h4>
<pre>protected static&nbsp;void&nbsp;clipForOverlappingReads(htsjdk.samtools.SAMRecord&nbsp;read1,
                                              htsjdk.samtools.SAMRecord&nbsp;read2)</pre>
<div class="block">Checks to see whether the ends of the reads overlap and soft clips reads
 them if necessary.</div>
</li>
</ul>
<a name="setValuesFromAlignment-htsjdk.samtools.SAMRecord-htsjdk.samtools.SAMRecord-boolean-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setValuesFromAlignment</h4>
<pre>protected&nbsp;void&nbsp;setValuesFromAlignment(htsjdk.samtools.SAMRecord&nbsp;rec,
                                      htsjdk.samtools.SAMRecord&nbsp;alignment,
                                      boolean&nbsp;needsSafeReverseComplement)</pre>
<div class="block">Sets the values from the alignment record on the unaligned BAM record.  This
 preserves all data from the unaligned record (ReadGroup, NoiseRead status, etc)
 and adds all the alignment info</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>rec</code> - The unaligned read record</dd>
<dd><code>alignment</code> - The alignment record</dd>
</dl>
</li>
</ul>
<a name="createNewCigarsIfMapsOffEndOfReference-htsjdk.samtools.SAMRecord-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createNewCigarsIfMapsOffEndOfReference</h4>
<pre>public static&nbsp;void&nbsp;createNewCigarsIfMapsOffEndOfReference(htsjdk.samtools.SAMRecord&nbsp;rec)</pre>
<div class="block">Soft-clip an alignment that hangs off the end of its reference sequence.  Checks both the read and its mate,
 if available.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>rec</code> - </dd>
</dl>
</li>
</ul>
<a name="updateCigarForTrimmedOrClippedBases-htsjdk.samtools.SAMRecord-htsjdk.samtools.SAMRecord-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>updateCigarForTrimmedOrClippedBases</h4>
<pre>protected&nbsp;void&nbsp;updateCigarForTrimmedOrClippedBases(htsjdk.samtools.SAMRecord&nbsp;rec,
                                                   htsjdk.samtools.SAMRecord&nbsp;alignment)</pre>
</li>
</ul>
<a name="getProgramRecord--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProgramRecord</h4>
<pre>protected&nbsp;htsjdk.samtools.SAMProgramRecord&nbsp;getProgramRecord()</pre>
</li>
</ul>
<a name="setProgramRecord-htsjdk.samtools.SAMProgramRecord-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setProgramRecord</h4>
<pre>protected&nbsp;void&nbsp;setProgramRecord(htsjdk.samtools.SAMProgramRecord&nbsp;pg)</pre>
</li>
</ul>
<a name="isReservedTag-java.lang.String-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isReservedTag</h4>
<pre>protected&nbsp;boolean&nbsp;isReservedTag(java.lang.String&nbsp;tag)</pre>
</li>
</ul>
<a name="getHeader--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getHeader</h4>
<pre>protected&nbsp;htsjdk.samtools.SAMFileHeader&nbsp;getHeader()</pre>
</li>
</ul>
<a name="resetRefSeqFileWalker--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>resetRefSeqFileWalker</h4>
<pre>protected&nbsp;void&nbsp;resetRefSeqFileWalker()</pre>
</li>
</ul>
<a name="isClipOverlappingReads--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isClipOverlappingReads</h4>
<pre>public&nbsp;boolean&nbsp;isClipOverlappingReads()</pre>
</li>
</ul>
<a name="setClipOverlappingReads-boolean-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setClipOverlappingReads</h4>
<pre>public&nbsp;void&nbsp;setClipOverlappingReads(boolean&nbsp;clipOverlappingReads)</pre>
</li>
</ul>
<a name="isKeepAlignerProperPairFlags--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isKeepAlignerProperPairFlags</h4>
<pre>public&nbsp;boolean&nbsp;isKeepAlignerProperPairFlags()</pre>
</li>
</ul>
<a name="setKeepAlignerProperPairFlags-boolean-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setKeepAlignerProperPairFlags</h4>
<pre>public&nbsp;void&nbsp;setKeepAlignerProperPairFlags(boolean&nbsp;keepAlignerProperPairFlags)</pre>
<div class="block">If true, keep the aligner's idea of proper pairs rather than letting alignment merger decide.</div>
</li>
</ul>
<a name="setIncludeSecondaryAlignments-boolean-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setIncludeSecondaryAlignments</h4>
<pre>public&nbsp;void&nbsp;setIncludeSecondaryAlignments(boolean&nbsp;includeSecondaryAlignments)</pre>
</li>
</ul>
<a name="close--">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>close</h4>
<pre>public&nbsp;void&nbsp;close()</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev&nbsp;Class</li>
<li><a href="../../picard/sam/AbstractAlignmentMerger.UnmappingReadStrategy.html" title="enum in picard.sam"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/sam/AbstractAlignmentMerger.html" target="_top">Frames</a></li>
<li><a href="AbstractAlignmentMerger.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
