<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_201) on Fri Oct 04 10:35:34 EDT 2019 -->
<title>SamAlignmentMerger (picard 2.21.1-SNAPSHOT API)</title>
<meta name="date" content="2019-10-04">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="SamAlignmentMerger (picard 2.21.1-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/sam/RevertSam.FileType.html" title="enum in picard.sam"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/sam/SamComparisonMetric.html" title="class in picard.sam"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/sam/SamAlignmentMerger.html" target="_top">Frames</a></li>
<li><a href="SamAlignmentMerger.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.classes.inherited.from.class.picard.sam.AbstractAlignmentMerger">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#fields.inherited.from.class.picard.sam.AbstractAlignmentMerger">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">picard.sam</div>
<h2 title="Class SamAlignmentMerger" class="title">Class SamAlignmentMerger</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li><a href="../../picard/sam/AbstractAlignmentMerger.html" title="class in picard.sam">picard.sam.AbstractAlignmentMerger</a></li>
<li>
<ul class="inheritance">
<li>picard.sam.SamAlignmentMerger</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public class <span class="typeNameLabel">SamAlignmentMerger</span>
extends <a href="../../picard/sam/AbstractAlignmentMerger.html" title="class in picard.sam">AbstractAlignmentMerger</a></pre>
<div class="block">Class that takes in a set of alignment information in SAM format and merges it with the set
 of all reads for which alignment was attempted, stored in an unmapped SAM file.  This
 class overrides mergeAlignment in AbstractAlignmentNMerger and proceeds on the assumption that
 the underlying alignment records are aleady in query-name sorted order (true for bwa).  If
 they are not, the mergeAlignment method catches the IllegalStateException, forces a sort
 of the underlying alignment records, and tries again.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="nested.classes.inherited.from.class.picard.sam.AbstractAlignmentMerger">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;picard.sam.<a href="../../picard/sam/AbstractAlignmentMerger.html" title="class in picard.sam">AbstractAlignmentMerger</a></h3>
<code><a href="../../picard/sam/AbstractAlignmentMerger.UnmappingReadStrategy.html" title="enum in picard.sam">AbstractAlignmentMerger.UnmappingReadStrategy</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="fields.inherited.from.class.picard.sam.AbstractAlignmentMerger">
<!--   -->
</a>
<h3>Fields inherited from class&nbsp;picard.sam.<a href="../../picard/sam/AbstractAlignmentMerger.html" title="class in picard.sam">AbstractAlignmentMerger</a></h3>
<code><a href="../../picard/sam/AbstractAlignmentMerger.html#MAX_RECORDS_IN_RAM">MAX_RECORDS_IN_RAM</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#referenceFasta">referenceFasta</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../picard/sam/SamAlignmentMerger.html#SamAlignmentMerger-java.io.File-java.io.File-java.io.File-htsjdk.samtools.SAMProgramRecord-boolean-boolean-boolean-java.util.List-int-java.util.List-java.util.List-java.lang.Integer-java.lang.Integer-java.util.List-java.util.List-java.util.List-htsjdk.samtools.SAMFileHeader.SortOrder-picard.sam.PrimaryAlignmentSelectionStrategy-boolean-boolean-int-">SamAlignmentMerger</a></span>(java.io.File&nbsp;unmappedBamFile,
                  java.io.File&nbsp;targetBamFile,
                  java.io.File&nbsp;referenceFasta,
                  htsjdk.samtools.SAMProgramRecord&nbsp;programRecord,
                  boolean&nbsp;clipAdapters,
                  boolean&nbsp;bisulfiteSequence,
                  boolean&nbsp;alignedReadsOnly,
                  java.util.List&lt;java.io.File&gt;&nbsp;alignedSamFile,
                  int&nbsp;maxGaps,
                  java.util.List&lt;java.lang.String&gt;&nbsp;attributesToRetain,
                  java.util.List&lt;java.lang.String&gt;&nbsp;attributesToRemove,
                  java.lang.Integer&nbsp;read1BasesTrimmed,
                  java.lang.Integer&nbsp;read2BasesTrimmed,
                  java.util.List&lt;java.io.File&gt;&nbsp;read1AlignedSamFile,
                  java.util.List&lt;java.io.File&gt;&nbsp;read2AlignedSamFile,
                  java.util.List&lt;htsjdk.samtools.SamPairUtil.PairOrientation&gt;&nbsp;expectedOrientations,
                  htsjdk.samtools.SAMFileHeader.SortOrder&nbsp;sortOrder,
                  <a href="../../picard/sam/PrimaryAlignmentSelectionStrategy.html" title="interface in picard.sam">PrimaryAlignmentSelectionStrategy</a>&nbsp;primaryAlignmentSelectionStrategy,
                  boolean&nbsp;addMateCigar,
                  boolean&nbsp;unmapContaminantReads,
                  int&nbsp;minUnclippedBases)</code>
<div class="block">Constructor with a default value for unmappingReadStrategy</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../picard/sam/SamAlignmentMerger.html#SamAlignmentMerger-java.io.File-java.io.File-java.io.File-htsjdk.samtools.SAMProgramRecord-boolean-boolean-boolean-java.util.List-int-java.util.List-java.util.List-java.lang.Integer-java.lang.Integer-java.util.List-java.util.List-java.util.List-htsjdk.samtools.SAMFileHeader.SortOrder-picard.sam.PrimaryAlignmentSelectionStrategy-boolean-boolean-int-picard.sam.AbstractAlignmentMerger.UnmappingReadStrategy-java.util.List-">SamAlignmentMerger</a></span>(java.io.File&nbsp;unmappedBamFile,
                  java.io.File&nbsp;targetBamFile,
                  java.io.File&nbsp;referenceFasta,
                  htsjdk.samtools.SAMProgramRecord&nbsp;programRecord,
                  boolean&nbsp;clipAdapters,
                  boolean&nbsp;bisulfiteSequence,
                  boolean&nbsp;alignedReadsOnly,
                  java.util.List&lt;java.io.File&gt;&nbsp;alignedSamFile,
                  int&nbsp;maxGaps,
                  java.util.List&lt;java.lang.String&gt;&nbsp;attributesToRetain,
                  java.util.List&lt;java.lang.String&gt;&nbsp;attributesToRemove,
                  java.lang.Integer&nbsp;read1BasesTrimmed,
                  java.lang.Integer&nbsp;read2BasesTrimmed,
                  java.util.List&lt;java.io.File&gt;&nbsp;read1AlignedSamFile,
                  java.util.List&lt;java.io.File&gt;&nbsp;read2AlignedSamFile,
                  java.util.List&lt;htsjdk.samtools.SamPairUtil.PairOrientation&gt;&nbsp;expectedOrientations,
                  htsjdk.samtools.SAMFileHeader.SortOrder&nbsp;sortOrder,
                  <a href="../../picard/sam/PrimaryAlignmentSelectionStrategy.html" title="interface in picard.sam">PrimaryAlignmentSelectionStrategy</a>&nbsp;primaryAlignmentSelectionStrategy,
                  boolean&nbsp;addMateCigar,
                  boolean&nbsp;unmapContaminantReads,
                  int&nbsp;minUnclippedBases,
                  <a href="../../picard/sam/AbstractAlignmentMerger.UnmappingReadStrategy.html" title="enum in picard.sam">AbstractAlignmentMerger.UnmappingReadStrategy</a>&nbsp;unmappingReadStrategy,
                  java.util.List&lt;java.lang.String&gt;&nbsp;requiredMatchingDictionaryTags)</code>
<div class="block">Constructor</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>protected htsjdk.samtools.SAMSequenceDictionary</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/SamAlignmentMerger.html#getDictionaryForMergedBam--">getDictionaryForMergedBam</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/SamAlignmentMerger.html#getForceSort--">getForceSort</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>protected htsjdk.samtools.util.CloseableIterator&lt;htsjdk.samtools.SAMRecord&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/SamAlignmentMerger.html#getQuerynameSortedAlignedRecords--">getQuerynameSortedAlignedRecords</a></span>()</code>
<div class="block">Reads the aligned SAM records into a SortingCollection and returns an iterator over that collection</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>protected boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/SamAlignmentMerger.html#ignoreAlignment-htsjdk.samtools.SAMRecord-">ignoreAlignment</a></span>(htsjdk.samtools.SAMRecord&nbsp;sam)</code>
<div class="block">For now, we ignore only those alignments that have more than <a href="../../picard/sam/SamAlignmentMerger.html#maxGaps"><code>maxGaps</code></a> insertions
 or deletions.</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>protected boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/SamAlignmentMerger.html#isContaminant-picard.sam.HitsForInsert-">isContaminant</a></span>(picard.sam.HitsForInsert&nbsp;hits)</code>
<div class="block">Criteria for contaminant reads:
 1.</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/sam/SamAlignmentMerger.html#mergeAlignment-java.io.File-">mergeAlignment</a></span>(java.io.File&nbsp;referenceFasta)</code>
<div class="block">Merges the alignment from the map file with the non-aligned records from the source BAM file.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.picard.sam.AbstractAlignmentMerger">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;picard.sam.<a href="../../picard/sam/AbstractAlignmentMerger.html" title="class in picard.sam">AbstractAlignmentMerger</a></h3>
<code><a href="../../picard/sam/AbstractAlignmentMerger.html#clipForOverlappingReads-htsjdk.samtools.SAMRecord-htsjdk.samtools.SAMRecord-">clipForOverlappingReads</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#close--">close</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#createNewCigarsIfMapsOffEndOfReference-htsjdk.samtools.SAMRecord-">createNewCigarsIfMapsOffEndOfReference</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#encodeMappingInformation-htsjdk.samtools.SAMRecord-">encodeMappingInformation</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#fixNmMdAndUq-htsjdk.samtools.SAMRecord-htsjdk.samtools.reference.ReferenceSequenceFileWalker-boolean-">fixNmMdAndUq</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#fixUq-htsjdk.samtools.SAMRecord-htsjdk.samtools.reference.ReferenceSequenceFileWalker-boolean-">fixUq</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#getAttributesToReverse--">getAttributesToReverse</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#getAttributesToReverseComplement--">getAttributesToReverseComplement</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#getHeader--">getHeader</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#getProgramRecord--">getProgramRecord</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#isClipOverlappingReads--">isClipOverlappingReads</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#isKeepAlignerProperPairFlags--">isKeepAlignerProperPairFlags</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#isReservedTag-java.lang.String-">isReservedTag</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#resetRefSeqFileWalker--">resetRefSeqFileWalker</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#setAddPGTagToReads-boolean-">setAddPGTagToReads</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#setAttributesToReverse-java.util.Set-">setAttributesToReverse</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#setAttributesToReverseComplement-java.util.Set-">setAttributesToReverseComplement</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#setClipOverlappingReads-boolean-">setClipOverlappingReads</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#setIncludeSecondaryAlignments-boolean-">setIncludeSecondaryAlignments</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#setKeepAlignerProperPairFlags-boolean-">setKeepAlignerProperPairFlags</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#setMaxRecordsInRam-int-">setMaxRecordsInRam</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#setProgramRecord-htsjdk.samtools.SAMProgramRecord-">setProgramRecord</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#setValuesFromAlignment-htsjdk.samtools.SAMRecord-htsjdk.samtools.SAMRecord-boolean-">setValuesFromAlignment</a>, <a href="../../picard/sam/AbstractAlignmentMerger.html#updateCigarForTrimmedOrClippedBases-htsjdk.samtools.SAMRecord-htsjdk.samtools.SAMRecord-">updateCigarForTrimmedOrClippedBases</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="SamAlignmentMerger-java.io.File-java.io.File-java.io.File-htsjdk.samtools.SAMProgramRecord-boolean-boolean-boolean-java.util.List-int-java.util.List-java.util.List-java.lang.Integer-java.lang.Integer-java.util.List-java.util.List-java.util.List-htsjdk.samtools.SAMFileHeader.SortOrder-picard.sam.PrimaryAlignmentSelectionStrategy-boolean-boolean-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SamAlignmentMerger</h4>
<pre>public&nbsp;SamAlignmentMerger(java.io.File&nbsp;unmappedBamFile,
                          java.io.File&nbsp;targetBamFile,
                          java.io.File&nbsp;referenceFasta,
                          htsjdk.samtools.SAMProgramRecord&nbsp;programRecord,
                          boolean&nbsp;clipAdapters,
                          boolean&nbsp;bisulfiteSequence,
                          boolean&nbsp;alignedReadsOnly,
                          java.util.List&lt;java.io.File&gt;&nbsp;alignedSamFile,
                          int&nbsp;maxGaps,
                          java.util.List&lt;java.lang.String&gt;&nbsp;attributesToRetain,
                          java.util.List&lt;java.lang.String&gt;&nbsp;attributesToRemove,
                          java.lang.Integer&nbsp;read1BasesTrimmed,
                          java.lang.Integer&nbsp;read2BasesTrimmed,
                          java.util.List&lt;java.io.File&gt;&nbsp;read1AlignedSamFile,
                          java.util.List&lt;java.io.File&gt;&nbsp;read2AlignedSamFile,
                          java.util.List&lt;htsjdk.samtools.SamPairUtil.PairOrientation&gt;&nbsp;expectedOrientations,
                          htsjdk.samtools.SAMFileHeader.SortOrder&nbsp;sortOrder,
                          <a href="../../picard/sam/PrimaryAlignmentSelectionStrategy.html" title="interface in picard.sam">PrimaryAlignmentSelectionStrategy</a>&nbsp;primaryAlignmentSelectionStrategy,
                          boolean&nbsp;addMateCigar,
                          boolean&nbsp;unmapContaminantReads,
                          int&nbsp;minUnclippedBases)</pre>
<div class="block">Constructor with a default value for unmappingReadStrategy</div>
</li>
</ul>
<a name="SamAlignmentMerger-java.io.File-java.io.File-java.io.File-htsjdk.samtools.SAMProgramRecord-boolean-boolean-boolean-java.util.List-int-java.util.List-java.util.List-java.lang.Integer-java.lang.Integer-java.util.List-java.util.List-java.util.List-htsjdk.samtools.SAMFileHeader.SortOrder-picard.sam.PrimaryAlignmentSelectionStrategy-boolean-boolean-int-picard.sam.AbstractAlignmentMerger.UnmappingReadStrategy-java.util.List-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>SamAlignmentMerger</h4>
<pre>public&nbsp;SamAlignmentMerger(java.io.File&nbsp;unmappedBamFile,
                          java.io.File&nbsp;targetBamFile,
                          java.io.File&nbsp;referenceFasta,
                          htsjdk.samtools.SAMProgramRecord&nbsp;programRecord,
                          boolean&nbsp;clipAdapters,
                          boolean&nbsp;bisulfiteSequence,
                          boolean&nbsp;alignedReadsOnly,
                          java.util.List&lt;java.io.File&gt;&nbsp;alignedSamFile,
                          int&nbsp;maxGaps,
                          java.util.List&lt;java.lang.String&gt;&nbsp;attributesToRetain,
                          java.util.List&lt;java.lang.String&gt;&nbsp;attributesToRemove,
                          java.lang.Integer&nbsp;read1BasesTrimmed,
                          java.lang.Integer&nbsp;read2BasesTrimmed,
                          java.util.List&lt;java.io.File&gt;&nbsp;read1AlignedSamFile,
                          java.util.List&lt;java.io.File&gt;&nbsp;read2AlignedSamFile,
                          java.util.List&lt;htsjdk.samtools.SamPairUtil.PairOrientation&gt;&nbsp;expectedOrientations,
                          htsjdk.samtools.SAMFileHeader.SortOrder&nbsp;sortOrder,
                          <a href="../../picard/sam/PrimaryAlignmentSelectionStrategy.html" title="interface in picard.sam">PrimaryAlignmentSelectionStrategy</a>&nbsp;primaryAlignmentSelectionStrategy,
                          boolean&nbsp;addMateCigar,
                          boolean&nbsp;unmapContaminantReads,
                          int&nbsp;minUnclippedBases,
                          <a href="../../picard/sam/AbstractAlignmentMerger.UnmappingReadStrategy.html" title="enum in picard.sam">AbstractAlignmentMerger.UnmappingReadStrategy</a>&nbsp;unmappingReadStrategy,
                          java.util.List&lt;java.lang.String&gt;&nbsp;requiredMatchingDictionaryTags)</pre>
<div class="block">Constructor</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>unmappedBamFile</code> - The BAM file that was used as the input to the aligner, which will
                                          include info on all the reads that did not map.  Required.</dd>
<dd><code>targetBamFile</code> - The file to which to write the merged SAM records. Required.</dd>
<dd><code>referenceFasta</code> - The reference sequence for the map files. Required.</dd>
<dd><code>programRecord</code> - Program record for taget file SAMRecords created.</dd>
<dd><code>clipAdapters</code> - Whether adapters marked in unmapped BAM file should be marked as
                                          soft clipped in the merged bam. Required.</dd>
<dd><code>bisulfiteSequence</code> - Whether the reads are bisulfite sequence (used when calculating the
                                          NM and UQ tags). Required.</dd>
<dd><code>alignedReadsOnly</code> - Whether to output only those reads that have alignment data</dd>
<dd><code>alignedSamFile</code> - The SAM file(s) with alignment information.  Optional.  If this is
                                          not provided, then read1AlignedSamFile and read2AlignedSamFile must be.</dd>
<dd><code>maxGaps</code> - The maximum number of insertions or deletions permitted in an
                                          alignment.  Alignments with more than this many gaps will be ignored.
                                          -1 means to allow any number of gaps.</dd>
<dd><code>attributesToRetain</code> - attributes from the alignment record that should be
                                          retained when merging, overridden by attributesToRemove if they share
                                          common tags.</dd>
<dd><code>attributesToRemove</code> - attributes from the alignment record that should be
                                          removed when merging.  This overrides attributesToRetain if they share
                                          common tags.</dd>
<dd><code>read1BasesTrimmed</code> - The number of bases trimmed from start of read 1 prior to alignment.  Optional.</dd>
<dd><code>read2BasesTrimmed</code> - The number of bases trimmed from start of read 2 prior to alignment.  Optional.</dd>
<dd><code>read1AlignedSamFile</code> - The alignment records for read1.  Used when the two ends of a read are
                                          aligned separately.  This is optional, but must be specified if
                                          alignedSamFile is not.</dd>
<dd><code>read2AlignedSamFile</code> - The alignment records for read1.  Used when the two ends of a read are
                                          aligned separately.  This is optional, but must be specified if
                                          alignedSamFile is not.</dd>
<dd><code>expectedOrientations</code> - A List of SamPairUtil.PairOrientations that are expected for
                                          aligned pairs.  Used to determine the properPair flag.</dd>
<dd><code>sortOrder</code> - The order in which the merged records should be output.  If null,
                                          output will be coordinate-sorted</dd>
<dd><code>primaryAlignmentSelectionStrategy</code> - How to handle multiple alignments for a fragment or read pair,
                                          in which none are primary, or more than one is marked primary</dd>
<dd><code>addMateCigar</code> - True if we are to add or maintain the mate CIGAR (MC) tag, false if we are to remove or not include.</dd>
<dd><code>unmapContaminantReads</code> - If true, identify reads having the signature of cross-species contamination (i.e. mostly clipped bases),
                                          and mark them as unmapped.</dd>
<dd><code>minUnclippedBases</code> - If unmapContaminantReads is set, require this many unclipped bases or else the read will be marked as contaminant.</dd>
<dd><code>unmappingReadStrategy</code> - An enum describing how to deal with reads whose mapping information are being removed (currently this happens due to cross-species
                                          contamination). Ignored unless unmapContaminantReads is true.</dd>
<dd><code>requiredMatchingDictionaryTags</code> - A list of SAMSequenceRecord tags that must be equal (if present) in the aligned bam and the reference dictionary.
                                          Program will issue a warning about other tags, if present in both files and are different.</dd>
</dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="mergeAlignment-java.io.File-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>mergeAlignment</h4>
<pre>public&nbsp;void&nbsp;mergeAlignment(java.io.File&nbsp;referenceFasta)</pre>
<div class="block">Merges the alignment from the map file with the non-aligned records from the source BAM file.
 Overrides mergeAlignment in AbstractAlignmentMerger.  Tries first to proceed on the assumption
 that the alignment records are pre-sorted.  If not, catches the exception, forces a sort, and
 tries again.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code><a href="../../picard/sam/AbstractAlignmentMerger.html#mergeAlignment-java.io.File-">mergeAlignment</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/sam/AbstractAlignmentMerger.html" title="class in picard.sam">AbstractAlignmentMerger</a></code></dd>
</dl>
</li>
</ul>
<a name="getDictionaryForMergedBam--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDictionaryForMergedBam</h4>
<pre>protected&nbsp;htsjdk.samtools.SAMSequenceDictionary&nbsp;getDictionaryForMergedBam()</pre>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../picard/sam/AbstractAlignmentMerger.html#getDictionaryForMergedBam--">getDictionaryForMergedBam</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/sam/AbstractAlignmentMerger.html" title="class in picard.sam">AbstractAlignmentMerger</a></code></dd>
</dl>
</li>
</ul>
<a name="getQuerynameSortedAlignedRecords--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getQuerynameSortedAlignedRecords</h4>
<pre>protected&nbsp;htsjdk.samtools.util.CloseableIterator&lt;htsjdk.samtools.SAMRecord&gt;&nbsp;getQuerynameSortedAlignedRecords()</pre>
<div class="block">Reads the aligned SAM records into a SortingCollection and returns an iterator over that collection</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../picard/sam/AbstractAlignmentMerger.html#getQuerynameSortedAlignedRecords--">getQuerynameSortedAlignedRecords</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/sam/AbstractAlignmentMerger.html" title="class in picard.sam">AbstractAlignmentMerger</a></code></dd>
</dl>
</li>
</ul>
<a name="ignoreAlignment-htsjdk.samtools.SAMRecord-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>ignoreAlignment</h4>
<pre>protected&nbsp;boolean&nbsp;ignoreAlignment(htsjdk.samtools.SAMRecord&nbsp;sam)</pre>
<div class="block">For now, we ignore only those alignments that have more than <a href="../../picard/sam/SamAlignmentMerger.html#maxGaps"><code>maxGaps</code></a> insertions
 or deletions.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code><a href="../../picard/sam/AbstractAlignmentMerger.html#ignoreAlignment-htsjdk.samtools.SAMRecord-">ignoreAlignment</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/sam/AbstractAlignmentMerger.html" title="class in picard.sam">AbstractAlignmentMerger</a></code></dd>
</dl>
</li>
</ul>
<a name="isContaminant-picard.sam.HitsForInsert-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isContaminant</h4>
<pre>protected&nbsp;boolean&nbsp;isContaminant(picard.sam.HitsForInsert&nbsp;hits)</pre>
<div class="block">Criteria for contaminant reads:
 1. primary alignment has fewer than minUnclippedBases unclipped bases
 2. primary alignment has both ends clipped</div>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code><a href="../../picard/sam/AbstractAlignmentMerger.html#isContaminant-picard.sam.HitsForInsert-">isContaminant</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/sam/AbstractAlignmentMerger.html" title="class in picard.sam">AbstractAlignmentMerger</a></code></dd>
</dl>
</li>
</ul>
<a name="getForceSort--">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getForceSort</h4>
<pre>public&nbsp;boolean&nbsp;getForceSort()</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/sam/RevertSam.FileType.html" title="enum in picard.sam"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/sam/SamComparisonMetric.html" title="class in picard.sam"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/sam/SamAlignmentMerger.html" target="_top">Frames</a></li>
<li><a href="SamAlignmentMerger.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.classes.inherited.from.class.picard.sam.AbstractAlignmentMerger">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#fields.inherited.from.class.picard.sam.AbstractAlignmentMerger">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
