<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_201) on Fri Oct 04 10:35:34 EDT 2019 -->
<title>RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector (picard 2.21.1-SNAPSHOT API)</title>
<meta name="date" content="2019-10-04">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector (picard 2.21.1-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-all.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../picard/analysis/directed/RnaSeqMetricsCollector.html" title="class in picard.analysis.directed"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../../picard/analysis/directed/RnaSeqMetricsCollector.StrandSpecificity.html" title="enum in picard.analysis.directed"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?picard/analysis/directed/RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector.html" target="_top">Frames</a></li>
<li><a href="RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">picard.analysis.directed</div>
<h2 title="Class RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector" class="title">Class RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>picard.analysis.directed.RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../picard/metrics/PerUnitMetricCollector.html" title="interface in picard.metrics">PerUnitMetricCollector</a>&lt;<a href="../../../picard/analysis/RnaSeqMetrics.html" title="class in picard.analysis">RnaSeqMetrics</a>,java.lang.Integer,htsjdk.samtools.SAMRecord&gt;</dd>
</dl>
<dl>
<dt>Enclosing class:</dt>
<dd><a href="../../../picard/analysis/directed/RnaSeqMetricsCollector.html" title="class in picard.analysis.directed">RnaSeqMetricsCollector</a></dd>
</dl>
<hr>
<br>
<pre>protected class <span class="typeNameLabel">RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector</span>
extends java.lang.Object
implements <a href="../../../picard/metrics/PerUnitMetricCollector.html" title="interface in picard.metrics">PerUnitMetricCollector</a>&lt;<a href="../../../picard/analysis/RnaSeqMetrics.html" title="class in picard.analysis">RnaSeqMetrics</a>,java.lang.Integer,htsjdk.samtools.SAMRecord&gt;</pre>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../picard/analysis/RnaSeqMetrics.html" title="class in picard.analysis">RnaSeqMetrics</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/analysis/directed/RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector.html#metrics">metrics</a></span></code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier</th>
<th class="colLast" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected </code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/analysis/directed/RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector.html#PerUnitRnaSeqMetricsCollector-picard.analysis.RnaSeqMetrics-java.lang.String-java.lang.String-java.lang.String-java.lang.Long-">PerUnitRnaSeqMetricsCollector</a></span>(<a href="../../../picard/analysis/RnaSeqMetrics.html" title="class in picard.analysis">RnaSeqMetrics</a>&nbsp;metrics,
                             java.lang.String&nbsp;sample,
                             java.lang.String&nbsp;library,
                             java.lang.String&nbsp;readGroup,
                             java.lang.Long&nbsp;ribosomalBasesInitialValue)</code>
<div class="block">Derived classes that need to capture some additional metrics can use this ctor to supply a metrics instance
 that is a subclass of RnaSeqMetrics.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/analysis/directed/RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector.html#PerUnitRnaSeqMetricsCollector-java.lang.String-java.lang.String-java.lang.String-java.lang.Long-">PerUnitRnaSeqMetricsCollector</a></span>(java.lang.String&nbsp;sample,
                             java.lang.String&nbsp;library,
                             java.lang.String&nbsp;readGroup,
                             java.lang.Long&nbsp;ribosomalBasesInitialValue)</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/analysis/directed/RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector.html#acceptRecord-htsjdk.samtools.SAMRecord-">acceptRecord</a></span>(htsjdk.samtools.SAMRecord&nbsp;rec)</code>
<div class="block">Add a SAMRecord (with ReferenceSequence and Read Group info) to the metric(s) being calculated)</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/analysis/directed/RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector.html#addMetricsToFile-htsjdk.samtools.metrics.MetricsFile-">addMetricsToFile</a></span>(htsjdk.samtools.metrics.MetricsFile&lt;<a href="../../../picard/analysis/RnaSeqMetrics.html" title="class in picard.analysis">RnaSeqMetrics</a>,java.lang.Integer&gt;&nbsp;file)</code>
<div class="block">Any metrics collected will be added to the metric file provided.</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/analysis/directed/RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector.html#finish--">finish</a></span>()</code>
<div class="block">When all records have been collected, compute any final values needed to finish constructing metrics/Histogram</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>protected java.util.Set&lt;<a href="../../../picard/annotation/Gene.html" title="class in picard.annotation">Gene</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/analysis/directed/RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector.html#getGenesForPickTranscripts--">getGenesForPickTranscripts</a></span>()</code>
<div class="block">Derived class may override this method in order to return a different set of genes for
 picking transcripts for computing coverage, or to cache the value returned by geneOverlapDetector.getAll(),
 because it can be expensive to compute repeatedly.</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/analysis/directed/RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector.html#getNumAlignedBases-htsjdk.samtools.SAMRecord-">getNumAlignedBases</a></span>(htsjdk.samtools.SAMRecord&nbsp;rec)</code>&nbsp;</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>java.util.Map&lt;<a href="../../../picard/annotation/Gene.Transcript.html" title="class in picard.annotation">Gene.Transcript</a>,int[]&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/analysis/directed/RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector.html#pickTranscripts-java.util.Map-">pickTranscripts</a></span>(java.util.Map&lt;<a href="../../../picard/annotation/Gene.Transcript.html" title="class in picard.annotation">Gene.Transcript</a>,int[]&gt;&nbsp;transcriptCoverage)</code>
<div class="block">Picks the set of transcripts on which the coverage metrics are to be calculated.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="metrics">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>metrics</h4>
<pre>protected final&nbsp;<a href="../../../picard/analysis/RnaSeqMetrics.html" title="class in picard.analysis">RnaSeqMetrics</a> metrics</pre>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="PerUnitRnaSeqMetricsCollector-picard.analysis.RnaSeqMetrics-java.lang.String-java.lang.String-java.lang.String-java.lang.Long-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PerUnitRnaSeqMetricsCollector</h4>
<pre>protected&nbsp;PerUnitRnaSeqMetricsCollector(<a href="../../../picard/analysis/RnaSeqMetrics.html" title="class in picard.analysis">RnaSeqMetrics</a>&nbsp;metrics,
                                        java.lang.String&nbsp;sample,
                                        java.lang.String&nbsp;library,
                                        java.lang.String&nbsp;readGroup,
                                        java.lang.Long&nbsp;ribosomalBasesInitialValue)</pre>
<div class="block">Derived classes that need to capture some additional metrics can use this ctor to supply a metrics instance
 that is a subclass of RnaSeqMetrics.</div>
</li>
</ul>
<a name="PerUnitRnaSeqMetricsCollector-java.lang.String-java.lang.String-java.lang.String-java.lang.Long-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>PerUnitRnaSeqMetricsCollector</h4>
<pre>public&nbsp;PerUnitRnaSeqMetricsCollector(java.lang.String&nbsp;sample,
                                     java.lang.String&nbsp;library,
                                     java.lang.String&nbsp;readGroup,
                                     java.lang.Long&nbsp;ribosomalBasesInitialValue)</pre>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="acceptRecord-htsjdk.samtools.SAMRecord-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>acceptRecord</h4>
<pre>public&nbsp;void&nbsp;acceptRecord(htsjdk.samtools.SAMRecord&nbsp;rec)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../picard/metrics/PerUnitMetricCollector.html#acceptRecord-ARGTYPE-">PerUnitMetricCollector</a></code></span></div>
<div class="block">Add a SAMRecord (with ReferenceSequence and Read Group info) to the metric(s) being calculated)</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../picard/metrics/PerUnitMetricCollector.html#acceptRecord-ARGTYPE-">acceptRecord</a></code>&nbsp;in interface&nbsp;<code><a href="../../../picard/metrics/PerUnitMetricCollector.html" title="interface in picard.metrics">PerUnitMetricCollector</a>&lt;<a href="../../../picard/analysis/RnaSeqMetrics.html" title="class in picard.analysis">RnaSeqMetrics</a>,java.lang.Integer,htsjdk.samtools.SAMRecord&gt;</code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>rec</code> - Contains SAMRecord, SAMReadGroupRecord, ReferenceSequence of current record and any previously
             computed values that might be needed for this class</dd>
</dl>
</li>
</ul>
<a name="getNumAlignedBases-htsjdk.samtools.SAMRecord-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumAlignedBases</h4>
<pre>protected&nbsp;int&nbsp;getNumAlignedBases(htsjdk.samtools.SAMRecord&nbsp;rec)</pre>
</li>
</ul>
<a name="finish--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>finish</h4>
<pre>public&nbsp;void&nbsp;finish()</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../picard/metrics/PerUnitMetricCollector.html#finish--">PerUnitMetricCollector</a></code></span></div>
<div class="block">When all records have been collected, compute any final values needed to finish constructing metrics/Histogram</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../picard/metrics/PerUnitMetricCollector.html#finish--">finish</a></code>&nbsp;in interface&nbsp;<code><a href="../../../picard/metrics/PerUnitMetricCollector.html" title="interface in picard.metrics">PerUnitMetricCollector</a>&lt;<a href="../../../picard/analysis/RnaSeqMetrics.html" title="class in picard.analysis">RnaSeqMetrics</a>,java.lang.Integer,htsjdk.samtools.SAMRecord&gt;</code></dd>
</dl>
</li>
</ul>
<a name="addMetricsToFile-htsjdk.samtools.metrics.MetricsFile-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addMetricsToFile</h4>
<pre>public&nbsp;void&nbsp;addMetricsToFile(htsjdk.samtools.metrics.MetricsFile&lt;<a href="../../../picard/analysis/RnaSeqMetrics.html" title="class in picard.analysis">RnaSeqMetrics</a>,java.lang.Integer&gt;&nbsp;file)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../picard/metrics/PerUnitMetricCollector.html#addMetricsToFile-htsjdk.samtools.metrics.MetricsFile-">PerUnitMetricCollector</a></code></span></div>
<div class="block">Any metrics collected will be added to the metric file provided.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../picard/metrics/PerUnitMetricCollector.html#addMetricsToFile-htsjdk.samtools.metrics.MetricsFile-">addMetricsToFile</a></code>&nbsp;in interface&nbsp;<code><a href="../../../picard/metrics/PerUnitMetricCollector.html" title="interface in picard.metrics">PerUnitMetricCollector</a>&lt;<a href="../../../picard/analysis/RnaSeqMetrics.html" title="class in picard.analysis">RnaSeqMetrics</a>,java.lang.Integer,htsjdk.samtools.SAMRecord&gt;</code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>file</code> - MetricsFile to which all metrics created by this collector should be added</dd>
</dl>
</li>
</ul>
<a name="getGenesForPickTranscripts--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenesForPickTranscripts</h4>
<pre>protected&nbsp;java.util.Set&lt;<a href="../../../picard/annotation/Gene.html" title="class in picard.annotation">Gene</a>&gt;&nbsp;getGenesForPickTranscripts()</pre>
<div class="block">Derived class may override this method in order to return a different set of genes for
 picking transcripts for computing coverage, or to cache the value returned by geneOverlapDetector.getAll(),
 because it can be expensive to compute repeatedly.</div>
</li>
</ul>
<a name="pickTranscripts-java.util.Map-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>pickTranscripts</h4>
<pre>public&nbsp;java.util.Map&lt;<a href="../../../picard/annotation/Gene.Transcript.html" title="class in picard.annotation">Gene.Transcript</a>,int[]&gt;&nbsp;pickTranscripts(java.util.Map&lt;<a href="../../../picard/annotation/Gene.Transcript.html" title="class in picard.annotation">Gene.Transcript</a>,int[]&gt;&nbsp;transcriptCoverage)</pre>
<div class="block">Picks the set of transcripts on which the coverage metrics are to be calculated.</div>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-all.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../picard/analysis/directed/RnaSeqMetricsCollector.html" title="class in picard.analysis.directed"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../../picard/analysis/directed/RnaSeqMetricsCollector.StrandSpecificity.html" title="enum in picard.analysis.directed"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?picard/analysis/directed/RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector.html" target="_top">Frames</a></li>
<li><a href="RnaSeqMetricsCollector.PerUnitRnaSeqMetricsCollector.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
