<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_201) on Fri Oct 04 10:35:33 EDT 2019 -->
<title>GenotypeConcordanceSummaryMetrics (picard 2.21.1-SNAPSHOT API)</title>
<meta name="date" content="2019-10-04">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="GenotypeConcordanceSummaryMetrics (picard 2.21.1-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/vcf/GenotypeConcordanceStates.TruthState.html" title="enum in picard.vcf"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/vcf/GvcfMetricAccumulator.html" title="class in picard.vcf"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/vcf/GenotypeConcordanceSummaryMetrics.html" target="_top">Frames</a></li>
<li><a href="GenotypeConcordanceSummaryMetrics.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#methods.inherited.from.class.htsjdk.samtools.metrics.MetricBase">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li>Method</li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">picard.vcf</div>
<h2 title="Class GenotypeConcordanceSummaryMetrics" class="title">Class GenotypeConcordanceSummaryMetrics</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>htsjdk.samtools.metrics.MetricBase</li>
<li>
<ul class="inheritance">
<li>picard.vcf.GenotypeConcordanceSummaryMetrics</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public class <span class="typeNameLabel">GenotypeConcordanceSummaryMetrics</span>
extends htsjdk.samtools.metrics.MetricBase</pre>
<div class="block">Class that holds summary metrics about Genotype Concordance</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordanceSummaryMetrics.html#CALL_SAMPLE">CALL_SAMPLE</a></span></code>
<div class="block">The name of the 'call' sample</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordanceSummaryMetrics.html#GENOTYPE_CONCORDANCE">GENOTYPE_CONCORDANCE</a></span></code>
<div class="block">The genotype concordance for all possible states.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordanceSummaryMetrics.html#HET_PPV">HET_PPV</a></span></code>
<div class="block">The ppv (positive predictive value) for all heterozygous variants (PPV is the TP / (TP + FP))</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordanceSummaryMetrics.html#HET_SENSITIVITY">HET_SENSITIVITY</a></span></code>
<div class="block">The sensitivity for all heterozygous variants (Sensitivity is TP / (TP + FN))</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordanceSummaryMetrics.html#HET_SPECIFICITY">HET_SPECIFICITY</a></span></code>
<div class="block">The specificity for all heterozygous variants cannot be calculated</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordanceSummaryMetrics.html#HOMVAR_PPV">HOMVAR_PPV</a></span></code>
<div class="block">The ppv (positive predictive value) for all homozygous variants (PPV is the TP / (TP + FP))</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordanceSummaryMetrics.html#HOMVAR_SENSITIVITY">HOMVAR_SENSITIVITY</a></span></code>
<div class="block">The sensitivity for all homozygous variants (Sensitivity is TP / (TP + FN))</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordanceSummaryMetrics.html#HOMVAR_SPECIFICITY">HOMVAR_SPECIFICITY</a></span></code>
<div class="block">The specificity for all homozygous variants cannot be calculated.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordanceSummaryMetrics.html#NON_REF_GENOTYPE_CONCORDANCE">NON_REF_GENOTYPE_CONCORDANCE</a></span></code>
<div class="block">The non-ref genotype concordance, ie for all var states only.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordanceSummaryMetrics.html#TRUTH_SAMPLE">TRUTH_SAMPLE</a></span></code>
<div class="block">The name of the 'truth' sample</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordanceSummaryMetrics.html#VAR_PPV">VAR_PPV</a></span></code>
<div class="block">The ppv (positive predictive value) for all (heterozygous and homozygous) variants (PPV is the TP / (TP + FP))</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordanceSummaryMetrics.html#VAR_SENSITIVITY">VAR_SENSITIVITY</a></span></code>
<div class="block">The sensitivity for all (heterozygous and homozygous) variants (Sensitivity is TP / (TP + FN))</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordanceSummaryMetrics.html#VAR_SPECIFICITY">VAR_SPECIFICITY</a></span></code>
<div class="block">The specificity for all (heterozygous and homozygous) variants (Specificity is TN / (FP + TN))</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>htsjdk.variant.variantcontext.VariantContext.Type</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordanceSummaryMetrics.html#VARIANT_TYPE">VARIANT_TYPE</a></span></code>
<div class="block">The type of the event (i.e.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordanceSummaryMetrics.html#GenotypeConcordanceSummaryMetrics--">GenotypeConcordanceSummaryMetrics</a></span>()</code>
<div class="block">Empty constructor - needed for unit tests</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordanceSummaryMetrics.html#GenotypeConcordanceSummaryMetrics-htsjdk.variant.variantcontext.VariantContext.Type-picard.vcf.GenotypeConcordanceCounts-java.lang.String-java.lang.String-boolean-">GenotypeConcordanceSummaryMetrics</a></span>(htsjdk.variant.variantcontext.VariantContext.Type&nbsp;variantType,
                                 <a href="../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&nbsp;concordanceCounts,
                                 java.lang.String&nbsp;truthSample,
                                 java.lang.String&nbsp;callSample,
                                 boolean&nbsp;missingSitesFlag)</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.htsjdk.samtools.metrics.MetricBase">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;htsjdk.samtools.metrics.MetricBase</h3>
<code>equals, hashCode, toString</code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, finalize, getClass, notify, notifyAll, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="VARIANT_TYPE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>VARIANT_TYPE</h4>
<pre>public&nbsp;htsjdk.variant.variantcontext.VariantContext.Type VARIANT_TYPE</pre>
<div class="block">The type of the event (i.e. either SNP or INDEL)</div>
</li>
</ul>
<a name="TRUTH_SAMPLE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TRUTH_SAMPLE</h4>
<pre>public&nbsp;java.lang.String TRUTH_SAMPLE</pre>
<div class="block">The name of the 'truth' sample</div>
</li>
</ul>
<a name="CALL_SAMPLE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>CALL_SAMPLE</h4>
<pre>public&nbsp;java.lang.String CALL_SAMPLE</pre>
<div class="block">The name of the 'call' sample</div>
</li>
</ul>
<a name="HET_SENSITIVITY">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>HET_SENSITIVITY</h4>
<pre>public&nbsp;double HET_SENSITIVITY</pre>
<div class="block">The sensitivity for all heterozygous variants (Sensitivity is TP / (TP + FN))</div>
</li>
</ul>
<a name="HET_PPV">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>HET_PPV</h4>
<pre>public&nbsp;double HET_PPV</pre>
<div class="block">The ppv (positive predictive value) for all heterozygous variants (PPV is the TP / (TP + FP))</div>
</li>
</ul>
<a name="HET_SPECIFICITY">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>HET_SPECIFICITY</h4>
<pre>public&nbsp;double HET_SPECIFICITY</pre>
<div class="block">The specificity for all heterozygous variants cannot be calculated</div>
</li>
</ul>
<a name="HOMVAR_SENSITIVITY">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>HOMVAR_SENSITIVITY</h4>
<pre>public&nbsp;double HOMVAR_SENSITIVITY</pre>
<div class="block">The sensitivity for all homozygous variants (Sensitivity is TP / (TP + FN))</div>
</li>
</ul>
<a name="HOMVAR_PPV">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>HOMVAR_PPV</h4>
<pre>public&nbsp;double HOMVAR_PPV</pre>
<div class="block">The ppv (positive predictive value) for all homozygous variants (PPV is the TP / (TP + FP))</div>
</li>
</ul>
<a name="HOMVAR_SPECIFICITY">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>HOMVAR_SPECIFICITY</h4>
<pre>public&nbsp;double HOMVAR_SPECIFICITY</pre>
<div class="block">The specificity for all homozygous variants cannot be calculated.</div>
</li>
</ul>
<a name="VAR_SENSITIVITY">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>VAR_SENSITIVITY</h4>
<pre>public&nbsp;double VAR_SENSITIVITY</pre>
<div class="block">The sensitivity for all (heterozygous and homozygous) variants (Sensitivity is TP / (TP + FN))</div>
</li>
</ul>
<a name="VAR_PPV">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>VAR_PPV</h4>
<pre>public&nbsp;double VAR_PPV</pre>
<div class="block">The ppv (positive predictive value) for all (heterozygous and homozygous) variants (PPV is the TP / (TP + FP))</div>
</li>
</ul>
<a name="VAR_SPECIFICITY">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>VAR_SPECIFICITY</h4>
<pre>public&nbsp;double VAR_SPECIFICITY</pre>
<div class="block">The specificity for all (heterozygous and homozygous) variants (Specificity is TN / (FP + TN))</div>
</li>
</ul>
<a name="GENOTYPE_CONCORDANCE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>GENOTYPE_CONCORDANCE</h4>
<pre>public&nbsp;double GENOTYPE_CONCORDANCE</pre>
<div class="block">The genotype concordance for all possible states.
  Genotype Concordance is the number of times the truth and call states match exactly / all truth and call combinations made</div>
</li>
</ul>
<a name="NON_REF_GENOTYPE_CONCORDANCE">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>NON_REF_GENOTYPE_CONCORDANCE</h4>
<pre>public&nbsp;double NON_REF_GENOTYPE_CONCORDANCE</pre>
<div class="block">The non-ref genotype concordance, ie for all var states only.
 Non Ref Genotype Concordance is the number of times the truth and call states match exactly for *vars only* / all truth and call *var* combinations made</div>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="GenotypeConcordanceSummaryMetrics--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>GenotypeConcordanceSummaryMetrics</h4>
<pre>public&nbsp;GenotypeConcordanceSummaryMetrics()</pre>
<div class="block">Empty constructor - needed for unit tests</div>
</li>
</ul>
<a name="GenotypeConcordanceSummaryMetrics-htsjdk.variant.variantcontext.VariantContext.Type-picard.vcf.GenotypeConcordanceCounts-java.lang.String-java.lang.String-boolean-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>GenotypeConcordanceSummaryMetrics</h4>
<pre>public&nbsp;GenotypeConcordanceSummaryMetrics(htsjdk.variant.variantcontext.VariantContext.Type&nbsp;variantType,
                                         <a href="../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&nbsp;concordanceCounts,
                                         java.lang.String&nbsp;truthSample,
                                         java.lang.String&nbsp;callSample,
                                         boolean&nbsp;missingSitesFlag)</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/vcf/GenotypeConcordanceStates.TruthState.html" title="enum in picard.vcf"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/vcf/GvcfMetricAccumulator.html" title="class in picard.vcf"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/vcf/GenotypeConcordanceSummaryMetrics.html" target="_top">Frames</a></li>
<li><a href="GenotypeConcordanceSummaryMetrics.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#methods.inherited.from.class.htsjdk.samtools.metrics.MetricBase">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li>Method</li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
