<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_201) on Fri Oct 04 10:35:33 EDT 2019 -->
<title>GenotypeConcordance (picard 2.21.1-SNAPSHOT API)</title>
<meta name="date" content="2019-10-04">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="GenotypeConcordance (picard 2.21.1-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":9,"i1":9,"i2":9,"i3":10,"i4":9,"i5":10,"i6":10,"i7":10,"i8":9,"i9":9};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/vcf/GatherVcfs.html" title="class in picard.vcf"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/vcf/GenotypeConcordance.Alleles.html" title="class in picard.vcf"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/vcf/GenotypeConcordance.html" target="_top">Frames</a></li>
<li><a href="GenotypeConcordance.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">picard.vcf</div>
<h2 title="Class GenotypeConcordance" class="title">Class GenotypeConcordance</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li><a href="../../picard/cmdline/CommandLineProgram.html" title="class in picard.cmdline">picard.cmdline.CommandLineProgram</a></li>
<li>
<ul class="inheritance">
<li>picard.vcf.GenotypeConcordance</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>@DocumentedFeature
public class <span class="typeNameLabel">GenotypeConcordance</span>
extends <a href="../../picard/cmdline/CommandLineProgram.html" title="class in picard.cmdline">CommandLineProgram</a></pre>
<div class="block"><h3>Summary</h3>
 Calculates the concordance between genotype data of one sample in each of two VCFs - one being considered the truth (or reference)
 the other being the call.  The concordance is broken into separate results sections for SNPs and indels.  Satistics
 are reported in three different files.

 <h3>Details</h3>
 This tool evaluates the concordance between genotype calls for a sample in different callsets where one is being considered
 as the \"truth\" (aka standard, or reference) and the other as the \"call\" that is being evaluated for accuracy. The
 Comparison can be restricted to a confidence interval which is typically used in order to enable proper assessment of
 False Positives and the False-Positive Rate (FPR).
 <br />
 <h3>Usage example</h3>
 <h4>Compare two VCFs within a confidence region</h4>
 <pre>
 java -jar picard.jar GenotypeConcordance \\
       CALL_VCF=input.vcf \\
       CALL_SAMPLE=sample_name \\
       O=gc_concordance.vcf \\
       TRUTH_VCF=truth_set.vcf \\
       TRUTH_SAMPLE=sample_in_truth \\
       INTERVALS=confident.interval_list \\
       MISSING_SITES_HOM_REF = true
 </pre>

 <h3>Output Metrics:</h3>
 Output metrics consists of GenotypeConcordanceContingencyMetrics, GenotypeConcordanceSummaryMetrics, and
 GenotypeConcordanceDetailMetrics.  For each set of metrics, the data is broken into separate sections for
 SNPs and INDELs.  Note that only SNP and INDEL variants are considered, MNP, Symbolic, and Mixed classes
  of variants are not included.
 <ul>
 <li><a href="../../picard/vcf/GenotypeConcordanceContingencyMetrics.html" title="class in picard.vcf"><code>GenotypeConcordanceContingencyMetrics</code></a> enumerate the constituents of each contingent in a callset including true-positive
 (TP), true-negative (TN), false-positive (FP), and false-negative (FN) calls.</li>
 <li><a href="../../picard/vcf/GenotypeConcordanceDetailMetrics.html" title="class in picard.vcf"><code>GenotypeConcordanceDetailMetrics</code></a> include the numbers of SNPs and INDELs for each contingent genotype as well as the
 number of validated genotypes.</li>
 <li><a href="../../picard/vcf/GenotypeConcordanceSummaryMetrics.html" title="class in picard.vcf"><code>GenotypeConcordanceSummaryMetrics</code></a> provide specific details for the variant caller performance on a callset including
 values for sensitivity, specificity, and positive predictive values.</li>
 </ul>
 <br />
 <br />
 Useful definitions applicable to alleles and genotypes:
 <ul>
 <li>Truthset - A callset (typically in VCF format) containing variant calls and genotypes that have been cross-validated
 with multiple technologies e.g. Genome In A Bottle Consortium (GIAB) (https://sites.stanford.edu/abms/giab)</li>
 <li>TP - True-positives are variant sites that match against the truth-set</li>
 <li>FP - False-positives are reference sites miscalled as variant</li>
 <li>FN - False-negatives are variant sites miscalled as reference</li>
 <li>TN - True-negatives are correctly called as reference</li>
 <li>Validated genotypes - are TP sites where the exact genotype (HET or HOM-VAR) appears in the truth-set</li>
 </ul>
 <h3>VCF Output:</h3>
 <ul>
 <li>The concordance state will be stored in the CONC_ST tag in the INFO field</li>
 <li>The truth sample name will be \"truth\" and call sample name will be \"call\"</li>
 </ul></div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.Alleles.html" title="class in picard.vcf">GenotypeConcordance.Alleles</a></span></code>
<div class="block">A simple structure to return the results of getAlleles.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#CALL_SAMPLE">CALL_SAMPLE</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.io.File</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#CALL_VCF">CALL_VCF</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#CONTINGENCY_METRICS_FILE_EXTENSION">CONTINGENCY_METRICS_FILE_EXTENSION</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static htsjdk.variant.vcf.VCFHeaderLine</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#CONTINGENCY_STATE_HEADER_LINE">CONTINGENCY_STATE_HEADER_LINE</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#CONTINGENCY_STATE_TAG">CONTINGENCY_STATE_TAG</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#DETAILED_METRICS_FILE_EXTENSION">DETAILED_METRICS_FILE_EXTENSION</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#IGNORE_FILTER_STATUS">IGNORE_FILTER_STATUS</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#indelCounter">indelCounter</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#INTERSECT_INTERVALS">INTERSECT_INTERVALS</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.List&lt;java.io.File&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#INTERVALS">INTERVALS</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#MIN_DP">MIN_DP</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#MIN_GQ">MIN_GQ</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#MISSING_SITES_HOM_REF">MISSING_SITES_HOM_REF</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.io.File</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#OUTPUT">OUTPUT</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#OUTPUT_ALL_ROWS">OUTPUT_ALL_ROWS</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#OUTPUT_VCF">OUTPUT_VCF</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#OUTPUT_VCF_CALL_SAMPLE_NAME">OUTPUT_VCF_CALL_SAMPLE_NAME</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#OUTPUT_VCF_FILE_EXTENSION">OUTPUT_VCF_FILE_EXTENSION</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#OUTPUT_VCF_TRUTH_SAMPLE_NAME">OUTPUT_VCF_TRUTH_SAMPLE_NAME</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#snpCounter">snpCounter</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#SUMMARY_METRICS_FILE_EXTENSION">SUMMARY_METRICS_FILE_EXTENSION</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#TRUTH_SAMPLE">TRUTH_SAMPLE</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.io.File</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#TRUTH_VCF">TRUTH_VCF</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#USE_VCF_INDEX">USE_VCF_INDEX</a></span></code>&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="fields.inherited.from.class.picard.cmdline.CommandLineProgram">
<!--   -->
</a>
<h3>Fields inherited from class&nbsp;picard.cmdline.<a href="../../picard/cmdline/CommandLineProgram.html" title="class in picard.cmdline">CommandLineProgram</a></h3>
<code><a href="../../picard/cmdline/CommandLineProgram.html#COMPRESSION_LEVEL">COMPRESSION_LEVEL</a>, <a href="../../picard/cmdline/CommandLineProgram.html#CREATE_INDEX">CREATE_INDEX</a>, <a href="../../picard/cmdline/CommandLineProgram.html#CREATE_MD5_FILE">CREATE_MD5_FILE</a>, <a href="../../picard/cmdline/CommandLineProgram.html#GA4GH_CLIENT_SECRETS">GA4GH_CLIENT_SECRETS</a>, <a href="../../picard/cmdline/CommandLineProgram.html#MAX_ALLOWABLE_ONE_LINE_SUMMARY_LENGTH">MAX_ALLOWABLE_ONE_LINE_SUMMARY_LENGTH</a>, <a href="../../picard/cmdline/CommandLineProgram.html#MAX_RECORDS_IN_RAM">MAX_RECORDS_IN_RAM</a>, <a href="../../picard/cmdline/CommandLineProgram.html#QUIET">QUIET</a>, <a href="../../picard/cmdline/CommandLineProgram.html#REFERENCE_SEQUENCE">REFERENCE_SEQUENCE</a>, <a href="../../picard/cmdline/CommandLineProgram.html#referenceSequence">referenceSequence</a>, <a href="../../picard/cmdline/CommandLineProgram.html#specialArgumentsCollection">specialArgumentsCollection</a>, <a href="../../picard/cmdline/CommandLineProgram.html#TMP_DIR">TMP_DIR</a>, <a href="../../picard/cmdline/CommandLineProgram.html#USE_JDK_DEFLATER">USE_JDK_DEFLATER</a>, <a href="../../picard/cmdline/CommandLineProgram.html#USE_JDK_INFLATER">USE_JDK_INFLATER</a>, <a href="../../picard/cmdline/CommandLineProgram.html#VALIDATION_STRINGENCY">VALIDATION_STRINGENCY</a>, <a href="../../picard/cmdline/CommandLineProgram.html#VERBOSITY">VERBOSITY</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#GenotypeConcordance--">GenotypeConcordance</a></span>()</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#addMissingTruthAndMissingCallStates-double-long-picard.vcf.GenotypeConcordanceCounts-">addMissingTruthAndMissingCallStates</a></span>(double&nbsp;numVariants,
                                   long&nbsp;intervalBaseCount,
                                   <a href="../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&nbsp;counter)</code>
<div class="block">Method to add missing sites that are KNOWN to be HOM_REF in the case of the NIST truth data set.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#classifyVariants-java.util.Optional-java.lang.String-java.util.Optional-java.lang.String-int-int-boolean-">classifyVariants</a></span>(java.util.Optional&lt;htsjdk.variant.variantcontext.VariantContext&gt;&nbsp;truthContext,
                java.lang.String&nbsp;truthSample,
                java.util.Optional&lt;htsjdk.variant.variantcontext.VariantContext&gt;&nbsp;callContext,
                java.lang.String&nbsp;callSample,
                int&nbsp;minGq,
                int&nbsp;minDp,
                boolean&nbsp;ignoreFilterStatus)</code>&nbsp;</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#classifyVariants-java.util.Optional-java.lang.String-java.util.Optional-java.lang.String-java.util.Optional-java.util.Optional-int-int-boolean-">classifyVariants</a></span>(java.util.Optional&lt;htsjdk.variant.variantcontext.VariantContext&gt;&nbsp;truthContext,
                java.lang.String&nbsp;truthSample,
                java.util.Optional&lt;htsjdk.variant.variantcontext.VariantContext&gt;&nbsp;callContext,
                java.lang.String&nbsp;callSample,
                java.util.Optional&lt;<a href="../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&gt;&nbsp;snpCounter,
                java.util.Optional&lt;<a href="../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&gt;&nbsp;indelCounter,
                int&nbsp;minGq,
                int&nbsp;minDp,
                boolean&nbsp;ignoreFilteredStatus)</code>
<div class="block">Attempts to determine the concordance state given the truth and all variant context and optionally increments the genotype concordance
 count for the given variant type (SNP or INDEL).</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>protected java.lang.String[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#customCommandLineValidation--">customCommandLineValidation</a></span>()</code>
<div class="block">Put any custom command-line validation in an override of this method.</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>static <a href="../../picard/vcf/GenotypeConcordanceStates.TruthAndCallStates.html" title="class in picard.vcf">GenotypeConcordanceStates.TruthAndCallStates</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#determineState-htsjdk.variant.variantcontext.VariantContext-java.lang.String-htsjdk.variant.variantcontext.VariantContext-java.lang.String-int-int-java.lang.Boolean-">determineState</a></span>(htsjdk.variant.variantcontext.VariantContext&nbsp;truthContext,
              java.lang.String&nbsp;truthSample,
              htsjdk.variant.variantcontext.VariantContext&nbsp;callContext,
              java.lang.String&nbsp;callSample,
              int&nbsp;minGq,
              int&nbsp;minDp,
              java.lang.Boolean&nbsp;ignoreFilteredStatus)</code>
<div class="block">A method to determine the truth and call states for a pair of variant contexts representing truth and call.</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#doWork--">doWork</a></span>()</code>
<div class="block">Do the work after command line has been parsed.</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code><a href="../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#getIndelCounter--">getIndelCounter</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code><a href="../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#getSnpCounter--">getSnpCounter</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>protected static <a href="../../picard/vcf/GenotypeConcordance.Alleles.html" title="class in picard.vcf">GenotypeConcordance.Alleles</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#normalizeAlleles-htsjdk.variant.variantcontext.VariantContext-java.lang.String-htsjdk.variant.variantcontext.VariantContext-java.lang.String-java.lang.Boolean-">normalizeAlleles</a></span>(htsjdk.variant.variantcontext.VariantContext&nbsp;truthContext,
                java.lang.String&nbsp;truthSample,
                htsjdk.variant.variantcontext.VariantContext&nbsp;callContext,
                java.lang.String&nbsp;callSample,
                java.lang.Boolean&nbsp;ignoreFilteredStatus)</code>
<div class="block">Gets the alleles for the truth and call genotypes.</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/vcf/GenotypeConcordance.html#outputDetailMetricsFile-htsjdk.variant.variantcontext.VariantContext.Type-htsjdk.samtools.metrics.MetricsFile-picard.vcf.GenotypeConcordanceCounts-java.lang.String-java.lang.String-boolean-boolean-">outputDetailMetricsFile</a></span>(htsjdk.variant.variantcontext.VariantContext.Type&nbsp;variantType,
                       htsjdk.samtools.metrics.MetricsFile&lt;<a href="../../picard/vcf/GenotypeConcordanceDetailMetrics.html" title="class in picard.vcf">GenotypeConcordanceDetailMetrics</a>,?&gt;&nbsp;genotypeConcordanceDetailMetricsFile,
                       <a href="../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&nbsp;counter,
                       java.lang.String&nbsp;truthSampleName,
                       java.lang.String&nbsp;callSampleName,
                       boolean&nbsp;missingSitesHomRef,
                       boolean&nbsp;outputAllRows)</code>
<div class="block">Outputs the detailed statistics tables for SNP and Indel match categories.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.picard.cmdline.CommandLineProgram">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;picard.cmdline.<a href="../../picard/cmdline/CommandLineProgram.html" title="class in picard.cmdline">CommandLineProgram</a></h3>
<code><a href="../../picard/cmdline/CommandLineProgram.html#getCommandLine--">getCommandLine</a>, <a href="../../picard/cmdline/CommandLineProgram.html#getCommandLineParser--">getCommandLineParser</a>, <a href="../../picard/cmdline/CommandLineProgram.html#getCommandLineParser-java.lang.Object-">getCommandLineParser</a>, <a href="../../picard/cmdline/CommandLineProgram.html#getDefaultHeaders--">getDefaultHeaders</a>, <a href="../../picard/cmdline/CommandLineProgram.html#getFaqLink--">getFaqLink</a>, <a href="../../picard/cmdline/CommandLineProgram.html#getMetricsFile--">getMetricsFile</a>, <a href="../../picard/cmdline/CommandLineProgram.html#getStandardUsagePreamble--">getStandardUsagePreamble</a>, <a href="../../picard/cmdline/CommandLineProgram.html#getStandardUsagePreamble-java.lang.Class-">getStandardUsagePreamble</a>, <a href="../../picard/cmdline/CommandLineProgram.html#getVersion--">getVersion</a>, <a href="../../picard/cmdline/CommandLineProgram.html#hasWebDocumentation-java.lang.Class-">hasWebDocumentation</a>, <a href="../../picard/cmdline/CommandLineProgram.html#instanceMain-java.lang.String:A-">instanceMain</a>, <a href="../../picard/cmdline/CommandLineProgram.html#instanceMainWithExit-java.lang.String:A-">instanceMainWithExit</a>, <a href="../../picard/cmdline/CommandLineProgram.html#makeReferenceArgumentCollection--">makeReferenceArgumentCollection</a>, <a href="../../picard/cmdline/CommandLineProgram.html#parseArgs-java.lang.String:A-">parseArgs</a>, <a href="../../picard/cmdline/CommandLineProgram.html#requiresReference--">requiresReference</a>, <a href="../../picard/cmdline/CommandLineProgram.html#setDefaultHeaders-java.util.List-">setDefaultHeaders</a>, <a href="../../picard/cmdline/CommandLineProgram.html#useLegacyParser-java.lang.Class-">useLegacyParser</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="TRUTH_VCF">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TRUTH_VCF</h4>
<pre>@Argument(shortName="TV",
          doc="The VCF containing the truth sample")
public&nbsp;java.io.File TRUTH_VCF</pre>
</li>
</ul>
<a name="CALL_VCF">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>CALL_VCF</h4>
<pre>@Argument(shortName="CV",
          doc="The VCF containing the call sample")
public&nbsp;java.io.File CALL_VCF</pre>
</li>
</ul>
<a name="OUTPUT">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>OUTPUT</h4>
<pre>@Argument(shortName="O",
          doc="Basename for the three metrics files that are to be written. Resulting files will be &lt;OUTPUT&gt;.genotype_concordance_summary_metrics, &lt;OUTPUT&gt;.genotype_concordance_detail_metrics, and &lt;OUTPUT&gt;.genotype_concordance_contingency_metrics.")
public&nbsp;java.io.File OUTPUT</pre>
</li>
</ul>
<a name="OUTPUT_VCF">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>OUTPUT_VCF</h4>
<pre>@Argument(doc="Output a VCF annotated with concordance information.")
public&nbsp;boolean OUTPUT_VCF</pre>
</li>
</ul>
<a name="TRUTH_SAMPLE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TRUTH_SAMPLE</h4>
<pre>@Argument(shortName="TS",
          doc="The name of the truth sample within the truth VCF. Not required if only one sample exists.",
          optional=true)
public&nbsp;java.lang.String TRUTH_SAMPLE</pre>
</li>
</ul>
<a name="CALL_SAMPLE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>CALL_SAMPLE</h4>
<pre>@Argument(shortName="CS",
          doc="The name of the call sample within the call VCF. Not required if only one sample exists.",
          optional=true)
public&nbsp;java.lang.String CALL_SAMPLE</pre>
</li>
</ul>
<a name="INTERVALS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>INTERVALS</h4>
<pre>@Argument(doc="One or more interval list files that will be used to limit the genotype concordance.  Note - if intervals are specified, the VCF files must be indexed.",
          optional=true)
public&nbsp;java.util.List&lt;java.io.File&gt; INTERVALS</pre>
</li>
</ul>
<a name="INTERSECT_INTERVALS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>INTERSECT_INTERVALS</h4>
<pre>@Argument(doc="If true, multiple interval lists will be intersected. If false multiple lists will be unioned.")
public&nbsp;boolean INTERSECT_INTERVALS</pre>
</li>
</ul>
<a name="MIN_GQ">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>MIN_GQ</h4>
<pre>@Argument(doc="Genotypes below this genotype quality will have genotypes classified as LowGq.")
public&nbsp;int MIN_GQ</pre>
</li>
</ul>
<a name="MIN_DP">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>MIN_DP</h4>
<pre>@Argument(doc="Genotypes below this depth will have genotypes classified as LowDp.")
public&nbsp;int MIN_DP</pre>
</li>
</ul>
<a name="OUTPUT_ALL_ROWS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>OUTPUT_ALL_ROWS</h4>
<pre>@Argument(doc="If true, output all rows in detailed statistics even when count == 0.  When false only output rows with non-zero counts.")
public&nbsp;boolean OUTPUT_ALL_ROWS</pre>
</li>
</ul>
<a name="USE_VCF_INDEX">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>USE_VCF_INDEX</h4>
<pre>@Argument(doc="If true, use the VCF index, else iterate over the entire VCF.",
          optional=true)
public&nbsp;boolean USE_VCF_INDEX</pre>
</li>
</ul>
<a name="MISSING_SITES_HOM_REF">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>MISSING_SITES_HOM_REF</h4>
<pre>@Argument(shortName="MISSING_HOM",
          doc="Default is false, which follows the GA4GH Scheme. If true, missing sites in the truth set will be treated as HOM_REF sites and sites missing in both the truth and call sets will be true negatives. Useful when hom ref sites are left out of the truth set. This flag can only be used with a high confidence interval list.")
public&nbsp;boolean MISSING_SITES_HOM_REF</pre>
</li>
</ul>
<a name="IGNORE_FILTER_STATUS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>IGNORE_FILTER_STATUS</h4>
<pre>@Argument(doc="Default is false. If true, filter status of sites will be ignored so that we include filtered sites when calculating genotype concordance. ",
          optional=true)
public&nbsp;boolean IGNORE_FILTER_STATUS</pre>
</li>
</ul>
<a name="SUMMARY_METRICS_FILE_EXTENSION">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SUMMARY_METRICS_FILE_EXTENSION</h4>
<pre>public static final&nbsp;java.lang.String SUMMARY_METRICS_FILE_EXTENSION</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../constant-values.html#picard.vcf.GenotypeConcordance.SUMMARY_METRICS_FILE_EXTENSION">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a name="DETAILED_METRICS_FILE_EXTENSION">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DETAILED_METRICS_FILE_EXTENSION</h4>
<pre>public static final&nbsp;java.lang.String DETAILED_METRICS_FILE_EXTENSION</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../constant-values.html#picard.vcf.GenotypeConcordance.DETAILED_METRICS_FILE_EXTENSION">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a name="CONTINGENCY_METRICS_FILE_EXTENSION">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>CONTINGENCY_METRICS_FILE_EXTENSION</h4>
<pre>public static final&nbsp;java.lang.String CONTINGENCY_METRICS_FILE_EXTENSION</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../constant-values.html#picard.vcf.GenotypeConcordance.CONTINGENCY_METRICS_FILE_EXTENSION">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a name="OUTPUT_VCF_FILE_EXTENSION">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>OUTPUT_VCF_FILE_EXTENSION</h4>
<pre>public static final&nbsp;java.lang.String OUTPUT_VCF_FILE_EXTENSION</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../constant-values.html#picard.vcf.GenotypeConcordance.OUTPUT_VCF_FILE_EXTENSION">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a name="snpCounter">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>snpCounter</h4>
<pre>protected&nbsp;<a href="../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a> snpCounter</pre>
</li>
</ul>
<a name="indelCounter">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>indelCounter</h4>
<pre>protected&nbsp;<a href="../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a> indelCounter</pre>
</li>
</ul>
<a name="CONTINGENCY_STATE_TAG">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>CONTINGENCY_STATE_TAG</h4>
<pre>public static final&nbsp;java.lang.String CONTINGENCY_STATE_TAG</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../constant-values.html#picard.vcf.GenotypeConcordance.CONTINGENCY_STATE_TAG">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a name="CONTINGENCY_STATE_HEADER_LINE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>CONTINGENCY_STATE_HEADER_LINE</h4>
<pre>public static final&nbsp;htsjdk.variant.vcf.VCFHeaderLine CONTINGENCY_STATE_HEADER_LINE</pre>
</li>
</ul>
<a name="OUTPUT_VCF_TRUTH_SAMPLE_NAME">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>OUTPUT_VCF_TRUTH_SAMPLE_NAME</h4>
<pre>public static final&nbsp;java.lang.String OUTPUT_VCF_TRUTH_SAMPLE_NAME</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../constant-values.html#picard.vcf.GenotypeConcordance.OUTPUT_VCF_TRUTH_SAMPLE_NAME">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a name="OUTPUT_VCF_CALL_SAMPLE_NAME">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>OUTPUT_VCF_CALL_SAMPLE_NAME</h4>
<pre>public static final&nbsp;java.lang.String OUTPUT_VCF_CALL_SAMPLE_NAME</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../constant-values.html#picard.vcf.GenotypeConcordance.OUTPUT_VCF_CALL_SAMPLE_NAME">Constant Field Values</a></dd>
</dl>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="GenotypeConcordance--">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>GenotypeConcordance</h4>
<pre>public&nbsp;GenotypeConcordance()</pre>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getSnpCounter--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSnpCounter</h4>
<pre>public&nbsp;<a href="../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&nbsp;getSnpCounter()</pre>
</li>
</ul>
<a name="getIndelCounter--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getIndelCounter</h4>
<pre>public&nbsp;<a href="../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&nbsp;getIndelCounter()</pre>
</li>
</ul>
<a name="customCommandLineValidation--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>customCommandLineValidation</h4>
<pre>protected&nbsp;java.lang.String[]&nbsp;customCommandLineValidation()</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from class:&nbsp;<code><a href="../../picard/cmdline/CommandLineProgram.html#customCommandLineValidation--">CommandLineProgram</a></code></span></div>
<div class="block">Put any custom command-line validation in an override of this method.
 clp is initialized at this point and can be used to print usage and access argv.
 Any options set by command-line parser can be validated.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code><a href="../../picard/cmdline/CommandLineProgram.html#customCommandLineValidation--">customCommandLineValidation</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/cmdline/CommandLineProgram.html" title="class in picard.cmdline">CommandLineProgram</a></code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>null if command line is valid.  If command line is invalid, returns an array of error message
 to be written to the appropriate place.</dd>
</dl>
</li>
</ul>
<a name="doWork--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>doWork</h4>
<pre>protected&nbsp;int&nbsp;doWork()</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from class:&nbsp;<code><a href="../../picard/cmdline/CommandLineProgram.html#doWork--">CommandLineProgram</a></code></span></div>
<div class="block">Do the work after command line has been parsed. RuntimeException may be
 thrown by this method, and are reported appropriately.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../picard/cmdline/CommandLineProgram.html#doWork--">doWork</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/cmdline/CommandLineProgram.html" title="class in picard.cmdline">CommandLineProgram</a></code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>program exit status.</dd>
</dl>
</li>
</ul>
<a name="classifyVariants-java.util.Optional-java.lang.String-java.util.Optional-java.lang.String-int-int-boolean-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>classifyVariants</h4>
<pre>public static&nbsp;boolean&nbsp;classifyVariants(java.util.Optional&lt;htsjdk.variant.variantcontext.VariantContext&gt;&nbsp;truthContext,
                                       java.lang.String&nbsp;truthSample,
                                       java.util.Optional&lt;htsjdk.variant.variantcontext.VariantContext&gt;&nbsp;callContext,
                                       java.lang.String&nbsp;callSample,
                                       int&nbsp;minGq,
                                       int&nbsp;minDp,
                                       boolean&nbsp;ignoreFilterStatus)</pre>
</li>
</ul>
<a name="classifyVariants-java.util.Optional-java.lang.String-java.util.Optional-java.lang.String-java.util.Optional-java.util.Optional-int-int-boolean-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>classifyVariants</h4>
<pre>public static&nbsp;boolean&nbsp;classifyVariants(java.util.Optional&lt;htsjdk.variant.variantcontext.VariantContext&gt;&nbsp;truthContext,
                                       java.lang.String&nbsp;truthSample,
                                       java.util.Optional&lt;htsjdk.variant.variantcontext.VariantContext&gt;&nbsp;callContext,
                                       java.lang.String&nbsp;callSample,
                                       java.util.Optional&lt;<a href="../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&gt;&nbsp;snpCounter,
                                       java.util.Optional&lt;<a href="../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&gt;&nbsp;indelCounter,
                                       int&nbsp;minGq,
                                       int&nbsp;minDp,
                                       boolean&nbsp;ignoreFilteredStatus)</pre>
<div class="block">Attempts to determine the concordance state given the truth and all variant context and optionally increments the genotype concordance
 count for the given variant type (SNP or INDEL).  This will ignore cases where an indel was found in the truth and a SNP was found in
 the call, and vice versa.  We typically fail to classify Mixed, Symbolic variants, or MNPs.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>truthContext</code> - A variant context representing truth</dd>
<dd><code>truthSample</code> - The name of the truth sample</dd>
<dd><code>callContext</code> - A variant context representing the call</dd>
<dd><code>callSample</code> - The name of the call sample</dd>
<dd><code>snpCounter</code> - optionally a place to increment the counts for SNP truth/call states</dd>
<dd><code>indelCounter</code> - optionally a place to increment the counts for INDEL truth/call states</dd>
<dd><code>minGq</code> - Threshold for filtering by genotype attribute GQ</dd>
<dd><code>minDp</code> - Threshold for filtering by genotype attribute DP</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>true if the concordance state could be classified.</dd>
</dl>
</li>
</ul>
<a name="addMissingTruthAndMissingCallStates-double-long-picard.vcf.GenotypeConcordanceCounts-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addMissingTruthAndMissingCallStates</h4>
<pre>public static&nbsp;void&nbsp;addMissingTruthAndMissingCallStates(double&nbsp;numVariants,
                                                       long&nbsp;intervalBaseCount,
                                                       <a href="../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&nbsp;counter)</pre>
<div class="block">Method to add missing sites that are KNOWN to be HOM_REF in the case of the NIST truth data set.</div>
</li>
</ul>
<a name="outputDetailMetricsFile-htsjdk.variant.variantcontext.VariantContext.Type-htsjdk.samtools.metrics.MetricsFile-picard.vcf.GenotypeConcordanceCounts-java.lang.String-java.lang.String-boolean-boolean-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>outputDetailMetricsFile</h4>
<pre>public static&nbsp;void&nbsp;outputDetailMetricsFile(htsjdk.variant.variantcontext.VariantContext.Type&nbsp;variantType,
                                           htsjdk.samtools.metrics.MetricsFile&lt;<a href="../../picard/vcf/GenotypeConcordanceDetailMetrics.html" title="class in picard.vcf">GenotypeConcordanceDetailMetrics</a>,?&gt;&nbsp;genotypeConcordanceDetailMetricsFile,
                                           <a href="../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&nbsp;counter,
                                           java.lang.String&nbsp;truthSampleName,
                                           java.lang.String&nbsp;callSampleName,
                                           boolean&nbsp;missingSitesHomRef,
                                           boolean&nbsp;outputAllRows)</pre>
<div class="block">Outputs the detailed statistics tables for SNP and Indel match categories.</div>
</li>
</ul>
<a name="normalizeAlleles-htsjdk.variant.variantcontext.VariantContext-java.lang.String-htsjdk.variant.variantcontext.VariantContext-java.lang.String-java.lang.Boolean-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>normalizeAlleles</h4>
<pre>protected static final&nbsp;<a href="../../picard/vcf/GenotypeConcordance.Alleles.html" title="class in picard.vcf">GenotypeConcordance.Alleles</a>&nbsp;normalizeAlleles(htsjdk.variant.variantcontext.VariantContext&nbsp;truthContext,
                                                                    java.lang.String&nbsp;truthSample,
                                                                    htsjdk.variant.variantcontext.VariantContext&nbsp;callContext,
                                                                    java.lang.String&nbsp;callSample,
                                                                    java.lang.Boolean&nbsp;ignoreFilteredStatus)</pre>
<div class="block">Gets the alleles for the truth and call genotypes.  In particular, this handles the case where indels can have different
 reference alleles.</div>
</li>
</ul>
<a name="determineState-htsjdk.variant.variantcontext.VariantContext-java.lang.String-htsjdk.variant.variantcontext.VariantContext-java.lang.String-int-int-java.lang.Boolean-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>determineState</h4>
<pre>public static final&nbsp;<a href="../../picard/vcf/GenotypeConcordanceStates.TruthAndCallStates.html" title="class in picard.vcf">GenotypeConcordanceStates.TruthAndCallStates</a>&nbsp;determineState(htsjdk.variant.variantcontext.VariantContext&nbsp;truthContext,
                                                                                java.lang.String&nbsp;truthSample,
                                                                                htsjdk.variant.variantcontext.VariantContext&nbsp;callContext,
                                                                                java.lang.String&nbsp;callSample,
                                                                                int&nbsp;minGq,
                                                                                int&nbsp;minDp,
                                                                                java.lang.Boolean&nbsp;ignoreFilteredStatus)</pre>
<div class="block">A method to determine the truth and call states for a pair of variant contexts representing truth and call.
 A variety of variant and genotype-level checks are first used to determine if either of the the variant contexts
 are filtered and after that a comparison of the called genotype alleles to determine appropriate truth and call state

 Note that this method does NOT check for SNP versus Indel.  It is assumed that that check is done by the caller and the results
 of this method are stored by SNP/Indel.
 Note that if a variant context has BOTH GQ and DP less than the specified threshold, then it will be of Truth/Call State LOW_GQ</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>truthContext</code> - A variant context representing truth</dd>
<dd><code>truthSample</code> - The name of the truth sample</dd>
<dd><code>callContext</code> - A variant context representing the call</dd>
<dd><code>callSample</code> - The name of the call sample</dd>
<dd><code>minGq</code> - Threshold for filtering by genotype attribute GQ</dd>
<dd><code>minDp</code> - Threshold for filtering by genotype attribute DP</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>TruthAndCallStates object containing the TruthState and CallState determined here.</dd>
</dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/vcf/GatherVcfs.html" title="class in picard.vcf"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/vcf/GenotypeConcordance.Alleles.html" title="class in picard.vcf"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/vcf/GenotypeConcordance.html" target="_top">Frames</a></li>
<li><a href="GenotypeConcordance.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
