<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_201) on Fri Oct 04 10:35:34 EDT 2019 -->
<title>AlleleSubsettingUtils (picard 2.21.1-SNAPSHOT API)</title>
<meta name="date" content="2019-10-04">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="AlleleSubsettingUtils (picard 2.21.1-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":9,"i1":9,"i2":9,"i3":9,"i4":9};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/util/AdapterPair.html" title="interface in picard.util"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/util/AsyncIterator.html" title="class in picard.util"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/util/AlleleSubsettingUtils.html" target="_top">Frames</a></li>
<li><a href="AlleleSubsettingUtils.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">picard.util</div>
<h2 title="Class AlleleSubsettingUtils" class="title">Class AlleleSubsettingUtils</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>picard.util.AlleleSubsettingUtils</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public final class <span class="typeNameLabel">AlleleSubsettingUtils</span>
extends java.lang.Object</pre>
<div class="block">Utilities class containing methods for restricting <code>VariantContext</code> and <code>GenotypesContext</code> objects to a
 reduced set of alleles, as well as for choosing the best set of alleles to keep and for cleaning up annotations and
 genotypes after subsetting.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../picard/util/AlleleSubsettingUtils.html#AlleleSubsettingUtils--">AlleleSubsettingUtils</a></span>()</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>static htsjdk.variant.variantcontext.GenotypesContext</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/util/AlleleSubsettingUtils.html#subsetAlleles-htsjdk.variant.variantcontext.GenotypesContext-java.util.List-java.util.List-">subsetAlleles</a></span>(htsjdk.variant.variantcontext.GenotypesContext&nbsp;originalGs,
             java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;originalAlleles,
             java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;allelesToKeep)</code>
<div class="block">Create the new GenotypesContext with the subsetted PLs and ADs
 Expects allelesToKeep to be in the same order
 in which they are in originalAlleles.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>static htsjdk.variant.variantcontext.VariantContext</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/util/AlleleSubsettingUtils.html#subsetAlleles-htsjdk.variant.variantcontext.VariantContext-java.util.List-">subsetAlleles</a></span>(htsjdk.variant.variantcontext.VariantContext&nbsp;originalVc,
             java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;allelesToKeep)</code>&nbsp;</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>static int[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/util/AlleleSubsettingUtils.html#subsettedPLIndices-java.util.List-java.util.List-">subsettedPLIndices</a></span>(java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;originalAlleles,
                  java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;newAlleles)</code>
<div class="block">Given a list of original alleles and a subset of new alleles to retain, find the array of old PL indices that correspond
 to new PL indices i.e.</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>static htsjdk.variant.variantcontext.VariantContext</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/util/AlleleSubsettingUtils.html#subsetVCToMatchSnp-htsjdk.variant.variantcontext.VariantContext-picard.fingerprint.Snp-">subsetVCToMatchSnp</a></span>(htsjdk.variant.variantcontext.VariantContext&nbsp;ctx,
                  <a href="../../picard/fingerprint/Snp.html" title="class in picard.fingerprint">Snp</a>&nbsp;snp)</code>
<div class="block">Method to subset the alleles in the VariantContext to those in the input snp.</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>static htsjdk.variant.variantcontext.VariantContext</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/util/AlleleSubsettingUtils.html#swapAlleles-htsjdk.variant.variantcontext.VariantContext-htsjdk.variant.variantcontext.Allele-htsjdk.variant.variantcontext.Allele-">swapAlleles</a></span>(htsjdk.variant.variantcontext.VariantContext&nbsp;originalVc,
           htsjdk.variant.variantcontext.Allele&nbsp;oldAllele,
           htsjdk.variant.variantcontext.Allele&nbsp;newAllele)</code>
<div class="block">Swaps one of the alleles in a VC (and its genotypes) with another.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="AlleleSubsettingUtils--">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>AlleleSubsettingUtils</h4>
<pre>public&nbsp;AlleleSubsettingUtils()</pre>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="subsetVCToMatchSnp-htsjdk.variant.variantcontext.VariantContext-picard.fingerprint.Snp-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>subsetVCToMatchSnp</h4>
<pre>public static&nbsp;htsjdk.variant.variantcontext.VariantContext&nbsp;subsetVCToMatchSnp(htsjdk.variant.variantcontext.VariantContext&nbsp;ctx,
                                                                              <a href="../../picard/fingerprint/Snp.html" title="class in picard.fingerprint">Snp</a>&nbsp;snp)</pre>
<div class="block">Method to subset the alleles in the VariantContext to those in the input snp.
 Will also subset appropriately the AD and PL fields in all the genotypes, and in
 the case of a monomorphic variant with a NON_REF allele, will replace the NON_REF
 allele with the non-reference allele from snp.
 <p>
 returns null if it is not possible to subset ctx to the alleles in snp.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>ctx</code> - The VariantContext to subset.</dd>
<dd><code>snp</code> - Snp whose alleles are used for subsetting ctx.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a VariantContext subsetted to the alleles in snp, or null if unable.</dd>
</dl>
</li>
</ul>
<a name="subsetAlleles-htsjdk.variant.variantcontext.VariantContext-java.util.List-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>subsetAlleles</h4>
<pre>public static&nbsp;htsjdk.variant.variantcontext.VariantContext&nbsp;subsetAlleles(htsjdk.variant.variantcontext.VariantContext&nbsp;originalVc,
                                                                         java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;allelesToKeep)</pre>
</li>
</ul>
<a name="swapAlleles-htsjdk.variant.variantcontext.VariantContext-htsjdk.variant.variantcontext.Allele-htsjdk.variant.variantcontext.Allele-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>swapAlleles</h4>
<pre>public static&nbsp;htsjdk.variant.variantcontext.VariantContext&nbsp;swapAlleles(htsjdk.variant.variantcontext.VariantContext&nbsp;originalVc,
                                                                       htsjdk.variant.variantcontext.Allele&nbsp;oldAllele,
                                                                       htsjdk.variant.variantcontext.Allele&nbsp;newAllele)
                                                                throws java.lang.IllegalArgumentException</pre>
<div class="block">Swaps one of the alleles in a VC (and its genotypes) with another.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>originalVc</code> - The <code>VariantContext</code> whose oldAllele will be swapped for newAllele.</dd>
<dd><code>oldAllele</code> - The <code>Allele</code> in originalVc that needs to be replaced.</dd>
<dd><code>newAllele</code> - The new <code>Allele</code> to use instead of oldAllele.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>A new <code>VariantContext</code> with newAllele swapped in for oldAllele.</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.IllegalArgumentException</code> - if originalVc doesn't contain oldAllele.</dd>
</dl>
</li>
</ul>
<a name="subsetAlleles-htsjdk.variant.variantcontext.GenotypesContext-java.util.List-java.util.List-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>subsetAlleles</h4>
<pre>public static&nbsp;htsjdk.variant.variantcontext.GenotypesContext&nbsp;subsetAlleles(htsjdk.variant.variantcontext.GenotypesContext&nbsp;originalGs,
                                                                           java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;originalAlleles,
                                                                           java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;allelesToKeep)</pre>
<div class="block">Create the new GenotypesContext with the subsetted PLs and ADs
 Expects allelesToKeep to be in the same order
 in which they are in originalAlleles.
 <p></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>originalGs</code> - the original GenotypesContext</dd>
<dd><code>originalAlleles</code> - the original alleles</dd>
<dd><code>allelesToKeep</code> - the subset of alleles to use with the new Genotypes</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a new non-null GenotypesContext</dd>
</dl>
</li>
</ul>
<a name="subsettedPLIndices-java.util.List-java.util.List-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>subsettedPLIndices</h4>
<pre>public static&nbsp;int[]&nbsp;subsettedPLIndices(java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;originalAlleles,
                                       java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;newAlleles)</pre>
<div class="block">Given a list of original alleles and a subset of new alleles to retain, find the array of old PL indices that correspond
 to new PL indices i.e. result[7] = old PL index of genotype containing same alleles as the new genotype with PL index 7.
 <p>
 This method is written in terms f indices rather than subsetting PLs directly in order to produce output that can be
 recycled from sample to sample, provided that the ploidy is the same.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>originalAlleles</code> - List of original alleles</dd>
<dd><code>newAlleles</code> - New alleles -- must be a subset of <code>originalAlleles</code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>old PL indices of new genotypes</dd>
</dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/util/AdapterPair.html" title="interface in picard.util"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/util/AsyncIterator.html" title="class in picard.util"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/util/AlleleSubsettingUtils.html" target="_top">Frames</a></li>
<li><a href="AlleleSubsettingUtils.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
