<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_201) on Fri Oct 04 10:35:34 EDT 2019 -->
<title>LiftoverUtils (picard 2.21.1-SNAPSHOT API)</title>
<meta name="date" content="2019-10-04">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="LiftoverUtils (picard 2.21.1-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":9,"i1":9,"i2":9,"i3":9,"i4":9,"i5":9,"i6":9,"i7":9};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/util/LiftOverIntervalList.html" title="class in picard.util"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/util/MathUtil.html" title="class in picard.util"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/util/LiftoverUtils.html" target="_top">Frames</a></li>
<li><a href="LiftoverUtils.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">picard.util</div>
<h2 title="Class LiftoverUtils" class="title">Class LiftoverUtils</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>picard.util.LiftoverUtils</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public class <span class="typeNameLabel">LiftoverUtils</span>
extends java.lang.Object</pre>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.util.Collection&lt;java.lang.String&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/util/LiftoverUtils.html#DEFAULT_TAGS_TO_DROP">DEFAULT_TAGS_TO_DROP</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.util.Collection&lt;java.lang.String&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/util/LiftoverUtils.html#DEFAULT_TAGS_TO_REVERSE">DEFAULT_TAGS_TO_REVERSE</a></span></code>
<div class="block">Default list of attributes that need to be reversed or dropped from the INFO field when alleles have been swapped.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static htsjdk.samtools.util.Log</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/util/LiftoverUtils.html#log">log</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/util/LiftoverUtils.html#REV_COMPED_ALLELES">REV_COMPED_ALLELES</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/util/LiftoverUtils.html#SWAPPED_ALLELES">SWAPPED_ALLELES</a></span></code>
<div class="block">Attribute used to store the fact that the alt and ref alleles of the variant have been swapped, while all the INFO annotations have not.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../picard/util/LiftoverUtils.html#LiftoverUtils--">LiftoverUtils</a></span>()</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>protected static java.util.List&lt;java.lang.String&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/util/LiftoverUtils.html#allelesToStringList-java.util.List-">allelesToStringList</a></span>(java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;alleles)</code>
<div class="block">This is a utility method that will convert a list of alleles into a list of base strings.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>protected static htsjdk.variant.variantcontext.GenotypesContext</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/util/LiftoverUtils.html#fixGenotypes-htsjdk.variant.variantcontext.GenotypesContext-java.util.List-java.util.List-">fixGenotypes</a></span>(htsjdk.variant.variantcontext.GenotypesContext&nbsp;originals,
            java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;originalAlleles,
            java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;newAlleles)</code>&nbsp;</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>protected static void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/util/LiftoverUtils.html#leftAlignVariant-htsjdk.variant.variantcontext.VariantContextBuilder-int-int-java.util.List-htsjdk.samtools.reference.ReferenceSequence-">leftAlignVariant</a></span>(htsjdk.variant.variantcontext.VariantContextBuilder&nbsp;builder,
                int&nbsp;start,
                int&nbsp;end,
                java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;alleles,
                htsjdk.samtools.reference.ReferenceSequence&nbsp;referenceSequence)</code>
<div class="block">Normalizes and left aligns a <code>VariantContextBuilder</code>.</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>protected static htsjdk.variant.variantcontext.VariantContextBuilder</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/util/LiftoverUtils.html#liftSimpleVariantContext-htsjdk.variant.variantcontext.VariantContext-htsjdk.samtools.util.Interval-">liftSimpleVariantContext</a></span>(htsjdk.variant.variantcontext.VariantContext&nbsp;source,
                        htsjdk.samtools.util.Interval&nbsp;target)</code>&nbsp;</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>static htsjdk.variant.variantcontext.VariantContext</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/util/LiftoverUtils.html#liftVariant-htsjdk.variant.variantcontext.VariantContext-htsjdk.samtools.util.Interval-htsjdk.samtools.reference.ReferenceSequence-boolean-boolean-">liftVariant</a></span>(htsjdk.variant.variantcontext.VariantContext&nbsp;source,
           htsjdk.samtools.util.Interval&nbsp;target,
           htsjdk.samtools.reference.ReferenceSequence&nbsp;refSeq,
           boolean&nbsp;writeOriginalPosition,
           boolean&nbsp;writeOriginalAlleles)</code>
<div class="block">This will take an input VariantContext and lift to the provided interval.</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>protected static boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/util/LiftoverUtils.html#referenceAlleleMatchesReferenceForIndel-java.util.List-htsjdk.samtools.reference.ReferenceSequence-int-int-">referenceAlleleMatchesReferenceForIndel</a></span>(java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;alleles,
                                       htsjdk.samtools.reference.ReferenceSequence&nbsp;referenceSequence,
                                       int&nbsp;start,
                                       int&nbsp;end)</code>
<div class="block">Checks whether the reference allele in the provided variant context actually matches the reference sequence</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>protected static htsjdk.variant.variantcontext.VariantContextBuilder</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/util/LiftoverUtils.html#reverseComplementVariantContext-htsjdk.variant.variantcontext.VariantContext-htsjdk.samtools.util.Interval-htsjdk.samtools.reference.ReferenceSequence-">reverseComplementVariantContext</a></span>(htsjdk.variant.variantcontext.VariantContext&nbsp;source,
                               htsjdk.samtools.util.Interval&nbsp;target,
                               htsjdk.samtools.reference.ReferenceSequence&nbsp;refSeq)</code>&nbsp;</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>static htsjdk.variant.variantcontext.VariantContext</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/util/LiftoverUtils.html#swapRefAlt-htsjdk.variant.variantcontext.VariantContext-java.util.Collection-java.util.Collection-">swapRefAlt</a></span>(htsjdk.variant.variantcontext.VariantContext&nbsp;vc,
          java.util.Collection&lt;java.lang.String&gt;&nbsp;annotationsToReverse,
          java.util.Collection&lt;java.lang.String&gt;&nbsp;annotationsToDrop)</code>
<div class="block">method to swap the reference and alt alleles of a bi-allelic, SNP</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="SWAPPED_ALLELES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SWAPPED_ALLELES</h4>
<pre>public static final&nbsp;java.lang.String SWAPPED_ALLELES</pre>
<div class="block">Attribute used to store the fact that the alt and ref alleles of the variant have been swapped, while all the INFO annotations have not.</div>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../constant-values.html#picard.util.LiftoverUtils.SWAPPED_ALLELES">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a name="REV_COMPED_ALLELES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>REV_COMPED_ALLELES</h4>
<pre>public static final&nbsp;java.lang.String REV_COMPED_ALLELES</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../constant-values.html#picard.util.LiftoverUtils.REV_COMPED_ALLELES">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a name="DEFAULT_TAGS_TO_REVERSE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DEFAULT_TAGS_TO_REVERSE</h4>
<pre>public static final&nbsp;java.util.Collection&lt;java.lang.String&gt; DEFAULT_TAGS_TO_REVERSE</pre>
<div class="block">Default list of attributes that need to be reversed or dropped from the INFO field when alleles have been swapped.</div>
</li>
</ul>
<a name="DEFAULT_TAGS_TO_DROP">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DEFAULT_TAGS_TO_DROP</h4>
<pre>public static final&nbsp;java.util.Collection&lt;java.lang.String&gt; DEFAULT_TAGS_TO_DROP</pre>
</li>
</ul>
<a name="log">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>log</h4>
<pre>public static final&nbsp;htsjdk.samtools.util.Log log</pre>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="LiftoverUtils--">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>LiftoverUtils</h4>
<pre>public&nbsp;LiftoverUtils()</pre>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="liftVariant-htsjdk.variant.variantcontext.VariantContext-htsjdk.samtools.util.Interval-htsjdk.samtools.reference.ReferenceSequence-boolean-boolean-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>liftVariant</h4>
<pre>public static&nbsp;htsjdk.variant.variantcontext.VariantContext&nbsp;liftVariant(htsjdk.variant.variantcontext.VariantContext&nbsp;source,
                                                                       htsjdk.samtools.util.Interval&nbsp;target,
                                                                       htsjdk.samtools.reference.ReferenceSequence&nbsp;refSeq,
                                                                       boolean&nbsp;writeOriginalPosition,
                                                                       boolean&nbsp;writeOriginalAlleles)</pre>
<div class="block">This will take an input VariantContext and lift to the provided interval.
 If this interval is in the opposite orientation, all alleles and genotypes will be reverse complemented
 and indels will be left-aligned.  Currently this is only able to invert biallelic indels, and null will be
 returned for any unsupported VC.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>source</code> - The VariantContext to lift</dd>
<dd><code>target</code> - The target interval</dd>
<dd><code>refSeq</code> - The reference sequence, which should match the target interval</dd>
<dd><code>writeOriginalPosition</code> - If true, INFO field annotations will be added to store the original position and contig</dd>
<dd><code>writeOriginalAlleles</code> - If true, an INFO field annotation will be added to store the original alleles in order.  This can be useful in the case of more complex liftovers, like reverse complements, left aligned indels or swapped REF/ALT</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>The lifted VariantContext.  This will be null if the input VariantContext could not be lifted.</dd>
</dl>
</li>
</ul>
<a name="allelesToStringList-java.util.List-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>allelesToStringList</h4>
<pre>protected static&nbsp;java.util.List&lt;java.lang.String&gt;&nbsp;allelesToStringList(java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;alleles)</pre>
<div class="block">This is a utility method that will convert a list of alleles into a list of base strings.  Reference status
 is ignored when creating these strings (i.e. 'A', not 'A*').  These strings should be sufficient
 to recreate an Allele using Allele.create()</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>alleles</code> - The list of alleles</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>A list of strings representing the bases of the input alleles.</dd>
</dl>
</li>
</ul>
<a name="liftSimpleVariantContext-htsjdk.variant.variantcontext.VariantContext-htsjdk.samtools.util.Interval-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>liftSimpleVariantContext</h4>
<pre>protected static&nbsp;htsjdk.variant.variantcontext.VariantContextBuilder&nbsp;liftSimpleVariantContext(htsjdk.variant.variantcontext.VariantContext&nbsp;source,
                                                                                              htsjdk.samtools.util.Interval&nbsp;target)</pre>
</li>
</ul>
<a name="reverseComplementVariantContext-htsjdk.variant.variantcontext.VariantContext-htsjdk.samtools.util.Interval-htsjdk.samtools.reference.ReferenceSequence-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reverseComplementVariantContext</h4>
<pre>protected static&nbsp;htsjdk.variant.variantcontext.VariantContextBuilder&nbsp;reverseComplementVariantContext(htsjdk.variant.variantcontext.VariantContext&nbsp;source,
                                                                                                     htsjdk.samtools.util.Interval&nbsp;target,
                                                                                                     htsjdk.samtools.reference.ReferenceSequence&nbsp;refSeq)</pre>
</li>
</ul>
<a name="fixGenotypes-htsjdk.variant.variantcontext.GenotypesContext-java.util.List-java.util.List-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fixGenotypes</h4>
<pre>protected static&nbsp;htsjdk.variant.variantcontext.GenotypesContext&nbsp;fixGenotypes(htsjdk.variant.variantcontext.GenotypesContext&nbsp;originals,
                                                                             java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;originalAlleles,
                                                                             java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;newAlleles)</pre>
</li>
</ul>
<a name="swapRefAlt-htsjdk.variant.variantcontext.VariantContext-java.util.Collection-java.util.Collection-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>swapRefAlt</h4>
<pre>public static&nbsp;htsjdk.variant.variantcontext.VariantContext&nbsp;swapRefAlt(htsjdk.variant.variantcontext.VariantContext&nbsp;vc,
                                                                      java.util.Collection&lt;java.lang.String&gt;&nbsp;annotationsToReverse,
                                                                      java.util.Collection&lt;java.lang.String&gt;&nbsp;annotationsToDrop)</pre>
<div class="block">method to swap the reference and alt alleles of a bi-allelic, SNP</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>vc</code> - the <code>VariantContext</code> (bi-allelic SNP) that needs to have it's REF and ALT alleles swapped.</dd>
<dd><code>annotationsToReverse</code> - INFO field annotations (of double value) that will be reversed (x->1-x)</dd>
<dd><code>annotationsToDrop</code> - INFO field annotations that will be dropped from the result since they are invalid when REF and ALT are swapped</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a new <code>VariantContext</code> with alleles swapped, INFO fields modified and in the genotypes, GT, AD and PL corrected appropriately</dd>
</dl>
</li>
</ul>
<a name="referenceAlleleMatchesReferenceForIndel-java.util.List-htsjdk.samtools.reference.ReferenceSequence-int-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>referenceAlleleMatchesReferenceForIndel</h4>
<pre>protected static&nbsp;boolean&nbsp;referenceAlleleMatchesReferenceForIndel(java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;alleles,
                                                                 htsjdk.samtools.reference.ReferenceSequence&nbsp;referenceSequence,
                                                                 int&nbsp;start,
                                                                 int&nbsp;end)</pre>
<div class="block">Checks whether the reference allele in the provided variant context actually matches the reference sequence</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>alleles</code> - list of alleles from which to find the reference allele</dd>
<dd><code>referenceSequence</code> - the ref sequence</dd>
<dd><code>start</code> - the start position of the actual indel</dd>
<dd><code>end</code> - the end position of the actual indel</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>true if they match, false otherwise</dd>
</dl>
</li>
</ul>
<a name="leftAlignVariant-htsjdk.variant.variantcontext.VariantContextBuilder-int-int-java.util.List-htsjdk.samtools.reference.ReferenceSequence-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>leftAlignVariant</h4>
<pre>protected static&nbsp;void&nbsp;leftAlignVariant(htsjdk.variant.variantcontext.VariantContextBuilder&nbsp;builder,
                                       int&nbsp;start,
                                       int&nbsp;end,
                                       java.util.List&lt;htsjdk.variant.variantcontext.Allele&gt;&nbsp;alleles,
                                       htsjdk.samtools.reference.ReferenceSequence&nbsp;referenceSequence)</pre>
<div class="block">Normalizes and left aligns a <code>VariantContextBuilder</code>.
 Note: this will modify the start/stop and alleles of this builder.
 Also note: if the reference allele does not match the reference sequence, this method will throw an exception

 Based on Adrian Tan, Gon&ccedil;alo R. Abecasis and Hyun Min Kang. (2015)
 Unified Representation of Genetic Variants. Bioinformatics.</div>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/util/LiftOverIntervalList.html" title="class in picard.util"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/util/MathUtil.html" title="class in picard.util"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/util/LiftoverUtils.html" target="_top">Frames</a></li>
<li><a href="LiftoverUtils.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
