<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_201) on Fri Oct 04 10:35:35 EDT 2019 -->
<title>picard.analysis (picard 2.21.1-SNAPSHOT API)</title>
<meta name="date" content="2019-10-04">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="picard.analysis (picard 2.21.1-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li class="navBarCell1Rev">Package</li>
<li>Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/package-summary.html">Prev&nbsp;Package</a></li>
<li><a href="../../picard/analysis/artifacts/package-summary.html">Next&nbsp;Package</a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/analysis/package-summary.html" target="_top">Frames</a></li>
<li><a href="package-summary.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h1 title="Package" class="title">Package&nbsp;picard.analysis</h1>
</div>
<div class="contentContainer">
<ul class="blockList">
<li class="blockList">
<table class="typeSummary" border="0" cellpadding="3" cellspacing="0" summary="Interface Summary table, listing interfaces, and an explanation">
<caption><span>Interface Summary</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Interface</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/CollectMultipleMetrics.ProgramInterface.html" title="interface in picard.analysis">CollectMultipleMetrics.ProgramInterface</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/WgsMetricsProcessor.html" title="interface in picard.analysis">WgsMetricsProcessor</a></td>
<td class="colLast">
<div class="block">Interface for processing data and generate result for CollectWgsMetrics</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<table class="typeSummary" border="0" cellpadding="3" cellspacing="0" summary="Class Summary table, listing classes, and an explanation">
<caption><span>Class Summary</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/AbstractWgsMetricsCollector.html" title="class in picard.analysis">AbstractWgsMetricsCollector</a>&lt;T extends htsjdk.samtools.util.AbstractRecordAndOffset&gt;</td>
<td class="colLast">
<div class="block">Class for collecting data on reference coverage, base qualities and excluded bases from one AbstractLocusInfo object for
 CollectWgsMetrics.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/AdapterUtility.html" title="class in picard.analysis">AdapterUtility</a></td>
<td class="colLast">
<div class="block">A utility class for matching reads to adapters.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/AlignmentSummaryMetrics.html" title="class in picard.analysis">AlignmentSummaryMetrics</a></td>
<td class="colLast">
<div class="block">High level metrics about the alignment of reads within a SAM file, produced by
 the CollectAlignmentSummaryMetrics program and usually stored in a file with
 the extension ".alignment_summary_metrics".</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/AlignmentSummaryMetricsCollector.html" title="class in picard.analysis">AlignmentSummaryMetricsCollector</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/BaseDistributionByCycleMetrics.html" title="class in picard.analysis">BaseDistributionByCycleMetrics</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/ChimeraUtil.html" title="class in picard.analysis">ChimeraUtil</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/CollectAlignmentSummaryMetrics.html" title="class in picard.analysis">CollectAlignmentSummaryMetrics</a></td>
<td class="colLast">
<div class="block">A command line tool to read a BAM file and produce standard alignment metrics that would be applicable to any alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/CollectAlignmentSummaryMetrics.CollectAlignmentRefArgCollection.html" title="class in picard.analysis">CollectAlignmentSummaryMetrics.CollectAlignmentRefArgCollection</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/CollectBaseDistributionByCycle.html" title="class in picard.analysis">CollectBaseDistributionByCycle</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/CollectGcBiasMetrics.html" title="class in picard.analysis">CollectGcBiasMetrics</a></td>
<td class="colLast">
<div class="block">Tool to collect information about GC bias in the reads in a given BAM file.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/CollectInsertSizeMetrics.html" title="class in picard.analysis">CollectInsertSizeMetrics</a></td>
<td class="colLast">
<div class="block">Command line program to read non-duplicate insert sizes, create a Histogram
 and report distribution statistics.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/CollectJumpingLibraryMetrics.html" title="class in picard.analysis">CollectJumpingLibraryMetrics</a></td>
<td class="colLast">
<div class="block">Command-line program to compute metrics about outward-facing pairs, inward-facing
 pairs, and chimeras in a jumping library.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/CollectMultipleMetrics.html" title="class in picard.analysis">CollectMultipleMetrics</a></td>
<td class="colLast">
<div class="block">Class that is designed to instantiate and execute multiple metrics programs that extend
 SinglePassSamProgram while making only a single pass through the SAM file and supplying
 each program with the records as it goes.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/CollectOxoGMetrics.html" title="class in picard.analysis">CollectOxoGMetrics</a></td>
<td class="colLast">
<div class="block">Class for trying to quantify the CpCG->CpCA error rate.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/CollectOxoGMetrics.CpcgMetrics.html" title="class in picard.analysis">CollectOxoGMetrics.CpcgMetrics</a></td>
<td class="colLast">
<div class="block">Metrics class for outputs.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/CollectQualityYieldMetrics.html" title="class in picard.analysis">CollectQualityYieldMetrics</a></td>
<td class="colLast">
<div class="block">Command line program to calibrate quality yield metrics</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/CollectQualityYieldMetrics.QualityYieldMetrics.html" title="class in picard.analysis">CollectQualityYieldMetrics.QualityYieldMetrics</a></td>
<td class="colLast">
<div class="block">A set of metrics used to describe the general quality of a BAM file</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/CollectQualityYieldMetrics.QualityYieldMetricsCollector.html" title="class in picard.analysis">CollectQualityYieldMetrics.QualityYieldMetricsCollector</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/CollectRawWgsMetrics.html" title="class in picard.analysis">CollectRawWgsMetrics</a></td>
<td class="colLast">
<div class="block">Computes a number of metrics that are useful for evaluating coverage and performance of whole genome sequencing
 experiments, same implementation as <a href="../../picard/analysis/CollectWgsMetrics.html" title="class in picard.analysis"><code>CollectWgsMetrics</code></a>, with different defaults: lacks baseQ and mappingQ filters
 and has much higher coverage cap.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/CollectRawWgsMetrics.RawWgsMetrics.html" title="class in picard.analysis">CollectRawWgsMetrics.RawWgsMetrics</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/CollectRnaSeqMetrics.html" title="class in picard.analysis">CollectRnaSeqMetrics</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/CollectRrbsMetrics.html" title="class in picard.analysis">CollectRrbsMetrics</a></td>
<td class="colLast">
<div class="block">Calculates and reports QC metrics for RRBS data based on the methylation status at individual C/G bases as well
 as CpG sites across all reads in the input BAM/SAM file.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/CollectRrbsMetrics.CollectRrbsMetricsReferenceArgumentCollection.html" title="class in picard.analysis">CollectRrbsMetrics.CollectRrbsMetricsReferenceArgumentCollection</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/CollectWgsMetrics.html" title="class in picard.analysis">CollectWgsMetrics</a></td>
<td class="colLast">
<div class="block">Computes a number of metrics that are useful for evaluating coverage and performance of whole genome sequencing experiments.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/CollectWgsMetrics.CollectWgsMetricsIntervalArgumentCollection.html" title="class in picard.analysis">CollectWgsMetrics.CollectWgsMetricsIntervalArgumentCollection</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/CollectWgsMetrics.WgsMetricsCollector.html" title="class in picard.analysis">CollectWgsMetrics.WgsMetricsCollector</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/CollectWgsMetricsWithNonZeroCoverage.html" title="class in picard.analysis">CollectWgsMetricsWithNonZeroCoverage</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/CollectWgsMetricsWithNonZeroCoverage.WgsMetricsWithNonZeroCoverage.html" title="class in picard.analysis">CollectWgsMetricsWithNonZeroCoverage.WgsMetricsWithNonZeroCoverage</a></td>
<td class="colLast">
<div class="block">Metrics for evaluating the performance of whole genome sequencing experiments.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/CompareMetrics.html" title="class in picard.analysis">CompareMetrics</a></td>
<td class="colLast">
<div class="block">Compare two metrics files.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/CounterManager.html" title="class in picard.analysis">CounterManager</a></td>
<td class="colLast">
<div class="block">Class for managing a list of Counters of integer,
 provides methods to access data from Counters with respect to an offset.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/FastWgsMetricsCollector.html" title="class in picard.analysis">FastWgsMetricsCollector</a></td>
<td class="colLast">
<div class="block">Class represents fast algorithm for collecting data from <code>AbstractLocusInfo</code>
 with a list of aligned <code>EdgingRecordAndOffset</code> objects.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/FingerprintingDetailMetrics.html" title="class in picard.analysis">FingerprintingDetailMetrics</a></td>
<td class="colLast">
<div class="block">Detailed metrics about an individual SNP/Haplotype comparison within a fingerprint comparison.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/FingerprintingSummaryMetrics.html" title="class in picard.analysis">FingerprintingSummaryMetrics</a></td>
<td class="colLast">
<div class="block">Summary fingerprinting metrics and statistics about the comparison of the sequence data
 from a single read group (lane or index within a lane) vs.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/GcBiasDetailMetrics.html" title="class in picard.analysis">GcBiasDetailMetrics</a></td>
<td class="colLast">
<div class="block">Class that holds detailed metrics about reads that fall within windows of a certain
 GC bin on the reference genome.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/GcBiasMetricsCollector.html" title="class in picard.analysis">GcBiasMetricsCollector</a></td>
<td class="colLast">
<div class="block">Calculates GC Bias Metrics on multiple levels
  Created by kbergin on 3/23/15.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/GcBiasSummaryMetrics.html" title="class in picard.analysis">GcBiasSummaryMetrics</a></td>
<td class="colLast">
<div class="block">High level metrics that capture how biased the coverage in a certain lane is.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/GcBiasUtils.html" title="class in picard.analysis">GcBiasUtils</a></td>
<td class="colLast">
<div class="block">Utilities to calculate GC Bias
 Created by kbergin on 9/23/15.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/InsertSizeMetrics.html" title="class in picard.analysis">InsertSizeMetrics</a></td>
<td class="colLast">
<div class="block">Metrics about the insert size distribution of a paired-end library, created by the
 CollectInsertSizeMetrics program and usually written to a file with the extension
 ".insert_size_metrics".</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/JumpingLibraryMetrics.html" title="class in picard.analysis">JumpingLibraryMetrics</a></td>
<td class="colLast">
<div class="block">High level metrics about the presence of  outward- and inward-facing pairs
 within a SAM file generated with a jumping library, produced by
 the CollectJumpingLibraryMetrics program and usually stored in a file with
 the extension ".jump_metrics".</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/MeanQualityByCycle.html" title="class in picard.analysis">MeanQualityByCycle</a></td>
<td class="colLast">
<div class="block">Program to generate a data table and chart of mean quality by cycle from a
 BAM file.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/MergeableMetricBase.html" title="class in picard.analysis">MergeableMetricBase</a></td>
<td class="colLast">
<div class="block">An extension of MetricBase that knows how to merge-by-adding fields that are appropriately annotated (<a href="../../picard/analysis/MergeableMetricBase.MergeByAdding.html" title="annotation in picard.analysis"><code>MergeByAdding</code></a>).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/QualityScoreDistribution.html" title="class in picard.analysis">QualityScoreDistribution</a></td>
<td class="colLast">
<div class="block">Charts quality score distribution within a BAM file.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/RnaSeqMetrics.html" title="class in picard.analysis">RnaSeqMetrics</a></td>
<td class="colLast">
<div class="block">Metrics about the alignment of RNA-seq reads within a SAM file to genes, produced by the CollectRnaSeqMetrics
 program and usually stored in a file with the extension ".rna_metrics".</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/RrbsCpgDetailMetrics.html" title="class in picard.analysis">RrbsCpgDetailMetrics</a></td>
<td class="colLast">
<div class="block">Holds information about CpG sites encountered for RRBS processing QC</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/RrbsMetricsCollector.html" title="class in picard.analysis">RrbsMetricsCollector</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/RrbsSummaryMetrics.html" title="class in picard.analysis">RrbsSummaryMetrics</a></td>
<td class="colLast">
<div class="block">Holds summary statistics from RRBS processing QC</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/SinglePassSamProgram.html" title="class in picard.analysis">SinglePassSamProgram</a></td>
<td class="colLast">
<div class="block">Super class that is designed to provide some consistent structure between subclasses that
 simply iterate once over a coordinate sorted BAM and collect information from the records
 as the go in order to produce some kind of output.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/TheoreticalSensitivity.html" title="class in picard.analysis">TheoreticalSensitivity</a></td>
<td class="colLast">
<div class="block">Created by David Benjamin on 5/13/15.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/TheoreticalSensitivity.RouletteWheel.html" title="class in picard.analysis">TheoreticalSensitivity.RouletteWheel</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/TheoreticalSensitivityMetrics.html" title="class in picard.analysis">TheoreticalSensitivityMetrics</a></td>
<td class="colLast">
<div class="block">TheoreticalSensitivityMetrics, are metrics calculated from TheoreticalSensitivity and parameters used in
 the calculation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/WgsMetrics.html" title="class in picard.analysis">WgsMetrics</a></td>
<td class="colLast">
<div class="block">Metrics for evaluating the performance of whole genome sequencing experiments.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/WgsMetricsProcessorImpl.html" title="class in picard.analysis">WgsMetricsProcessorImpl</a>&lt;T extends htsjdk.samtools.util.AbstractRecordAndOffset&gt;</td>
<td class="colLast">
<div class="block">Implementation of <a href="../../picard/analysis/WgsMetricsProcessor.html" title="interface in picard.analysis"><code>WgsMetricsProcessor</code></a> that gets input data from a given iterator
 and processes it with a help of collector</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<table class="typeSummary" border="0" cellpadding="3" cellspacing="0" summary="Enum Summary table, listing enums, and an explanation">
<caption><span>Enum Summary</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Enum</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/AlignmentSummaryMetrics.Category.html" title="enum in picard.analysis">AlignmentSummaryMetrics.Category</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/CollectMultipleMetrics.Program.html" title="enum in picard.analysis">CollectMultipleMetrics.Program</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/CollectWgsMetricsWithNonZeroCoverage.WgsMetricsWithNonZeroCoverage.Category.html" title="enum in picard.analysis">CollectWgsMetricsWithNonZeroCoverage.WgsMetricsWithNonZeroCoverage.Category</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/MetricAccumulationLevel.html" title="enum in picard.analysis">MetricAccumulationLevel</a></td>
<td class="colLast">
<div class="block">For use with Picard metrics programs that may output metrics for multiple levels
 of aggregation with an analysis.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<table class="typeSummary" border="0" cellpadding="3" cellspacing="0" summary="Annotation Types Summary table, listing annotation types, and an explanation">
<caption><span>Annotation Types Summary</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Annotation Type</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/MergeableMetricBase.MergeByAdding.html" title="annotation in picard.analysis">MergeableMetricBase.MergeByAdding</a></td>
<td class="colLast">
<div class="block">Metrics whose values can be merged by adding.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/MergeableMetricBase.MergeByAssertEquals.html" title="annotation in picard.analysis">MergeableMetricBase.MergeByAssertEquals</a></td>
<td class="colLast">
<div class="block">Metrics whose values should be equal when merging.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/MergeableMetricBase.MergingIsManual.html" title="annotation in picard.analysis">MergeableMetricBase.MergingIsManual</a></td>
<td class="colLast">
<div class="block">Metrics that are merged manually in the <a href="../../picard/analysis/MergeableMetricBase.html#merge-picard.analysis.MergeableMetricBase-"><code>MergeableMetricBase.merge(MergeableMetricBase)</code></a> ()}.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/analysis/MergeableMetricBase.NoMergingIsDerived.html" title="annotation in picard.analysis">MergeableMetricBase.NoMergingIsDerived</a></td>
<td class="colLast">
<div class="block">Metrics that are not merged, but are subsequently derived from other metrics, for example by
 <a href="../../picard/analysis/MergeableMetricBase.html#calculateDerivedFields--"><code>MergeableMetricBase.calculateDerivedFields()</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/analysis/MergeableMetricBase.NoMergingKeepsValue.html" title="annotation in picard.analysis">MergeableMetricBase.NoMergingKeepsValue</a></td>
<td class="colLast">
<div class="block">Metrics that are not merged.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li class="navBarCell1Rev">Package</li>
<li>Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/package-summary.html">Prev&nbsp;Package</a></li>
<li><a href="../../picard/analysis/artifacts/package-summary.html">Next&nbsp;Package</a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/analysis/package-summary.html" target="_top">Frames</a></li>
<li><a href="package-summary.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
