<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_201) on Fri Oct 04 10:35:33 EDT 2019 -->
<title>RnaSeqMetrics (picard 2.21.1-SNAPSHOT API)</title>
<meta name="date" content="2019-10-04">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="RnaSeqMetrics (picard 2.21.1-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/analysis/QualityScoreDistribution.html" title="class in picard.analysis"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/analysis/RrbsCpgDetailMetrics.html" title="class in picard.analysis"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/analysis/RnaSeqMetrics.html" target="_top">Frames</a></li>
<li><a href="RnaSeqMetrics.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#methods.inherited.from.class.htsjdk.samtools.metrics.MetricBase">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li>Method</li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">picard.analysis</div>
<h2 title="Class RnaSeqMetrics" class="title">Class RnaSeqMetrics</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>htsjdk.samtools.metrics.MetricBase</li>
<li>
<ul class="inheritance">
<li><a href="../../picard/metrics/MultilevelMetrics.html" title="class in picard.metrics">picard.metrics.MultilevelMetrics</a></li>
<li>
<ul class="inheritance">
<li>picard.analysis.RnaSeqMetrics</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public class <span class="typeNameLabel">RnaSeqMetrics</span>
extends <a href="../../picard/metrics/MultilevelMetrics.html" title="class in picard.metrics">MultilevelMetrics</a></pre>
<div class="block">Metrics about the alignment of RNA-seq reads within a SAM file to genes, produced by the CollectRnaSeqMetrics
 program and usually stored in a file with the extension ".rna_metrics".</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#CODING_BASES">CODING_BASES</a></span></code>
<div class="block">Number of bases in primary alignments that align to a non-UTR coding base for some gene, and not ribosomal sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#CORRECT_STRAND_READS">CORRECT_STRAND_READS</a></span></code>
<div class="block">Number of aligned reads that are mapped to the correct strand.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#IGNORED_READS">IGNORED_READS</a></span></code>
<div class="block">Number of primary alignments that are mapped to a sequence specified on command-line as IGNORED_SEQUENCE.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#INCORRECT_STRAND_READS">INCORRECT_STRAND_READS</a></span></code>
<div class="block">Number of aligned reads that are mapped to the incorrect strand.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#INTERGENIC_BASES">INTERGENIC_BASES</a></span></code>
<div class="block">Number of bases in primary alignments that do not align to any gene.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#INTRONIC_BASES">INTRONIC_BASES</a></span></code>
<div class="block">Number of bases in primary alignments that align to an intronic base for some gene, and not a coding or UTR base.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#MEDIAN_3PRIME_BIAS">MEDIAN_3PRIME_BIAS</a></span></code>
<div class="block">The median 3 prime bias of the 1000 most highly expressed transcripts, where 3 prime bias is calculated per
 transcript as: mean coverage of the 3 prime-most 100 bases divided by the mean coverage of the whole transcript.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#MEDIAN_5PRIME_BIAS">MEDIAN_5PRIME_BIAS</a></span></code>
<div class="block">The median 5 prime bias of the 1000 most highly expressed transcripts.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#MEDIAN_5PRIME_TO_3PRIME_BIAS">MEDIAN_5PRIME_TO_3PRIME_BIAS</a></span></code>
<div class="block">The ratio of coverage at the 5 prime end to the 3 prime end based on the 1000 most highly expressed transcripts.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#MEDIAN_CV_COVERAGE">MEDIAN_CV_COVERAGE</a></span></code>
<div class="block">The median coefficient of variation (CV) or stdev/mean for coverage values of the 1000 most highly expressed transcripts.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#NUM_R1_TRANSCRIPT_STRAND_READS">NUM_R1_TRANSCRIPT_STRAND_READS</a></span></code>
<div class="block">The number of reads that support the model where R1 is on the strand of transcription and R2 is on the
 opposite strand.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#NUM_R2_TRANSCRIPT_STRAND_READS">NUM_R2_TRANSCRIPT_STRAND_READS</a></span></code>
<div class="block">The fraction of reads that support the model where R2 is on the strand of transcription and R1 is on the opposite
 strand.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#NUM_UNEXPLAINED_READS">NUM_UNEXPLAINED_READS</a></span></code>
<div class="block">The fraction of reads for which the transcription strand model could not be inferred.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#PCT_CODING_BASES">PCT_CODING_BASES</a></span></code>
<div class="block">Fraction of PF_ALIGNED_BASES that mapped to protein coding regions of genes, CODING_BASES/PF_ALIGNED_BASES</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#PCT_CORRECT_STRAND_READS">PCT_CORRECT_STRAND_READS</a></span></code>
<div class="block">Fraction of reads corresponding to mRNA transcripts which map to the correct strand of a reference genome
    = CORRECT_STRAND_READS/(CORRECT_STRAND_READS + INCORRECT_STRAND_READS).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#PCT_INTERGENIC_BASES">PCT_INTERGENIC_BASES</a></span></code>
<div class="block">Fraction of PF_ALIGNED_BASES that mapped to intergenic regions of genomic DNA, INTERGENIC_BASES/PF_ALIGNED_BASES</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#PCT_INTRONIC_BASES">PCT_INTRONIC_BASES</a></span></code>
<div class="block">Fraction of PF_ALIGNED_BASES that correspond to gene introns, INTRONIC_BASES/PF_ALIGNED_BASES</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#PCT_MRNA_BASES">PCT_MRNA_BASES</a></span></code>
<div class="block">Sum of bases mapped to regions corresponding to UTRs and coding regions of mRNA transcripts, PCT_UTR_BASES + PCT_CODING_BASES</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#PCT_R1_TRANSCRIPT_STRAND_READS">PCT_R1_TRANSCRIPT_STRAND_READS</a></span></code>
<div class="block">The fraction of reads that support the model where R1 is on the strand of transcription and R2 is on the
 opposite strand.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#PCT_R2_TRANSCRIPT_STRAND_READS">PCT_R2_TRANSCRIPT_STRAND_READS</a></span></code>
<div class="block">The fraction of reads that support the model where R2 is on the strand of transcription and R1 is on the opposite
 strand.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.lang.Double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#PCT_RIBOSOMAL_BASES">PCT_RIBOSOMAL_BASES</a></span></code>
<div class="block">Fraction of PF_ALIGNED_BASES that mapped to regions encoding ribosomal RNA, RIBOSOMAL_BASES/PF_ALIGNED_BASES</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#PCT_USABLE_BASES">PCT_USABLE_BASES</a></span></code>
<div class="block">The fraction of bases mapping to mRNA divided by the total number of PF bases, (CODING_BASES + UTR_BASES)/PF_BASES.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#PCT_UTR_BASES">PCT_UTR_BASES</a></span></code>
<div class="block">Fraction of PF_ALIGNED_BASES that mapped to untranslated regions (UTR) of genes, UTR_BASES/PF_ALIGNED_BASES</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#PF_ALIGNED_BASES">PF_ALIGNED_BASES</a></span></code>
<div class="block">The total number of aligned PF bases.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#PF_BASES">PF_BASES</a></span></code>
<div class="block">The total number of PF bases including non-aligned reads.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.Long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#RIBOSOMAL_BASES">RIBOSOMAL_BASES</a></span></code>
<div class="block">Number of bases in primary alignments that align to ribosomal sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#UTR_BASES">UTR_BASES</a></span></code>
<div class="block">Number of bases in primary alignments that align to a UTR base for some gene, and not a coding base.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="fields.inherited.from.class.picard.metrics.MultilevelMetrics">
<!--   -->
</a>
<h3>Fields inherited from class&nbsp;picard.metrics.<a href="../../picard/metrics/MultilevelMetrics.html" title="class in picard.metrics">MultilevelMetrics</a></h3>
<code><a href="../../picard/metrics/MultilevelMetrics.html#LIBRARY">LIBRARY</a>, <a href="../../picard/metrics/MultilevelMetrics.html#READ_GROUP">READ_GROUP</a>, <a href="../../picard/metrics/MultilevelMetrics.html#SAMPLE">SAMPLE</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../picard/analysis/RnaSeqMetrics.html#RnaSeqMetrics--">RnaSeqMetrics</a></span>()</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.htsjdk.samtools.metrics.MetricBase">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;htsjdk.samtools.metrics.MetricBase</h3>
<code>equals, hashCode, toString</code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, finalize, getClass, notify, notifyAll, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="PF_BASES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PF_BASES</h4>
<pre>public&nbsp;long PF_BASES</pre>
<div class="block">The total number of PF bases including non-aligned reads.</div>
</li>
</ul>
<a name="PF_ALIGNED_BASES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PF_ALIGNED_BASES</h4>
<pre>public&nbsp;long PF_ALIGNED_BASES</pre>
<div class="block">The total number of aligned PF bases.  Non-primary alignments are not counted. Bases in aligned reads that
 do not correspond to reference (e.g. soft clips, insertions) are not counted.</div>
</li>
</ul>
<a name="RIBOSOMAL_BASES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>RIBOSOMAL_BASES</h4>
<pre>public&nbsp;java.lang.Long RIBOSOMAL_BASES</pre>
<div class="block">Number of bases in primary alignments that align to ribosomal sequence.</div>
</li>
</ul>
<a name="CODING_BASES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>CODING_BASES</h4>
<pre>public&nbsp;long CODING_BASES</pre>
<div class="block">Number of bases in primary alignments that align to a non-UTR coding base for some gene, and not ribosomal sequence.</div>
</li>
</ul>
<a name="UTR_BASES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>UTR_BASES</h4>
<pre>public&nbsp;long UTR_BASES</pre>
<div class="block">Number of bases in primary alignments that align to a UTR base for some gene, and not a coding base.</div>
</li>
</ul>
<a name="INTRONIC_BASES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>INTRONIC_BASES</h4>
<pre>public&nbsp;long INTRONIC_BASES</pre>
<div class="block">Number of bases in primary alignments that align to an intronic base for some gene, and not a coding or UTR base.</div>
</li>
</ul>
<a name="INTERGENIC_BASES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>INTERGENIC_BASES</h4>
<pre>public&nbsp;long INTERGENIC_BASES</pre>
<div class="block">Number of bases in primary alignments that do not align to any gene.</div>
</li>
</ul>
<a name="IGNORED_READS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>IGNORED_READS</h4>
<pre>public&nbsp;long IGNORED_READS</pre>
<div class="block">Number of primary alignments that are mapped to a sequence specified on command-line as IGNORED_SEQUENCE.  These are not
 counted in PF_ALIGNED_BASES, CORRECT_STRAND_READS, INCORRECT_STRAND_READS, or any of the base-counting metrics.
 These reads are counted in PF_BASES.</div>
</li>
</ul>
<a name="CORRECT_STRAND_READS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>CORRECT_STRAND_READS</h4>
<pre>public&nbsp;long CORRECT_STRAND_READS</pre>
<div class="block">Number of aligned reads that are mapped to the correct strand.  0 if library is not strand-specific.</div>
</li>
</ul>
<a name="INCORRECT_STRAND_READS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>INCORRECT_STRAND_READS</h4>
<pre>public&nbsp;long INCORRECT_STRAND_READS</pre>
<div class="block">Number of aligned reads that are mapped to the incorrect strand.  0 if library is not strand-specific.</div>
</li>
</ul>
<a name="NUM_R1_TRANSCRIPT_STRAND_READS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>NUM_R1_TRANSCRIPT_STRAND_READS</h4>
<pre>public&nbsp;long NUM_R1_TRANSCRIPT_STRAND_READS</pre>
<div class="block">The number of reads that support the model where R1 is on the strand of transcription and R2 is on the
 opposite strand.</div>
</li>
</ul>
<a name="NUM_R2_TRANSCRIPT_STRAND_READS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>NUM_R2_TRANSCRIPT_STRAND_READS</h4>
<pre>public&nbsp;long NUM_R2_TRANSCRIPT_STRAND_READS</pre>
<div class="block">The fraction of reads that support the model where R2 is on the strand of transcription and R1 is on the opposite
 strand.</div>
</li>
</ul>
<a name="NUM_UNEXPLAINED_READS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>NUM_UNEXPLAINED_READS</h4>
<pre>public&nbsp;long NUM_UNEXPLAINED_READS</pre>
<div class="block">The fraction of reads for which the transcription strand model could not be inferred.</div>
</li>
</ul>
<a name="PCT_R1_TRANSCRIPT_STRAND_READS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_R1_TRANSCRIPT_STRAND_READS</h4>
<pre>public&nbsp;double PCT_R1_TRANSCRIPT_STRAND_READS</pre>
<div class="block">The fraction of reads that support the model where R1 is on the strand of transcription and R2 is on the
 opposite strand.  For unpaired reads, it is the fraction of reads that are on the transcription strand (out of all
 the reads).</div>
</li>
</ul>
<a name="PCT_R2_TRANSCRIPT_STRAND_READS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_R2_TRANSCRIPT_STRAND_READS</h4>
<pre>public&nbsp;double PCT_R2_TRANSCRIPT_STRAND_READS</pre>
<div class="block">The fraction of reads that support the model where R2 is on the strand of transcription and R1 is on the opposite
 strand.  For unpaired reads, it is the fraction of reads that are on opposite strand than that of the the
 transcription strand (out of all the reads).</div>
</li>
</ul>
<a name="PCT_RIBOSOMAL_BASES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_RIBOSOMAL_BASES</h4>
<pre>public&nbsp;java.lang.Double PCT_RIBOSOMAL_BASES</pre>
<div class="block">Fraction of PF_ALIGNED_BASES that mapped to regions encoding ribosomal RNA, RIBOSOMAL_BASES/PF_ALIGNED_BASES</div>
</li>
</ul>
<a name="PCT_CODING_BASES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_CODING_BASES</h4>
<pre>public&nbsp;double PCT_CODING_BASES</pre>
<div class="block">Fraction of PF_ALIGNED_BASES that mapped to protein coding regions of genes, CODING_BASES/PF_ALIGNED_BASES</div>
</li>
</ul>
<a name="PCT_UTR_BASES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_UTR_BASES</h4>
<pre>public&nbsp;double PCT_UTR_BASES</pre>
<div class="block">Fraction of PF_ALIGNED_BASES that mapped to untranslated regions (UTR) of genes, UTR_BASES/PF_ALIGNED_BASES</div>
</li>
</ul>
<a name="PCT_INTRONIC_BASES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_INTRONIC_BASES</h4>
<pre>public&nbsp;double PCT_INTRONIC_BASES</pre>
<div class="block">Fraction of PF_ALIGNED_BASES that correspond to gene introns, INTRONIC_BASES/PF_ALIGNED_BASES</div>
</li>
</ul>
<a name="PCT_INTERGENIC_BASES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_INTERGENIC_BASES</h4>
<pre>public&nbsp;double PCT_INTERGENIC_BASES</pre>
<div class="block">Fraction of PF_ALIGNED_BASES that mapped to intergenic regions of genomic DNA, INTERGENIC_BASES/PF_ALIGNED_BASES</div>
</li>
</ul>
<a name="PCT_MRNA_BASES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_MRNA_BASES</h4>
<pre>public&nbsp;double PCT_MRNA_BASES</pre>
<div class="block">Sum of bases mapped to regions corresponding to UTRs and coding regions of mRNA transcripts, PCT_UTR_BASES + PCT_CODING_BASES</div>
</li>
</ul>
<a name="PCT_USABLE_BASES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_USABLE_BASES</h4>
<pre>public&nbsp;double PCT_USABLE_BASES</pre>
<div class="block">The fraction of bases mapping to mRNA divided by the total number of PF bases, (CODING_BASES + UTR_BASES)/PF_BASES.</div>
</li>
</ul>
<a name="PCT_CORRECT_STRAND_READS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_CORRECT_STRAND_READS</h4>
<pre>public&nbsp;double PCT_CORRECT_STRAND_READS</pre>
<div class="block">Fraction of reads corresponding to mRNA transcripts which map to the correct strand of a reference genome
    = CORRECT_STRAND_READS/(CORRECT_STRAND_READS + INCORRECT_STRAND_READS).  0 if library is not strand-specific.</div>
</li>
</ul>
<a name="MEDIAN_CV_COVERAGE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>MEDIAN_CV_COVERAGE</h4>
<pre>public&nbsp;double MEDIAN_CV_COVERAGE</pre>
<div class="block">The median coefficient of variation (CV) or stdev/mean for coverage values of the 1000 most highly expressed transcripts. Ideal value = 0.</div>
</li>
</ul>
<a name="MEDIAN_5PRIME_BIAS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>MEDIAN_5PRIME_BIAS</h4>
<pre>public&nbsp;double MEDIAN_5PRIME_BIAS</pre>
<div class="block">The median 5 prime bias of the 1000 most highly expressed transcripts. The 5 prime bias is calculated per
 transcript as: mean coverage of the 5 prime-most 100 bases divided by the mean coverage of the whole transcript.</div>
</li>
</ul>
<a name="MEDIAN_3PRIME_BIAS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>MEDIAN_3PRIME_BIAS</h4>
<pre>public&nbsp;double MEDIAN_3PRIME_BIAS</pre>
<div class="block">The median 3 prime bias of the 1000 most highly expressed transcripts, where 3 prime bias is calculated per
 transcript as: mean coverage of the 3 prime-most 100 bases divided by the mean coverage of the whole transcript.</div>
</li>
</ul>
<a name="MEDIAN_5PRIME_TO_3PRIME_BIAS">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>MEDIAN_5PRIME_TO_3PRIME_BIAS</h4>
<pre>public&nbsp;double MEDIAN_5PRIME_TO_3PRIME_BIAS</pre>
<div class="block">The ratio of coverage at the 5 prime end to the 3 prime end based on the 1000 most highly expressed transcripts.</div>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="RnaSeqMetrics--">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>RnaSeqMetrics</h4>
<pre>public&nbsp;RnaSeqMetrics()</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/analysis/QualityScoreDistribution.html" title="class in picard.analysis"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/analysis/RrbsCpgDetailMetrics.html" title="class in picard.analysis"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/analysis/RnaSeqMetrics.html" target="_top">Frames</a></li>
<li><a href="RnaSeqMetrics.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#methods.inherited.from.class.htsjdk.samtools.metrics.MetricBase">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li>Method</li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
