<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_201) on Fri Oct 04 10:35:33 EDT 2019 -->
<title>WgsMetrics (picard 2.21.1-SNAPSHOT API)</title>
<meta name="date" content="2019-10-04">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="WgsMetrics (picard 2.21.1-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":10,"i1":10};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/analysis/TheoreticalSensitivityMetrics.html" title="class in picard.analysis"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/analysis/WgsMetricsProcessor.html" title="interface in picard.analysis"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/analysis/WgsMetrics.html" target="_top">Frames</a></li>
<li><a href="WgsMetrics.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.classes.inherited.from.class.picard.analysis.MergeableMetricBase">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">picard.analysis</div>
<h2 title="Class WgsMetrics" class="title">Class WgsMetrics</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>htsjdk.samtools.metrics.MetricBase</li>
<li>
<ul class="inheritance">
<li><a href="../../picard/analysis/MergeableMetricBase.html" title="class in picard.analysis">picard.analysis.MergeableMetricBase</a></li>
<li>
<ul class="inheritance">
<li>picard.analysis.WgsMetrics</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>Direct Known Subclasses:</dt>
<dd><a href="../../picard/analysis/CollectRawWgsMetrics.RawWgsMetrics.html" title="class in picard.analysis">CollectRawWgsMetrics.RawWgsMetrics</a>, <a href="../../picard/analysis/CollectWgsMetricsWithNonZeroCoverage.WgsMetricsWithNonZeroCoverage.html" title="class in picard.analysis">CollectWgsMetricsWithNonZeroCoverage.WgsMetricsWithNonZeroCoverage</a></dd>
</dl>
<hr>
<br>
<pre>public class <span class="typeNameLabel">WgsMetrics</span>
extends <a href="../../picard/analysis/MergeableMetricBase.html" title="class in picard.analysis">MergeableMetricBase</a></pre>
<div class="block">Metrics for evaluating the performance of whole genome sequencing experiments.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="nested.classes.inherited.from.class.picard.analysis.MergeableMetricBase">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;picard.analysis.<a href="../../picard/analysis/MergeableMetricBase.html" title="class in picard.analysis">MergeableMetricBase</a></h3>
<code><a href="../../picard/analysis/MergeableMetricBase.MergeByAdding.html" title="annotation in picard.analysis">MergeableMetricBase.MergeByAdding</a>, <a href="../../picard/analysis/MergeableMetricBase.MergeByAssertEquals.html" title="annotation in picard.analysis">MergeableMetricBase.MergeByAssertEquals</a>, <a href="../../picard/analysis/MergeableMetricBase.MergingIsManual.html" title="annotation in picard.analysis">MergeableMetricBase.MergingIsManual</a>, <a href="../../picard/analysis/MergeableMetricBase.NoMergingIsDerived.html" title="annotation in picard.analysis">MergeableMetricBase.NoMergingIsDerived</a>, <a href="../../picard/analysis/MergeableMetricBase.NoMergingKeepsValue.html" title="annotation in picard.analysis">MergeableMetricBase.NoMergingKeepsValue</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#coverageCap">coverageCap</a></span></code>
<div class="block">The maximum depth/coverage to consider.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#GENOME_TERRITORY">GENOME_TERRITORY</a></span></code>
<div class="block">The number of non-N bases in the genome reference over which coverage will be evaluated.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#HET_SNP_Q">HET_SNP_Q</a></span></code>
<div class="block">The Phred Scaled Q Score of the theoretical HET SNP sensitivity.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#HET_SNP_SENSITIVITY">HET_SNP_SENSITIVITY</a></span></code>
<div class="block">The theoretical HET SNP sensitivity.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected htsjdk.samtools.util.Histogram&lt;java.lang.Integer&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#highQualityDepthHistogram">highQualityDepthHistogram</a></span></code>
<div class="block">Count of sites with a given depth of coverage.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected htsjdk.samtools.util.IntervalList</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#intervals">intervals</a></span></code>
<div class="block">The intervals over which this metric was computed.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#MAD_COVERAGE">MAD_COVERAGE</a></span></code>
<div class="block">The median absolute deviation of coverage of the genome after all filters are applied.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#MEAN_COVERAGE">MEAN_COVERAGE</a></span></code>
<div class="block">The mean coverage in bases of the genome territory, after all filters are applied.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#MEDIAN_COVERAGE">MEDIAN_COVERAGE</a></span></code>
<div class="block">The median coverage in bases of the genome territory, after all filters are applied.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_100X">PCT_100X</a></span></code>
<div class="block">The fraction of bases that attained at least 100X sequence coverage in post-filtering bases.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_10X">PCT_10X</a></span></code>
<div class="block">The fraction of bases that attained at least 10X sequence coverage in post-filtering bases.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_15X">PCT_15X</a></span></code>
<div class="block">The fraction of bases that attained at least 15X sequence coverage in post-filtering bases.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_1X">PCT_1X</a></span></code>
<div class="block">The fraction of bases that attained at least 1X sequence coverage in post-filtering bases.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_20X">PCT_20X</a></span></code>
<div class="block">The fraction of bases that attained at least 20X sequence coverage in post-filtering bases.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_25X">PCT_25X</a></span></code>
<div class="block">The fraction of bases that attained at least 25X sequence coverage in post-filtering bases.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_30X">PCT_30X</a></span></code>
<div class="block">The fraction of bases that attained at least 30X sequence coverage in post-filtering bases.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_40X">PCT_40X</a></span></code>
<div class="block">The fraction of bases that attained at least 40X sequence coverage in post-filtering bases.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_50X">PCT_50X</a></span></code>
<div class="block">The fraction of bases that attained at least 50X sequence coverage in post-filtering bases.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_5X">PCT_5X</a></span></code>
<div class="block">The fraction of bases that attained at least 5X sequence coverage in post-filtering bases.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_60X">PCT_60X</a></span></code>
<div class="block">The fraction of bases that attained at least 60X sequence coverage in post-filtering bases.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_70X">PCT_70X</a></span></code>
<div class="block">The fraction of bases that attained at least 70X sequence coverage in post-filtering bases.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_80X">PCT_80X</a></span></code>
<div class="block">The fraction of bases that attained at least 80X sequence coverage in post-filtering bases.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_90X">PCT_90X</a></span></code>
<div class="block">The fraction of bases that attained at least 90X sequence coverage in post-filtering bases.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_EXC_ADAPTER">PCT_EXC_ADAPTER</a></span></code>
<div class="block">The fraction of aligned bases that were filtered out because they were in reads with mapping quality 0 and the looked like adapter reads.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_EXC_BASEQ">PCT_EXC_BASEQ</a></span></code>
<div class="block">The fraction of aligned bases that were filtered out because they were of low base quality (lower than MIN_BASE_QUALITY).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_EXC_CAPPED">PCT_EXC_CAPPED</a></span></code>
<div class="block">The fraction of aligned bases that were filtered out because they would have raised coverage above COVERAGE_CAP.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_EXC_DUPE">PCT_EXC_DUPE</a></span></code>
<div class="block">The fraction of aligned bases that were filtered out because they were in reads marked as duplicates.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_EXC_MAPQ">PCT_EXC_MAPQ</a></span></code>
<div class="block">The fraction of aligned bases that were filtered out because they were in reads with low mapping quality (lower than MIN_MAPPING_QUALITY).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_EXC_OVERLAP">PCT_EXC_OVERLAP</a></span></code>
<div class="block">The fraction of aligned bases that were filtered out because they were the second observation from an insert with overlapping reads.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_EXC_TOTAL">PCT_EXC_TOTAL</a></span></code>
<div class="block">The total fraction of aligned bases excluded due to all filters.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#PCT_EXC_UNPAIRED">PCT_EXC_UNPAIRED</a></span></code>
<div class="block">The fraction of aligned bases that were filtered out because they were in reads without a mapped mate pair.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#SD_COVERAGE">SD_COVERAGE</a></span></code>
<div class="block">The standard deviation of coverage of the genome after all filters are applied.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#theoreticalHetSensitivitySampleSize">theoreticalHetSensitivitySampleSize</a></span></code>
<div class="block">The sample size used for theoretical het sensitivity.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected htsjdk.samtools.util.Histogram&lt;java.lang.Integer&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#unfilteredBaseQHistogram">unfilteredBaseQHistogram</a></span></code>
<div class="block">The count of bases observed with a given base quality.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected htsjdk.samtools.util.Histogram&lt;java.lang.Integer&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#unfilteredDepthHistogram">unfilteredDepthHistogram</a></span></code>
<div class="block">Count of sites with a given depth of coverage.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#WgsMetrics--">WgsMetrics</a></span>()</code>
<div class="block">Create an instance of this metric that is not mergeable.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#WgsMetrics-htsjdk.samtools.util.IntervalList-htsjdk.samtools.util.Histogram-htsjdk.samtools.util.Histogram-double-double-double-double-double-double-double-double-int-htsjdk.samtools.util.Histogram-int-">WgsMetrics</a></span>(htsjdk.samtools.util.IntervalList&nbsp;intervals,
          htsjdk.samtools.util.Histogram&lt;java.lang.Integer&gt;&nbsp;highQualityDepthHistogram,
          htsjdk.samtools.util.Histogram&lt;java.lang.Integer&gt;&nbsp;unfilteredDepthHistogram,
          double&nbsp;pctExcludedByAdapter,
          double&nbsp;pctExcludedByMapq,
          double&nbsp;pctExcludedByDupes,
          double&nbsp;pctExcludedByPairing,
          double&nbsp;pctExcludedByBaseq,
          double&nbsp;pctExcludedByOverlap,
          double&nbsp;pctExcludedByCapping,
          double&nbsp;pctExcludeTotal,
          int&nbsp;coverageCap,
          htsjdk.samtools.util.Histogram&lt;java.lang.Integer&gt;&nbsp;unfilteredBaseQHistogram,
          int&nbsp;theoreticalHetSensitivitySampleSize)</code>
<div class="block">Create an instance of this metric that is mergeable.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#calculateDerivedFields--">calculateDerivedFields</a></span>()</code>
<div class="block">Placeholder method that will calculate the derived fields from the other ones.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code><a href="../../picard/analysis/MergeableMetricBase.html" title="class in picard.analysis">MergeableMetricBase</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/WgsMetrics.html#merge-picard.analysis.MergeableMetricBase-">merge</a></span>(<a href="../../picard/analysis/MergeableMetricBase.html" title="class in picard.analysis">MergeableMetricBase</a>&nbsp;other)</code>
<div class="block">Merges the various PCT_EXC_* metrics.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.picard.analysis.MergeableMetricBase">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;picard.analysis.<a href="../../picard/analysis/MergeableMetricBase.html" title="class in picard.analysis">MergeableMetricBase</a></h3>
<code><a href="../../picard/analysis/MergeableMetricBase.html#canMerge-picard.analysis.MergeableMetricBase-">canMerge</a>, <a href="../../picard/analysis/MergeableMetricBase.html#merge-java.util.Collection-">merge</a>, <a href="../../picard/analysis/MergeableMetricBase.html#mergeIfCan-picard.analysis.MergeableMetricBase-">mergeIfCan</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.htsjdk.samtools.metrics.MetricBase">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;htsjdk.samtools.metrics.MetricBase</h3>
<code>equals, hashCode, toString</code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, finalize, getClass, notify, notifyAll, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="intervals">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>intervals</h4>
<pre>protected&nbsp;htsjdk.samtools.util.IntervalList intervals</pre>
<div class="block">The intervals over which this metric was computed.</div>
</li>
</ul>
<a name="highQualityDepthHistogram">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>highQualityDepthHistogram</h4>
<pre>protected final&nbsp;htsjdk.samtools.util.Histogram&lt;java.lang.Integer&gt; highQualityDepthHistogram</pre>
<div class="block">Count of sites with a given depth of coverage. Excludes bases with quality below MINIMUM_BASE_QUALITY</div>
</li>
</ul>
<a name="unfilteredDepthHistogram">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>unfilteredDepthHistogram</h4>
<pre>protected final&nbsp;htsjdk.samtools.util.Histogram&lt;java.lang.Integer&gt; unfilteredDepthHistogram</pre>
<div class="block">Count of sites with a given depth of coverage. Includes all but quality 2 bases</div>
</li>
</ul>
<a name="unfilteredBaseQHistogram">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>unfilteredBaseQHistogram</h4>
<pre>protected final&nbsp;htsjdk.samtools.util.Histogram&lt;java.lang.Integer&gt; unfilteredBaseQHistogram</pre>
<div class="block">The count of bases observed with a given base quality.</div>
</li>
</ul>
<a name="coverageCap">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>coverageCap</h4>
<pre>protected final&nbsp;int coverageCap</pre>
<div class="block">The maximum depth/coverage to consider.</div>
</li>
</ul>
<a name="theoreticalHetSensitivitySampleSize">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>theoreticalHetSensitivitySampleSize</h4>
<pre>protected final&nbsp;int theoreticalHetSensitivitySampleSize</pre>
<div class="block">The sample size used for theoretical het sensitivity.</div>
</li>
</ul>
<a name="GENOME_TERRITORY">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>GENOME_TERRITORY</h4>
<pre>public&nbsp;long GENOME_TERRITORY</pre>
<div class="block">The number of non-N bases in the genome reference over which coverage will be evaluated.</div>
</li>
</ul>
<a name="MEAN_COVERAGE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>MEAN_COVERAGE</h4>
<pre>public&nbsp;double MEAN_COVERAGE</pre>
<div class="block">The mean coverage in bases of the genome territory, after all filters are applied.</div>
</li>
</ul>
<a name="SD_COVERAGE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SD_COVERAGE</h4>
<pre>public&nbsp;double SD_COVERAGE</pre>
<div class="block">The standard deviation of coverage of the genome after all filters are applied.</div>
</li>
</ul>
<a name="MEDIAN_COVERAGE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>MEDIAN_COVERAGE</h4>
<pre>public&nbsp;double MEDIAN_COVERAGE</pre>
<div class="block">The median coverage in bases of the genome territory, after all filters are applied.</div>
</li>
</ul>
<a name="MAD_COVERAGE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>MAD_COVERAGE</h4>
<pre>public&nbsp;double MAD_COVERAGE</pre>
<div class="block">The median absolute deviation of coverage of the genome after all filters are applied.</div>
</li>
</ul>
<a name="PCT_EXC_ADAPTER">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_EXC_ADAPTER</h4>
<pre>public&nbsp;double PCT_EXC_ADAPTER</pre>
<div class="block">The fraction of aligned bases that were filtered out because they were in reads with mapping quality 0 and the looked like adapter reads.</div>
</li>
</ul>
<a name="PCT_EXC_MAPQ">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_EXC_MAPQ</h4>
<pre>public&nbsp;double PCT_EXC_MAPQ</pre>
<div class="block">The fraction of aligned bases that were filtered out because they were in reads with low mapping quality (lower than MIN_MAPPING_QUALITY).</div>
</li>
</ul>
<a name="PCT_EXC_DUPE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_EXC_DUPE</h4>
<pre>public&nbsp;double PCT_EXC_DUPE</pre>
<div class="block">The fraction of aligned bases that were filtered out because they were in reads marked as duplicates.</div>
</li>
</ul>
<a name="PCT_EXC_UNPAIRED">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_EXC_UNPAIRED</h4>
<pre>public&nbsp;double PCT_EXC_UNPAIRED</pre>
<div class="block">The fraction of aligned bases that were filtered out because they were in reads without a mapped mate pair.</div>
</li>
</ul>
<a name="PCT_EXC_BASEQ">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_EXC_BASEQ</h4>
<pre>public&nbsp;double PCT_EXC_BASEQ</pre>
<div class="block">The fraction of aligned bases that were filtered out because they were of low base quality (lower than MIN_BASE_QUALITY).</div>
</li>
</ul>
<a name="PCT_EXC_OVERLAP">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_EXC_OVERLAP</h4>
<pre>public&nbsp;double PCT_EXC_OVERLAP</pre>
<div class="block">The fraction of aligned bases that were filtered out because they were the second observation from an insert with overlapping reads.</div>
</li>
</ul>
<a name="PCT_EXC_CAPPED">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_EXC_CAPPED</h4>
<pre>public&nbsp;double PCT_EXC_CAPPED</pre>
<div class="block">The fraction of aligned bases that were filtered out because they would have raised coverage above COVERAGE_CAP.</div>
</li>
</ul>
<a name="PCT_EXC_TOTAL">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_EXC_TOTAL</h4>
<pre>public&nbsp;double PCT_EXC_TOTAL</pre>
<div class="block">The total fraction of aligned bases excluded due to all filters.</div>
</li>
</ul>
<a name="PCT_1X">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_1X</h4>
<pre>public&nbsp;double PCT_1X</pre>
<div class="block">The fraction of bases that attained at least 1X sequence coverage in post-filtering bases.</div>
</li>
</ul>
<a name="PCT_5X">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_5X</h4>
<pre>public&nbsp;double PCT_5X</pre>
<div class="block">The fraction of bases that attained at least 5X sequence coverage in post-filtering bases.</div>
</li>
</ul>
<a name="PCT_10X">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_10X</h4>
<pre>public&nbsp;double PCT_10X</pre>
<div class="block">The fraction of bases that attained at least 10X sequence coverage in post-filtering bases.</div>
</li>
</ul>
<a name="PCT_15X">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_15X</h4>
<pre>public&nbsp;double PCT_15X</pre>
<div class="block">The fraction of bases that attained at least 15X sequence coverage in post-filtering bases.</div>
</li>
</ul>
<a name="PCT_20X">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_20X</h4>
<pre>public&nbsp;double PCT_20X</pre>
<div class="block">The fraction of bases that attained at least 20X sequence coverage in post-filtering bases.</div>
</li>
</ul>
<a name="PCT_25X">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_25X</h4>
<pre>public&nbsp;double PCT_25X</pre>
<div class="block">The fraction of bases that attained at least 25X sequence coverage in post-filtering bases.</div>
</li>
</ul>
<a name="PCT_30X">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_30X</h4>
<pre>public&nbsp;double PCT_30X</pre>
<div class="block">The fraction of bases that attained at least 30X sequence coverage in post-filtering bases.</div>
</li>
</ul>
<a name="PCT_40X">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_40X</h4>
<pre>public&nbsp;double PCT_40X</pre>
<div class="block">The fraction of bases that attained at least 40X sequence coverage in post-filtering bases.</div>
</li>
</ul>
<a name="PCT_50X">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_50X</h4>
<pre>public&nbsp;double PCT_50X</pre>
<div class="block">The fraction of bases that attained at least 50X sequence coverage in post-filtering bases.</div>
</li>
</ul>
<a name="PCT_60X">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_60X</h4>
<pre>public&nbsp;double PCT_60X</pre>
<div class="block">The fraction of bases that attained at least 60X sequence coverage in post-filtering bases.</div>
</li>
</ul>
<a name="PCT_70X">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_70X</h4>
<pre>public&nbsp;double PCT_70X</pre>
<div class="block">The fraction of bases that attained at least 70X sequence coverage in post-filtering bases.</div>
</li>
</ul>
<a name="PCT_80X">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_80X</h4>
<pre>public&nbsp;double PCT_80X</pre>
<div class="block">The fraction of bases that attained at least 80X sequence coverage in post-filtering bases.</div>
</li>
</ul>
<a name="PCT_90X">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_90X</h4>
<pre>public&nbsp;double PCT_90X</pre>
<div class="block">The fraction of bases that attained at least 90X sequence coverage in post-filtering bases.</div>
</li>
</ul>
<a name="PCT_100X">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PCT_100X</h4>
<pre>public&nbsp;double PCT_100X</pre>
<div class="block">The fraction of bases that attained at least 100X sequence coverage in post-filtering bases.</div>
</li>
</ul>
<a name="HET_SNP_SENSITIVITY">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>HET_SNP_SENSITIVITY</h4>
<pre>public&nbsp;double HET_SNP_SENSITIVITY</pre>
<div class="block">The theoretical HET SNP sensitivity.</div>
</li>
</ul>
<a name="HET_SNP_Q">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>HET_SNP_Q</h4>
<pre>public&nbsp;double HET_SNP_Q</pre>
<div class="block">The Phred Scaled Q Score of the theoretical HET SNP sensitivity.</div>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="WgsMetrics--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>WgsMetrics</h4>
<pre>public&nbsp;WgsMetrics()</pre>
<div class="block">Create an instance of this metric that is not mergeable.</div>
</li>
</ul>
<a name="WgsMetrics-htsjdk.samtools.util.IntervalList-htsjdk.samtools.util.Histogram-htsjdk.samtools.util.Histogram-double-double-double-double-double-double-double-double-int-htsjdk.samtools.util.Histogram-int-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>WgsMetrics</h4>
<pre>public&nbsp;WgsMetrics(htsjdk.samtools.util.IntervalList&nbsp;intervals,
                  htsjdk.samtools.util.Histogram&lt;java.lang.Integer&gt;&nbsp;highQualityDepthHistogram,
                  htsjdk.samtools.util.Histogram&lt;java.lang.Integer&gt;&nbsp;unfilteredDepthHistogram,
                  double&nbsp;pctExcludedByAdapter,
                  double&nbsp;pctExcludedByMapq,
                  double&nbsp;pctExcludedByDupes,
                  double&nbsp;pctExcludedByPairing,
                  double&nbsp;pctExcludedByBaseq,
                  double&nbsp;pctExcludedByOverlap,
                  double&nbsp;pctExcludedByCapping,
                  double&nbsp;pctExcludeTotal,
                  int&nbsp;coverageCap,
                  htsjdk.samtools.util.Histogram&lt;java.lang.Integer&gt;&nbsp;unfilteredBaseQHistogram,
                  int&nbsp;theoreticalHetSensitivitySampleSize)</pre>
<div class="block">Create an instance of this metric that is mergeable.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>highQualityDepthHistogram</code> - the count of genomic positions observed for each observed depth. Excludes bases with quality below MINIMUM_BASE_QUALITY.</dd>
<dd><code>unfilteredDepthHistogram</code> - the depth histogram that includes all but quality 2 bases.</dd>
<dd><code>pctExcludedByAdapter</code> - the fraction of aligned bases that were filtered out because they were in reads with 0 mapping quality that looked like adapter sequence.</dd>
<dd><code>pctExcludedByMapq</code> - the fraction of aligned bases that were filtered out because they were in reads with low mapping quality.</dd>
<dd><code>pctExcludedByDupes</code> - the fraction of aligned bases that were filtered out because they were in reads marked as duplicates.</dd>
<dd><code>pctExcludedByPairing</code> - the fraction of bases that were filtered out because they were in reads without a mapped mate pair.</dd>
<dd><code>pctExcludedByBaseq</code> - the fraction of aligned bases that were filtered out because they were of low base quality.</dd>
<dd><code>pctExcludedByOverlap</code> - the fraction of aligned bases that were filtered out because they were the second observation from an insert with overlapping reads.</dd>
<dd><code>pctExcludedByCapping</code> - the fraction of aligned bases that were filtered out because they would have raised coverage above the capped value.</dd>
<dd><code>pctExcludeTotal</code> - the fraction of bases excluded across all filters.</dd>
<dd><code>coverageCap</code> - Treat positions with coverage exceeding this value as if they had coverage at this value.</dd>
<dd><code>unfilteredBaseQHistogram</code> - the count of bases observed with a given quality. Includes all but quality 2 bases.</dd>
<dd><code>theoreticalHetSensitivitySampleSize</code> - the sample size used for theoretical het sensitivity sampling.</dd>
</dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="merge-picard.analysis.MergeableMetricBase-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>merge</h4>
<pre>public&nbsp;<a href="../../picard/analysis/MergeableMetricBase.html" title="class in picard.analysis">MergeableMetricBase</a>&nbsp;merge(<a href="../../picard/analysis/MergeableMetricBase.html" title="class in picard.analysis">MergeableMetricBase</a>&nbsp;other)</pre>
<div class="block">Merges the various PCT_EXC_* metrics.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code><a href="../../picard/analysis/MergeableMetricBase.html#merge-picard.analysis.MergeableMetricBase-">merge</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/analysis/MergeableMetricBase.html" title="class in picard.analysis">MergeableMetricBase</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>other</code> - metric to merge into this one.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>result of merging, also known as "this"</dd>
</dl>
</li>
</ul>
<a name="calculateDerivedFields--">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>calculateDerivedFields</h4>
<pre>public&nbsp;void&nbsp;calculateDerivedFields()</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from class:&nbsp;<code><a href="../../picard/analysis/MergeableMetricBase.html#calculateDerivedFields--">MergeableMetricBase</a></code></span></div>
<div class="block">Placeholder method that will calculate the derived fields from the other ones.
 Classes that are derived from non-trivial derived classes should consider calling super.calculateDerivedFields() as well.
 Fields whose value will change due to this method should be annotated with <a href="../../picard/analysis/MergeableMetricBase.NoMergingIsDerived.html" title="annotation in picard.analysis"><code>NoMergingKeepsValue</code></a>.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code><a href="../../picard/analysis/MergeableMetricBase.html#calculateDerivedFields--">calculateDerivedFields</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/analysis/MergeableMetricBase.html" title="class in picard.analysis">MergeableMetricBase</a></code></dd>
</dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/analysis/TheoreticalSensitivityMetrics.html" title="class in picard.analysis"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/analysis/WgsMetricsProcessor.html" title="interface in picard.analysis"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/analysis/WgsMetrics.html" target="_top">Frames</a></li>
<li><a href="WgsMetrics.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.classes.inherited.from.class.picard.analysis.MergeableMetricBase">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
