<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_201) on Fri Oct 04 10:35:33 EDT 2019 -->
<title>CrosscheckFingerprints (picard 2.21.1-SNAPSHOT API)</title>
<meta name="date" content="2019-10-04">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="CrosscheckFingerprints (picard 2.21.1-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":10,"i1":10};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/fingerprint/ClusteredCrosscheckMetric.html" title="class in picard.fingerprint"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/fingerprint/CrosscheckMetric.html" title="class in picard.fingerprint"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/fingerprint/CrosscheckFingerprints.html" target="_top">Frames</a></li>
<li><a href="CrosscheckFingerprints.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">picard.fingerprint</div>
<h2 title="Class CrosscheckFingerprints" class="title">Class CrosscheckFingerprints</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li><a href="../../picard/cmdline/CommandLineProgram.html" title="class in picard.cmdline">picard.cmdline.CommandLineProgram</a></li>
<li>
<ul class="inheritance">
<li>picard.fingerprint.CrosscheckFingerprints</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>Direct Known Subclasses:</dt>
<dd><a href="../../picard/fingerprint/CrosscheckReadGroupFingerprints.html" title="class in picard.fingerprint">CrosscheckReadGroupFingerprints</a></dd>
</dl>
<hr>
<br>
<pre>@DocumentedFeature
public class <span class="typeNameLabel">CrosscheckFingerprints</span>
extends <a href="../../picard/cmdline/CommandLineProgram.html" title="class in picard.cmdline">CommandLineProgram</a></pre>
<div class="block">Checks that all data in the set of input files appear to come from the same
 individual. Can be used to compare according to readgroups, libraries, samples, or files.
 Operates on bams/sams and vcfs (including gvcfs).

 <h3>Summary</h3>
 Checks if all the genetic data within a set of files appear to come from the same individual.
 It quickly determines whether a "group's" genotype matches that of an input SAM/BAM/VCF by selective sampling,
 and has been designed to work well even for low-depth SAM/BAMs.
 <br/>
 The tool collects "fingerprints" (essentially genotype information from different parts of the genome)
 at the finest level available in the data (readgroup for SAM files
 and sample for VCF files) and then optionally aggregates it by library, sample or file, to increase power and provide
 results at the desired resolution. Output is in a "Moltenized" format, one row per comparison. The results will
 be emitted into a metric file for the class <a href="../../picard/fingerprint/CrosscheckMetric.html" title="class in picard.fingerprint"><code>CrosscheckMetric</code></a>.
 In this format the output will include the LOD score and also tumor-aware LOD score which can
 help assess identity even in the presence of a severe loss of heterozygosity with high purity (which could
 otherwise fail to notice that samples are from the same individual.)
 A matrix output is also available to facilitate visual inspection of crosscheck results.
 <br/>
 Since there can be many rows of output in the metric file, we recommend the use of <a href="../../picard/fingerprint/ClusterCrosscheckMetrics.html" title="class in picard.fingerprint"><code>ClusterCrosscheckMetrics</code></a>
 as a follow-up step to running CrosscheckFingerprints.
 <br/>
 There are cases where one would like to identify a few groups out of a collection of many possible groups (say
 to link a bam to it's correct sample in a multi-sample vcf. In this case one would not case for the cross-checking
 of the various samples in the VCF against each other, but only in checking the identity of the bam against the various
 samples in the vcf. The <a href="../../picard/fingerprint/CrosscheckFingerprints.html#SECOND_INPUT"><code>SECOND_INPUT</code></a> is provided for this use-case. With <a href="../../picard/fingerprint/CrosscheckFingerprints.html#SECOND_INPUT"><code>SECOND_INPUT</code></a> provided, CrosscheckFingerprints
 does the following:
 <il>
 <li>aggregation of data happens independently for the input files in <a href="../../picard/fingerprint/CrosscheckFingerprints.html#INPUT"><code>INPUT</code></a> and <a href="../../picard/fingerprint/CrosscheckFingerprints.html#SECOND_INPUT"><code>SECOND_INPUT</code></a>.</li>
 <li>aggregation of data happens at the SAMPLE level.</li>
 <li>each samples from <a href="../../picard/fingerprint/CrosscheckFingerprints.html#INPUT"><code>INPUT</code></a> will only be compared to that same sample in <a href="../../picard/fingerprint/CrosscheckFingerprints.html#INPUT"><code>INPUT</code></a>.</li>
 <li><a href="../../picard/fingerprint/CrosscheckFingerprints.html#MATRIX_OUTPUT"><code>MATRIX_OUTPUT</code></a> is disabled.</li>
 </il>
 <br/>
 In some cases, the groups collected may not have any observations (calls for a vcf, reads for a bam) at fingerprinting sites, or a sample in INPUT may be missing from the SECOND_INPUT.
 These cases are handled as follows:  If running in CHECK_SAME_SAMPLES mode with INPUT and SECOND_INPUT, and either INPUT or SECOND_INPUT includes a sample
 not found in the other, or contains a sample with no observations at any fingerprinting sites, an error will be logged and the tool will return EXIT_CODE_WHEN_MISMATCH.
 In all other running modes, when any group which is being crosschecked does not have any observations at fingerprinting sites, a warning is logged.  As long as there is at least
 one comparison where both sides have observations at fingerprinting sites, the tool will return zero.  However, if all comparisons have at least one side with no observations
 at fingerprinting sites, an error will be logged and the tool will return EXIT_CODE_WHEN_NO_VALID_CHECKS.
 <h3>Examples</h3>
 <h4>Check that all the readgroups from a sample match each other:</h4>
 <pre>
     java -jar picard.jar CrosscheckFingerprints \
          INPUT=sample.with.many.readgroups.bam \
          HAPLOTYPE_MAP=fingerprinting_haplotype_database.txt \
          LOD_THRESHOLD=-5 \
          OUTPUT=sample.crosscheck_metrics
 </pre>


 <h4>Check that all the readgroups match as expected when providing reads from two samples from the same individual:</h4>
 <pre>
     java -jar picard.jar CrosscheckFingerprints \
          INPUT=sample.one.with.many.readgroups.bam \
          INPUT=sample.two.with.many.readgroups.bam \
          HAPLOTYPE_MAP=fingerprinting_haplotype_database.txt \
          LOD_THRESHOLD=-5 \
          EXPECT_ALL_GROUPS_TO_MATCH=true \
          OUTPUT=sample.crosscheck_metrics
 </pre>


 <h3> Detailed Explanation</h3>

 This tool calculates the LOD score for identity check between "groups" of data in the INPUT files as defined by
 the CROSSCHECK_BY argument. A positive value indicates that the data seems to have come from the same individual
 or, in other words the identity checks out. The scale is logarithmic (base 10), so a LOD of 6 indicates
 that it is 1,000,000 more likely that the data matches the genotypes than not. A negative value indicates
 that the data do not match. A score that is near zero is inconclusive and can result from low coverage
 or non-informative genotypes. Each group is assigned a sample identifier (for SAM this is taken from the SM tag in
 the appropriate readgroup header line, for VCF this is taken from the column label in the file-header.
 After combining all the data from the same "group" together, an all-against-all comparison is performed. Results are
 categorized a <a href="../../picard/fingerprint/CrosscheckMetric.FingerprintResult.html" title="enum in picard.fingerprint"><code>CrosscheckMetric.FingerprintResult</code></a> enum: EXPECTED_MATCH, EXPECTED_MISMATCH, UNEXPECTED_MATCH, UNEXPECTED_MISMATCH,
 or AMBIGUOUS depending on the LOD score and on whether the sample identifiers of the groups agree: LOD scores that are
 less than LOD_THRESHOLD are considered mismatches, and those greater than -LOD_THRESHOLD are matches (between is ambiguous).
 If the sample identifiers are equal, the groups are expected to match. They are expected to mismatch otherwise.
 <br/>

 The identity check makes use of haplotype blocks defined in the HAPLOTYPE_MAP file to enable it to have higher
 statistical power for detecting identity or swap by aggregating data from several SNPs in the haplotype block. This
 enables an identity check of samples with very low coverage (e.g. ~1x mean coverage).
 <br/>
 When provided a VCF, the identity check looks at the PL, GL and GT fields (in that order) and uses the first one that
 it finds.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#ALLOW_DUPLICATE_READS">ALLOW_DUPLICATE_READS</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#CALCULATE_TUMOR_AWARE_RESULTS">CALCULATE_TUMOR_AWARE_RESULTS</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../picard/fingerprint/CrosscheckMetric.DataType.html" title="enum in picard.fingerprint">CrosscheckMetric.DataType</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#CROSSCHECK_BY">CROSSCHECK_BY</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>picard.fingerprint.Fingerprint.CrosscheckMode</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#CROSSCHECK_MODE">CROSSCHECK_MODE</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#EXIT_CODE_WHEN_MISMATCH">EXIT_CODE_WHEN_MISMATCH</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#EXIT_CODE_WHEN_NO_VALID_CHECKS">EXIT_CODE_WHEN_NO_VALID_CHECKS</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#EXPECT_ALL_GROUPS_TO_MATCH">EXPECT_ALL_GROUPS_TO_MATCH</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#GENOTYPING_ERROR_RATE">GENOTYPING_ERROR_RATE</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.io.File</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#HAPLOTYPE_MAP">HAPLOTYPE_MAP</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.List&lt;java.lang.String&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#INPUT">INPUT</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.io.File</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#INPUT_SAMPLE_FILE_MAP">INPUT_SAMPLE_FILE_MAP</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.io.File</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#INPUT_SAMPLE_MAP">INPUT_SAMPLE_MAP</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#LOD_THRESHOLD">LOD_THRESHOLD</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#LOSS_OF_HET_RATE">LOSS_OF_HET_RATE</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.io.File</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#MATRIX_OUTPUT">MATRIX_OUTPUT</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#NUM_THREADS">NUM_THREADS</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.io.File</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#OUTPUT">OUTPUT</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#OUTPUT_ERRORS_ONLY">OUTPUT_ERRORS_ONLY</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.List&lt;java.lang.String&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#SECOND_INPUT">SECOND_INPUT</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.io.File</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#SECOND_INPUT_SAMPLE_MAP">SECOND_INPUT_SAMPLE_MAP</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#TEST_INPUT_READABILITY">TEST_INPUT_READABILITY</a></span></code>&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="fields.inherited.from.class.picard.cmdline.CommandLineProgram">
<!--   -->
</a>
<h3>Fields inherited from class&nbsp;picard.cmdline.<a href="../../picard/cmdline/CommandLineProgram.html" title="class in picard.cmdline">CommandLineProgram</a></h3>
<code><a href="../../picard/cmdline/CommandLineProgram.html#COMPRESSION_LEVEL">COMPRESSION_LEVEL</a>, <a href="../../picard/cmdline/CommandLineProgram.html#CREATE_INDEX">CREATE_INDEX</a>, <a href="../../picard/cmdline/CommandLineProgram.html#CREATE_MD5_FILE">CREATE_MD5_FILE</a>, <a href="../../picard/cmdline/CommandLineProgram.html#GA4GH_CLIENT_SECRETS">GA4GH_CLIENT_SECRETS</a>, <a href="../../picard/cmdline/CommandLineProgram.html#MAX_ALLOWABLE_ONE_LINE_SUMMARY_LENGTH">MAX_ALLOWABLE_ONE_LINE_SUMMARY_LENGTH</a>, <a href="../../picard/cmdline/CommandLineProgram.html#MAX_RECORDS_IN_RAM">MAX_RECORDS_IN_RAM</a>, <a href="../../picard/cmdline/CommandLineProgram.html#QUIET">QUIET</a>, <a href="../../picard/cmdline/CommandLineProgram.html#REFERENCE_SEQUENCE">REFERENCE_SEQUENCE</a>, <a href="../../picard/cmdline/CommandLineProgram.html#referenceSequence">referenceSequence</a>, <a href="../../picard/cmdline/CommandLineProgram.html#specialArgumentsCollection">specialArgumentsCollection</a>, <a href="../../picard/cmdline/CommandLineProgram.html#TMP_DIR">TMP_DIR</a>, <a href="../../picard/cmdline/CommandLineProgram.html#USE_JDK_DEFLATER">USE_JDK_DEFLATER</a>, <a href="../../picard/cmdline/CommandLineProgram.html#USE_JDK_INFLATER">USE_JDK_INFLATER</a>, <a href="../../picard/cmdline/CommandLineProgram.html#VALIDATION_STRINGENCY">VALIDATION_STRINGENCY</a>, <a href="../../picard/cmdline/CommandLineProgram.html#VERBOSITY">VERBOSITY</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#CrosscheckFingerprints--">CrosscheckFingerprints</a></span>()</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>protected java.lang.String[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#customCommandLineValidation--">customCommandLineValidation</a></span>()</code>
<div class="block">Put any custom command-line validation in an override of this method.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/CrosscheckFingerprints.html#doWork--">doWork</a></span>()</code>
<div class="block">Do the work after command line has been parsed.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.picard.cmdline.CommandLineProgram">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;picard.cmdline.<a href="../../picard/cmdline/CommandLineProgram.html" title="class in picard.cmdline">CommandLineProgram</a></h3>
<code><a href="../../picard/cmdline/CommandLineProgram.html#getCommandLine--">getCommandLine</a>, <a href="../../picard/cmdline/CommandLineProgram.html#getCommandLineParser--">getCommandLineParser</a>, <a href="../../picard/cmdline/CommandLineProgram.html#getCommandLineParser-java.lang.Object-">getCommandLineParser</a>, <a href="../../picard/cmdline/CommandLineProgram.html#getDefaultHeaders--">getDefaultHeaders</a>, <a href="../../picard/cmdline/CommandLineProgram.html#getFaqLink--">getFaqLink</a>, <a href="../../picard/cmdline/CommandLineProgram.html#getMetricsFile--">getMetricsFile</a>, <a href="../../picard/cmdline/CommandLineProgram.html#getStandardUsagePreamble--">getStandardUsagePreamble</a>, <a href="../../picard/cmdline/CommandLineProgram.html#getStandardUsagePreamble-java.lang.Class-">getStandardUsagePreamble</a>, <a href="../../picard/cmdline/CommandLineProgram.html#getVersion--">getVersion</a>, <a href="../../picard/cmdline/CommandLineProgram.html#hasWebDocumentation-java.lang.Class-">hasWebDocumentation</a>, <a href="../../picard/cmdline/CommandLineProgram.html#instanceMain-java.lang.String:A-">instanceMain</a>, <a href="../../picard/cmdline/CommandLineProgram.html#instanceMainWithExit-java.lang.String:A-">instanceMainWithExit</a>, <a href="../../picard/cmdline/CommandLineProgram.html#makeReferenceArgumentCollection--">makeReferenceArgumentCollection</a>, <a href="../../picard/cmdline/CommandLineProgram.html#parseArgs-java.lang.String:A-">parseArgs</a>, <a href="../../picard/cmdline/CommandLineProgram.html#requiresReference--">requiresReference</a>, <a href="../../picard/cmdline/CommandLineProgram.html#setDefaultHeaders-java.util.List-">setDefaultHeaders</a>, <a href="../../picard/cmdline/CommandLineProgram.html#useLegacyParser-java.lang.Class-">useLegacyParser</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="INPUT">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>INPUT</h4>
<pre>@Argument(shortName="I",
          doc="One or more input files (or lists of files) with which to compare fingerprints.",
          minElements=1)
public&nbsp;java.util.List&lt;java.lang.String&gt; INPUT</pre>
</li>
</ul>
<a name="INPUT_SAMPLE_MAP">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>INPUT_SAMPLE_MAP</h4>
<pre>@Argument(doc="A tsv with two columns representing the sample as it appears in the INPUT data (in column 1) and the sample as it should be used for comparisons to SECOND_INPUT (in the second column). Need only include the samples that change. Values in column 1 should be unique. Values in column 2 should be unique even in union with the remaining unmapped samples. Should only be used with SECOND_INPUT. ",
          optional=true,
          mutex="INPUT_SAMPLE_FILE_MAP")
public&nbsp;java.io.File INPUT_SAMPLE_MAP</pre>
</li>
</ul>
<a name="INPUT_SAMPLE_FILE_MAP">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>INPUT_SAMPLE_FILE_MAP</h4>
<pre>@Argument(doc="A tsv with two columns representing the sample as it should be used for comparisons to SECOND_INPUT (in the first column) and  the source file (in INPUT) for the fingerprint (in the second column). Need only to include the samples that change. Values in column 1 should be unique even in union with the remaining unmapped samples. Values in column 2 should be unique in the file. Will error if more than one sample is found in a file (multi-sample vcf) pointed to in column 2. Should only be used in the presence of SECOND_INPUT. ",
          optional=true,
          mutex="INPUT_SAMPLE_MAP")
public&nbsp;java.io.File INPUT_SAMPLE_FILE_MAP</pre>
</li>
</ul>
<a name="SECOND_INPUT">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SECOND_INPUT</h4>
<pre>@Argument(shortName="SI",
          optional=true,
          mutex="MATRIX_OUTPUT",
          doc="A second set of input files (or lists of files) with which to compare fingerprints. If this option is provided the tool compares each sample in INPUT with the sample from SECOND_INPUT that has the same sample ID. In addition, data will be grouped by SAMPLE regardless of the value of CROSSCHECK_BY. When operating in this mode, each sample in INPUT must also have a corresponding sample in SECOND_INPUT. If this is violated, the tool will proceed to check the matching samples, but report the missing samples and return a non-zero error-code.")
public&nbsp;java.util.List&lt;java.lang.String&gt; SECOND_INPUT</pre>
</li>
</ul>
<a name="SECOND_INPUT_SAMPLE_MAP">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SECOND_INPUT_SAMPLE_MAP</h4>
<pre>@Argument(doc="A tsv with two columns representing the sample as it appears in the SECOND_INPUT data (in column 1) and the sample as it should be used for comparisons to INPUT (in the second column). Note that in case of unrolling files (file-of-filenames) one would need to reference the final file, i.e. the file that contains the genomic data. Need only include the samples that change. Values in column 1 should be unique. Values in column 2 should be unique even in union with the remaining unmapped samples. Should only be used with SECOND_INPUT. ",
          optional=true)
public&nbsp;java.io.File SECOND_INPUT_SAMPLE_MAP</pre>
</li>
</ul>
<a name="CROSSCHECK_MODE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>CROSSCHECK_MODE</h4>
<pre>@Argument(doc="An argument that controls how crosschecking with both INPUT and SECOND_INPUT should occur. ")
public&nbsp;picard.fingerprint.Fingerprint.CrosscheckMode CROSSCHECK_MODE</pre>
</li>
</ul>
<a name="OUTPUT">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>OUTPUT</h4>
<pre>@Argument(shortName="O",
          optional=true,
          doc="Optional output file to write metrics to. Default is to write to stdout.")
public&nbsp;java.io.File OUTPUT</pre>
</li>
</ul>
<a name="MATRIX_OUTPUT">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>MATRIX_OUTPUT</h4>
<pre>@Argument(shortName="MO",
          optional=true,
          doc="Optional output file to write matrix of LOD scores to. This is less informative than the metrics output and only contains Normal-Normal LOD score (i.e. doesn\'t account for Loss of Heterozygosity). It is however sometimes easier to use visually.",
          mutex="SECOND_INPUT")
public&nbsp;java.io.File MATRIX_OUTPUT</pre>
</li>
</ul>
<a name="HAPLOTYPE_MAP">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>HAPLOTYPE_MAP</h4>
<pre>@Argument(shortName="H",
          doc="The file lists a set of SNPs, optionally arranged in high-LD blocks, to be used for fingerprinting. See https://software.broadinstitute.org/gatk/documentation/article?id=9526 for details.")
public&nbsp;java.io.File HAPLOTYPE_MAP</pre>
</li>
</ul>
<a name="LOD_THRESHOLD">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>LOD_THRESHOLD</h4>
<pre>@Argument(shortName="LOD",
          doc="If any two groups (with the same sample name) match with a LOD score lower than the threshold the tool will exit with a non-zero code to indicate error. Program will also exit with an error if it finds two groups with different sample name that match with a LOD score greater than -LOD_THRESHOLD.\n\nLOD score 0 means equal likelihood that the groups match vs. come from different individuals, negative LOD score -N, mean 10^N time more likely that the groups are from different individuals, and +N means 10^N times more likely that the groups are from the same individual. ")
public&nbsp;double LOD_THRESHOLD</pre>
</li>
</ul>
<a name="CROSSCHECK_BY">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>CROSSCHECK_BY</h4>
<pre>@Argument(doc="Specificies which data-type should be used as the basic comparison unit. Fingerprints from readgroups can be \"rolled-up\" to the LIBRARY, SAMPLE, or FILE level before being compared. Fingerprints from VCF can be be compared by SAMPLE or FILE.")
public&nbsp;<a href="../../picard/fingerprint/CrosscheckMetric.DataType.html" title="enum in picard.fingerprint">CrosscheckMetric.DataType</a> CROSSCHECK_BY</pre>
</li>
</ul>
<a name="NUM_THREADS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>NUM_THREADS</h4>
<pre>@Argument(doc="The number of threads to use to process files and generate fingerprints.")
public&nbsp;int NUM_THREADS</pre>
</li>
</ul>
<a name="CALCULATE_TUMOR_AWARE_RESULTS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>CALCULATE_TUMOR_AWARE_RESULTS</h4>
<pre>@Argument(doc="specifies whether the Tumor-aware result should be calculated. These are time consuming and can roughly double the runtime of the tool. When crosschecking many groups not calculating the tumor-aware  results can result in a significant speedup.")
public&nbsp;boolean CALCULATE_TUMOR_AWARE_RESULTS</pre>
</li>
</ul>
<a name="ALLOW_DUPLICATE_READS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>ALLOW_DUPLICATE_READS</h4>
<pre>@Argument(doc="Allow the use of duplicate reads in performing the comparison. Can be useful when duplicate marking has been overly aggressive and coverage is low.")
public&nbsp;boolean ALLOW_DUPLICATE_READS</pre>
</li>
</ul>
<a name="GENOTYPING_ERROR_RATE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>GENOTYPING_ERROR_RATE</h4>
<pre>@Argument(doc="Assumed genotyping error rate that provides a floor on the probability that a genotype comes from the expected sample. Must be greater than zero. ")
public&nbsp;double GENOTYPING_ERROR_RATE</pre>
</li>
</ul>
<a name="OUTPUT_ERRORS_ONLY">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>OUTPUT_ERRORS_ONLY</h4>
<pre>@Argument(doc="If true then only groups that do not relate to each other as expected will have their LODs reported.")
public&nbsp;boolean OUTPUT_ERRORS_ONLY</pre>
</li>
</ul>
<a name="LOSS_OF_HET_RATE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>LOSS_OF_HET_RATE</h4>
<pre>@Argument(doc="The rate at which a heterozygous genotype in a normal sample turns into a homozygous (via loss of heterozygosity) in the tumor (model assumes independent events, so this needs to be larger than reality).",
          optional=true)
public&nbsp;double LOSS_OF_HET_RATE</pre>
</li>
</ul>
<a name="EXPECT_ALL_GROUPS_TO_MATCH">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>EXPECT_ALL_GROUPS_TO_MATCH</h4>
<pre>@Argument(doc="Expect all groups\' fingerprints to match, irrespective of their sample names.  By default (with this value set to false), groups (readgroups, libraries, files, or samples) with different sample names are expected to mismatch, and those with the same sample name are expected to match. ")
public&nbsp;boolean EXPECT_ALL_GROUPS_TO_MATCH</pre>
</li>
</ul>
<a name="EXIT_CODE_WHEN_MISMATCH">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>EXIT_CODE_WHEN_MISMATCH</h4>
<pre>@Argument(doc="When one or more mismatches between groups is detected, exit with this value instead of 0.")
public&nbsp;int EXIT_CODE_WHEN_MISMATCH</pre>
</li>
</ul>
<a name="EXIT_CODE_WHEN_NO_VALID_CHECKS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>EXIT_CODE_WHEN_NO_VALID_CHECKS</h4>
<pre>@Argument(doc="When all LOD score are zero, exit with this value.")
public&nbsp;int EXIT_CODE_WHEN_NO_VALID_CHECKS</pre>
</li>
</ul>
<a name="TEST_INPUT_READABILITY">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>TEST_INPUT_READABILITY</h4>
<pre>@Hidden
 @Argument(doc="When true code will check for readability on input files (this can be slow on cloud access)")
public&nbsp;boolean TEST_INPUT_READABILITY</pre>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="CrosscheckFingerprints--">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>CrosscheckFingerprints</h4>
<pre>public&nbsp;CrosscheckFingerprints()</pre>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="customCommandLineValidation--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>customCommandLineValidation</h4>
<pre>protected&nbsp;java.lang.String[]&nbsp;customCommandLineValidation()</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from class:&nbsp;<code><a href="../../picard/cmdline/CommandLineProgram.html#customCommandLineValidation--">CommandLineProgram</a></code></span></div>
<div class="block">Put any custom command-line validation in an override of this method.
 clp is initialized at this point and can be used to print usage and access argv.
 Any options set by command-line parser can be validated.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code><a href="../../picard/cmdline/CommandLineProgram.html#customCommandLineValidation--">customCommandLineValidation</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/cmdline/CommandLineProgram.html" title="class in picard.cmdline">CommandLineProgram</a></code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>null if command line is valid.  If command line is invalid, returns an array of error message
 to be written to the appropriate place.</dd>
</dl>
</li>
</ul>
<a name="doWork--">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>doWork</h4>
<pre>protected&nbsp;int&nbsp;doWork()</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from class:&nbsp;<code><a href="../../picard/cmdline/CommandLineProgram.html#doWork--">CommandLineProgram</a></code></span></div>
<div class="block">Do the work after command line has been parsed. RuntimeException may be
 thrown by this method, and are reported appropriately.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../picard/cmdline/CommandLineProgram.html#doWork--">doWork</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/cmdline/CommandLineProgram.html" title="class in picard.cmdline">CommandLineProgram</a></code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>program exit status.</dd>
</dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/fingerprint/ClusteredCrosscheckMetric.html" title="class in picard.fingerprint"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/fingerprint/CrosscheckMetric.html" title="class in picard.fingerprint"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/fingerprint/CrosscheckFingerprints.html" target="_top">Frames</a></li>
<li><a href="CrosscheckFingerprints.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
