<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_201) on Fri Oct 04 10:35:33 EDT 2019 -->
<title>FingerprintChecker (picard 2.21.1-SNAPSHOT API)</title>
<meta name="date" content="2019-10-04">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="FingerprintChecker (picard 2.21.1-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":9,"i1":9,"i2":9,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":10,"i10":10,"i11":10,"i12":10,"i13":10,"i14":10,"i15":10,"i16":10,"i17":10,"i18":9,"i19":10,"i20":10,"i21":10,"i22":10,"i23":10,"i24":10,"i25":10,"i26":10};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/fingerprint/FingerprintIdDetails.html" title="class in picard.fingerprint"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/fingerprint/FingerprintChecker.html" target="_top">Frames</a></li>
<li><a href="FingerprintChecker.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">picard.fingerprint</div>
<h2 title="Class FingerprintChecker" class="title">Class FingerprintChecker</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>picard.fingerprint.FingerprintChecker</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public class <span class="typeNameLabel">FingerprintChecker</span>
extends java.lang.Object</pre>
<div class="block">Major class that coordinates the activities involved in comparing genetic fingerprint
 data whether the source is from a genotyping platform or derived from sequence data.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#DEFAULT_GENOTYPING_ERROR_RATE">DEFAULT_GENOTYPING_ERROR_RATE</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#DEFAULT_MAXIMAL_PL_DIFFERENCE">DEFAULT_MAXIMAL_PL_DIFFERENCE</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#DEFAULT_MINIMUM_BASE_QUALITY">DEFAULT_MINIMUM_BASE_QUALITY</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#DEFAULT_MINIMUM_MAPPING_QUALITY">DEFAULT_MINIMUM_MAPPING_QUALITY</a></span></code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#FingerprintChecker-java.io.File-">FingerprintChecker</a></span>(java.io.File&nbsp;haplotypeData)</code>
<div class="block">Creates a fingerprint checker that will work with the set of haplotypes stored in
 the supplied file.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#FingerprintChecker-picard.fingerprint.HaplotypeMap-">FingerprintChecker</a></span>(<a href="../../picard/fingerprint/HaplotypeMap.html" title="class in picard.fingerprint">HaplotypeMap</a>&nbsp;haplotypes)</code>
<div class="block">Creates a fingerprint checker that will work with the set of haplotyped provided.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>static <a href="../../picard/fingerprint/MatchResults.html" title="class in picard.fingerprint">MatchResults</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#calculateMatchResults-picard.fingerprint.Fingerprint-picard.fingerprint.Fingerprint-">calculateMatchResults</a></span>(<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&nbsp;observedFp,
                     <a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&nbsp;expectedFp)</code>
<div class="block">Compares two fingerprints and calculates a MatchResults object which contains detailed
 information about the match (or mismatch) between fingerprints including the LOD score
 for whether or not the two are likely from the same sample.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>static <a href="../../picard/fingerprint/MatchResults.html" title="class in picard.fingerprint">MatchResults</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#calculateMatchResults-picard.fingerprint.Fingerprint-picard.fingerprint.Fingerprint-double-double-">calculateMatchResults</a></span>(<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&nbsp;observedFp,
                     <a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&nbsp;expectedFp,
                     double&nbsp;minPExpected,
                     double&nbsp;pLoH)</code>&nbsp;</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>static <a href="../../picard/fingerprint/MatchResults.html" title="class in picard.fingerprint">MatchResults</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#calculateMatchResults-picard.fingerprint.Fingerprint-picard.fingerprint.Fingerprint-double-double-boolean-boolean-">calculateMatchResults</a></span>(<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&nbsp;observedFp,
                     <a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&nbsp;expectedFp,
                     double&nbsp;minPExpected,
                     double&nbsp;pLoH,
                     boolean&nbsp;calculateLocusInfo,
                     boolean&nbsp;calculateTumorAwareLod)</code>
<div class="block">Compares two fingerprints and calculates a MatchResults object which contains detailed
 information about the match (or mismatch) between fingerprints including the LOD score
 for whether or not the two are likely from the same sample.</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../picard/fingerprint/FingerprintResults.html" title="class in picard.fingerprint">FingerprintResults</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#checkFingerprints-java.util.List-java.util.List-java.lang.String-boolean-">checkFingerprints</a></span>(java.util.List&lt;java.nio.file.Path&gt;&nbsp;samFiles,
                 java.util.List&lt;java.nio.file.Path&gt;&nbsp;genotypeFiles,
                 java.lang.String&nbsp;specificSample,
                 boolean&nbsp;ignoreReadGroups)</code>
<div class="block">Top level method to take a set of one or more SAM files and one or more Genotype files and compare
 each read group in each SAM file to each set of fingerprint genotypes.</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../picard/fingerprint/FingerprintResults.html" title="class in picard.fingerprint">FingerprintResults</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#checkFingerprintsFromPaths-java.util.List-java.util.List-java.lang.String-java.lang.String-">checkFingerprintsFromPaths</a></span>(java.util.List&lt;java.nio.file.Path&gt;&nbsp;observedGenotypeFiles,
                          java.util.List&lt;java.nio.file.Path&gt;&nbsp;expectedGenotypeFiles,
                          java.lang.String&nbsp;observedSample,
                          java.lang.String&nbsp;expectedSample)</code>
<div class="block">Top level method to take a set of one or more observed genotype (VCF) files and one or more expected genotype (VCF) files and compare
 one or more sample in the observed genotype file with one or more in the expected file and generate results for each set.</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>java.util.Map&lt;<a href="../../picard/fingerprint/FingerprintIdDetails.html" title="class in picard.fingerprint">FingerprintIdDetails</a>,<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#fingerprintFiles-java.util.Collection-int-int-java.util.concurrent.TimeUnit-">fingerprintFiles</a></span>(java.util.Collection&lt;java.nio.file.Path&gt;&nbsp;files,
                int&nbsp;threads,
                int&nbsp;waitTime,
                java.util.concurrent.TimeUnit&nbsp;waitTimeUnit)</code>
<div class="block">Fingerprints one or more SAM/BAM/VCF files at all available loci within the haplotype map, using multiple threads
 to speed up the processing.</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>java.util.Map&lt;<a href="../../picard/fingerprint/FingerprintIdDetails.html" title="class in picard.fingerprint">FingerprintIdDetails</a>,<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#fingerprintSamFile-java.nio.file.Path-htsjdk.samtools.util.IntervalList-">fingerprintSamFile</a></span>(java.nio.file.Path&nbsp;samFile,
                  htsjdk.samtools.util.IntervalList&nbsp;loci)</code>
<div class="block">Generates a Fingerprint per read group in the supplied SAM file using the loci provided in
 the interval list.</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>java.util.Map&lt;<a href="../../picard/fingerprint/FingerprintIdDetails.html" title="class in picard.fingerprint">FingerprintIdDetails</a>,<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#fingerprintVcf-java.nio.file.Path-">fingerprintVcf</a></span>(java.nio.file.Path&nbsp;vcfFile)</code>&nbsp;</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>htsjdk.samtools.SAMFileHeader</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#getHeader--">getHeader</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>htsjdk.samtools.util.IntervalList</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#getLociToGenotype-java.util.Collection-">getLociToGenotype</a></span>(java.util.Collection&lt;<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;&nbsp;fingerprints)</code>
<div class="block">Takes a set of fingerprints and returns an IntervalList containing all the loci that
 can be productively examined in sequencing data to compare to one or more of the
 fingerprints.</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>java.io.File</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#getReferenceFasta--">getReferenceFasta</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>htsjdk.samtools.ValidationStringency</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#getValidationStringency--">getValidationStringency</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>java.util.Map&lt;java.lang.String,<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#identifyContaminant-java.nio.file.Path-double-int-">identifyContaminant</a></span>(java.nio.file.Path&nbsp;samFile,
                   double&nbsp;contamination,
                   int&nbsp;locusMaxReads)</code>
<div class="block">Generates a per-sample Fingerprint for the contaminant in the supplied SAM file.</div>
</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>java.util.Map&lt;java.lang.String,<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#loadFingerprints-java.nio.file.Path-java.lang.String-">loadFingerprints</a></span>(java.nio.file.Path&nbsp;fingerprintFile,
                java.lang.String&nbsp;specificSample)</code>
<div class="block">Loads genotypes from the supplied file into one or more Fingerprint objects and returns them in a
 Map of Sample->Fingerprint.</div>
</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>java.util.Map&lt;java.lang.String,<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#loadFingerprintsFromIndexedVcf-java.nio.file.Path-java.lang.String-">loadFingerprintsFromIndexedVcf</a></span>(java.nio.file.Path&nbsp;fingerprintFile,
                              java.lang.String&nbsp;specificSample)</code>
<div class="block">Loads genotypes from the supplied file into one or more Fingerprint objects and returns them in a
 Map of Sample->Fingerprint.</div>
</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>java.util.Map&lt;java.lang.String,<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#loadFingerprintsFromNonIndexedVcf-java.nio.file.Path-java.lang.String-">loadFingerprintsFromNonIndexedVcf</a></span>(java.nio.file.Path&nbsp;fingerprintFile,
                                 java.lang.String&nbsp;specificSample)</code>
<div class="block">Loads genotypes from the supplied file into one or more Fingerprint objects and returns them in a
 Map of Sample->Fingerprint.</div>
</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>java.util.Map&lt;java.lang.String,<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#loadFingerprintsFromQueriableReader-htsjdk.variant.vcf.VCFFileReader-java.lang.String-java.nio.file.Path-">loadFingerprintsFromQueriableReader</a></span>(htsjdk.variant.vcf.VCFFileReader&nbsp;reader,
                                   java.lang.String&nbsp;specificSample,
                                   java.nio.file.Path&nbsp;source)</code>
<div class="block">Loads genotypes from the supplied reader into one or more Fingerprint objects and returns them in a
 Map of Sample->Fingerprint.</div>
</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code>java.util.Map&lt;java.lang.String,<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#loadFingerprintsFromVariantContexts-java.lang.Iterable-java.lang.String-java.nio.file.Path-">loadFingerprintsFromVariantContexts</a></span>(java.lang.Iterable&lt;htsjdk.variant.variantcontext.VariantContext&gt;&nbsp;iterable,
                                   java.lang.String&nbsp;specificSample,
                                   java.nio.file.Path&nbsp;source)</code>
<div class="block">Loads genotypes from the supplied file into one or more Fingerprint objects and returns them in a
 Map of Sample->Fingerprint.</div>
</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code>protected static &lt;T&gt;&nbsp;java.util.List&lt;T&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#randomSublist-java.util.List-int-">randomSublist</a></span>(java.util.List&lt;T&gt;&nbsp;list,
             int&nbsp;n)</code>
<div class="block">A small utility function to choose n random elements (un-shuffled) from a list</div>
</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#setAllowDuplicateReads-boolean-">setAllowDuplicateReads</a></span>(boolean&nbsp;allowDuplicateReads)</code>
<div class="block">Sets whether duplicate reads should be allowed when calling genotypes from SAM files.</div>
</td>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#setGenotypingErrorRate-double-">setGenotypingErrorRate</a></span>(double&nbsp;genotypingErrorRate)</code>
<div class="block">Sets the assumed genotyping error rate used when accurate error rates are not available.</div>
</td>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#setmaximalPLDifference-int-">setmaximalPLDifference</a></span>(int&nbsp;maximalPLDifference)</code>
<div class="block">Sets the maximal difference in PL scores considered when reading PLs from a VCF.</div>
</td>
</tr>
<tr id="i22" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#setMinimumBaseQuality-int-">setMinimumBaseQuality</a></span>(int&nbsp;minimumBaseQuality)</code>
<div class="block">Sets the minimum base quality for bases used when computing a fingerprint from sequence data.</div>
</td>
</tr>
<tr id="i23" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#setMinimumMappingQuality-int-">setMinimumMappingQuality</a></span>(int&nbsp;minimumMappingQuality)</code>
<div class="block">Sets the minimum mapping quality for reads used when computing fingerprints from sequence data.</div>
</td>
</tr>
<tr id="i24" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#setpLossofHet-double-">setpLossofHet</a></span>(double&nbsp;pLossofHet)</code>&nbsp;</td>
</tr>
<tr id="i25" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#setReferenceFasta-java.io.File-">setReferenceFasta</a></span>(java.io.File&nbsp;referenceFasta)</code>&nbsp;</td>
</tr>
<tr id="i26" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/FingerprintChecker.html#setValidationStringency-htsjdk.samtools.ValidationStringency-">setValidationStringency</a></span>(htsjdk.samtools.ValidationStringency&nbsp;validationStringency)</code>&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="DEFAULT_GENOTYPING_ERROR_RATE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DEFAULT_GENOTYPING_ERROR_RATE</h4>
<pre>public static final&nbsp;double DEFAULT_GENOTYPING_ERROR_RATE</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../constant-values.html#picard.fingerprint.FingerprintChecker.DEFAULT_GENOTYPING_ERROR_RATE">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a name="DEFAULT_MINIMUM_MAPPING_QUALITY">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DEFAULT_MINIMUM_MAPPING_QUALITY</h4>
<pre>public static final&nbsp;int DEFAULT_MINIMUM_MAPPING_QUALITY</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../constant-values.html#picard.fingerprint.FingerprintChecker.DEFAULT_MINIMUM_MAPPING_QUALITY">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a name="DEFAULT_MINIMUM_BASE_QUALITY">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DEFAULT_MINIMUM_BASE_QUALITY</h4>
<pre>public static final&nbsp;int DEFAULT_MINIMUM_BASE_QUALITY</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../constant-values.html#picard.fingerprint.FingerprintChecker.DEFAULT_MINIMUM_BASE_QUALITY">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a name="DEFAULT_MAXIMAL_PL_DIFFERENCE">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>DEFAULT_MAXIMAL_PL_DIFFERENCE</h4>
<pre>public static final&nbsp;int DEFAULT_MAXIMAL_PL_DIFFERENCE</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../constant-values.html#picard.fingerprint.FingerprintChecker.DEFAULT_MAXIMAL_PL_DIFFERENCE">Constant Field Values</a></dd>
</dl>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="FingerprintChecker-java.io.File-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>FingerprintChecker</h4>
<pre>public&nbsp;FingerprintChecker(java.io.File&nbsp;haplotypeData)</pre>
<div class="block">Creates a fingerprint checker that will work with the set of haplotypes stored in
 the supplied file.</div>
</li>
</ul>
<a name="FingerprintChecker-picard.fingerprint.HaplotypeMap-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>FingerprintChecker</h4>
<pre>public&nbsp;FingerprintChecker(<a href="../../picard/fingerprint/HaplotypeMap.html" title="class in picard.fingerprint">HaplotypeMap</a>&nbsp;haplotypes)</pre>
<div class="block">Creates a fingerprint checker that will work with the set of haplotyped provided.</div>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getValidationStringency--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getValidationStringency</h4>
<pre>public&nbsp;htsjdk.samtools.ValidationStringency&nbsp;getValidationStringency()</pre>
</li>
</ul>
<a name="setValidationStringency-htsjdk.samtools.ValidationStringency-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setValidationStringency</h4>
<pre>public&nbsp;void&nbsp;setValidationStringency(htsjdk.samtools.ValidationStringency&nbsp;validationStringency)</pre>
</li>
</ul>
<a name="getReferenceFasta--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReferenceFasta</h4>
<pre>public&nbsp;java.io.File&nbsp;getReferenceFasta()</pre>
</li>
</ul>
<a name="setMinimumBaseQuality-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMinimumBaseQuality</h4>
<pre>public&nbsp;void&nbsp;setMinimumBaseQuality(int&nbsp;minimumBaseQuality)</pre>
<div class="block">Sets the minimum base quality for bases used when computing a fingerprint from sequence data.</div>
</li>
</ul>
<a name="setMinimumMappingQuality-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setMinimumMappingQuality</h4>
<pre>public&nbsp;void&nbsp;setMinimumMappingQuality(int&nbsp;minimumMappingQuality)</pre>
<div class="block">Sets the minimum mapping quality for reads used when computing fingerprints from sequence data.</div>
</li>
</ul>
<a name="setGenotypingErrorRate-double-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGenotypingErrorRate</h4>
<pre>public&nbsp;void&nbsp;setGenotypingErrorRate(double&nbsp;genotypingErrorRate)</pre>
<div class="block">Sets the assumed genotyping error rate used when accurate error rates are not available.</div>
</li>
</ul>
<a name="setmaximalPLDifference-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setmaximalPLDifference</h4>
<pre>public&nbsp;void&nbsp;setmaximalPLDifference(int&nbsp;maximalPLDifference)</pre>
<div class="block">Sets the maximal difference in PL scores considered when reading PLs from a VCF.</div>
</li>
</ul>
<a name="getHeader--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getHeader</h4>
<pre>public&nbsp;htsjdk.samtools.SAMFileHeader&nbsp;getHeader()</pre>
</li>
</ul>
<a name="setAllowDuplicateReads-boolean-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAllowDuplicateReads</h4>
<pre>public&nbsp;void&nbsp;setAllowDuplicateReads(boolean&nbsp;allowDuplicateReads)</pre>
<div class="block">Sets whether duplicate reads should be allowed when calling genotypes from SAM files. This is
 useful when comparing read groups within a SAM file and individual read groups show artifactually
 high duplication (e.g. a single-ended read group mixed in with paired-end read groups).</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>allowDuplicateReads</code> - should fingerprinting use duplicate reads?</dd>
</dl>
</li>
</ul>
<a name="setpLossofHet-double-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setpLossofHet</h4>
<pre>public&nbsp;void&nbsp;setpLossofHet(double&nbsp;pLossofHet)</pre>
</li>
</ul>
<a name="loadFingerprints-java.nio.file.Path-java.lang.String-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadFingerprints</h4>
<pre>public&nbsp;java.util.Map&lt;java.lang.String,<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;&nbsp;loadFingerprints(java.nio.file.Path&nbsp;fingerprintFile,
                                                                    java.lang.String&nbsp;specificSample)</pre>
<div class="block">Loads genotypes from the supplied file into one or more Fingerprint objects and returns them in a
 Map of Sample->Fingerprint.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>fingerprintFile</code> - - VCF file containing genotypes for one or more samples</dd>
<dd><code>specificSample</code> - - null to load genotypes for all samples contained in the file or the name
                        of an individual sample to load (and exclude all others).</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a Map of Sample name to Fingerprint</dd>
</dl>
</li>
</ul>
<a name="loadFingerprintsFromNonIndexedVcf-java.nio.file.Path-java.lang.String-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadFingerprintsFromNonIndexedVcf</h4>
<pre>public&nbsp;java.util.Map&lt;java.lang.String,<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;&nbsp;loadFingerprintsFromNonIndexedVcf(java.nio.file.Path&nbsp;fingerprintFile,
                                                                                     java.lang.String&nbsp;specificSample)</pre>
<div class="block">Loads genotypes from the supplied file into one or more Fingerprint objects and returns them in a
 Map of Sample->Fingerprint.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>fingerprintFile</code> - - VCF file containing genotypes for one or more samples</dd>
<dd><code>specificSample</code> - - null to load genotypes for all samples contained in the file or the name
                        of an individual sample to load (and exclude all others).</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a Map of Sample name to Fingerprint</dd>
</dl>
</li>
</ul>
<a name="loadFingerprintsFromVariantContexts-java.lang.Iterable-java.lang.String-java.nio.file.Path-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadFingerprintsFromVariantContexts</h4>
<pre>public&nbsp;java.util.Map&lt;java.lang.String,<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;&nbsp;loadFingerprintsFromVariantContexts(java.lang.Iterable&lt;htsjdk.variant.variantcontext.VariantContext&gt;&nbsp;iterable,
                                                                                       java.lang.String&nbsp;specificSample,
                                                                                       java.nio.file.Path&nbsp;source)</pre>
<div class="block">Loads genotypes from the supplied file into one or more Fingerprint objects and returns them in a
 Map of Sample->Fingerprint.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>iterable</code> - - an iterable over variantContexts containing genotypes for one or more samples</dd>
<dd><code>specificSample</code> - - null to load genotypes for all samples contained in the file or the name
                       of an individual sample to load (and exclude all others).</dd>
<dd><code>source</code> - The path of the source file used. used to emit errors, and annotate the fingerprints.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a Map of Sample name to Fingerprint</dd>
</dl>
</li>
</ul>
<a name="loadFingerprintsFromIndexedVcf-java.nio.file.Path-java.lang.String-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadFingerprintsFromIndexedVcf</h4>
<pre>public&nbsp;java.util.Map&lt;java.lang.String,<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;&nbsp;loadFingerprintsFromIndexedVcf(java.nio.file.Path&nbsp;fingerprintFile,
                                                                                  java.lang.String&nbsp;specificSample)</pre>
<div class="block">Loads genotypes from the supplied file into one or more Fingerprint objects and returns them in a
 Map of Sample->Fingerprint.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>fingerprintFile</code> - - VCF file containing genotypes for one or more samples</dd>
<dd><code>specificSample</code> - - null to load genotypes for all samples contained in the file or the name
                        of an individual sample to load (and exclude all others).</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a Map of Sample name to Fingerprint</dd>
</dl>
</li>
</ul>
<a name="loadFingerprintsFromQueriableReader-htsjdk.variant.vcf.VCFFileReader-java.lang.String-java.nio.file.Path-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>loadFingerprintsFromQueriableReader</h4>
<pre>public&nbsp;java.util.Map&lt;java.lang.String,<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;&nbsp;loadFingerprintsFromQueriableReader(htsjdk.variant.vcf.VCFFileReader&nbsp;reader,
                                                                                       java.lang.String&nbsp;specificSample,
                                                                                       java.nio.file.Path&nbsp;source)</pre>
<div class="block">Loads genotypes from the supplied reader into one or more Fingerprint objects and returns them in a
 Map of Sample->Fingerprint.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>reader</code> - - VCF reader containing genotypes for one or more samples</dd>
<dd><code>specificSample</code> - - null to load genotypes for all samples contained in the file or the name
                       of an individual sample to load (and exclude all others).</dd>
<dd><code>source</code> - The path of the source file used. used to emit errors.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a Map of Sample name to Fingerprint</dd>
</dl>
</li>
</ul>
<a name="getLociToGenotype-java.util.Collection-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLociToGenotype</h4>
<pre>public&nbsp;htsjdk.samtools.util.IntervalList&nbsp;getLociToGenotype(java.util.Collection&lt;<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;&nbsp;fingerprints)</pre>
<div class="block">Takes a set of fingerprints and returns an IntervalList containing all the loci that
 can be productively examined in sequencing data to compare to one or more of the
 fingerprints.</div>
</li>
</ul>
<a name="fingerprintVcf-java.nio.file.Path-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fingerprintVcf</h4>
<pre>public&nbsp;java.util.Map&lt;<a href="../../picard/fingerprint/FingerprintIdDetails.html" title="class in picard.fingerprint">FingerprintIdDetails</a>,<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;&nbsp;fingerprintVcf(java.nio.file.Path&nbsp;vcfFile)</pre>
</li>
</ul>
<a name="fingerprintSamFile-java.nio.file.Path-htsjdk.samtools.util.IntervalList-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fingerprintSamFile</h4>
<pre>public&nbsp;java.util.Map&lt;<a href="../../picard/fingerprint/FingerprintIdDetails.html" title="class in picard.fingerprint">FingerprintIdDetails</a>,<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;&nbsp;fingerprintSamFile(java.nio.file.Path&nbsp;samFile,
                                                                          htsjdk.samtools.util.IntervalList&nbsp;loci)</pre>
<div class="block">Generates a Fingerprint per read group in the supplied SAM file using the loci provided in
 the interval list.</div>
</li>
</ul>
<a name="identifyContaminant-java.nio.file.Path-double-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>identifyContaminant</h4>
<pre>public&nbsp;java.util.Map&lt;java.lang.String,<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;&nbsp;identifyContaminant(java.nio.file.Path&nbsp;samFile,
                                                                       double&nbsp;contamination,
                                                                       int&nbsp;locusMaxReads)</pre>
<div class="block">Generates a per-sample Fingerprint for the contaminant in the supplied SAM file.
 Data is aggregated by sample, not read-group.</div>
</li>
</ul>
<a name="randomSublist-java.util.List-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>randomSublist</h4>
<pre>protected static&nbsp;&lt;T&gt;&nbsp;java.util.List&lt;T&gt;&nbsp;randomSublist(java.util.List&lt;T&gt;&nbsp;list,
                                                     int&nbsp;n)</pre>
<div class="block">A small utility function to choose n random elements (un-shuffled) from a list</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>list</code> - A list of elements</dd>
<dd><code>n</code> - a number of elements requested from list</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a list of n randomly chosen (but in the original order) elements from list.
 If the list has less than n elements it is returned in its entirety.</dd>
</dl>
</li>
</ul>
<a name="fingerprintFiles-java.util.Collection-int-int-java.util.concurrent.TimeUnit-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fingerprintFiles</h4>
<pre>public&nbsp;java.util.Map&lt;<a href="../../picard/fingerprint/FingerprintIdDetails.html" title="class in picard.fingerprint">FingerprintIdDetails</a>,<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&gt;&nbsp;fingerprintFiles(java.util.Collection&lt;java.nio.file.Path&gt;&nbsp;files,
                                                                        int&nbsp;threads,
                                                                        int&nbsp;waitTime,
                                                                        java.util.concurrent.TimeUnit&nbsp;waitTimeUnit)</pre>
<div class="block">Fingerprints one or more SAM/BAM/VCF files at all available loci within the haplotype map, using multiple threads
 to speed up the processing.</div>
</li>
</ul>
<a name="checkFingerprints-java.util.List-java.util.List-java.lang.String-boolean-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>checkFingerprints</h4>
<pre>public&nbsp;java.util.List&lt;<a href="../../picard/fingerprint/FingerprintResults.html" title="class in picard.fingerprint">FingerprintResults</a>&gt;&nbsp;checkFingerprints(java.util.List&lt;java.nio.file.Path&gt;&nbsp;samFiles,
                                                            java.util.List&lt;java.nio.file.Path&gt;&nbsp;genotypeFiles,
                                                            java.lang.String&nbsp;specificSample,
                                                            boolean&nbsp;ignoreReadGroups)</pre>
<div class="block">Top level method to take a set of one or more SAM files and one or more Genotype files and compare
 each read group in each SAM file to each set of fingerprint genotypes.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>samFiles</code> - the list of SAM files to fingerprint</dd>
<dd><code>genotypeFiles</code> - the list of genotype files from which to pull fingerprint genotypes</dd>
<dd><code>specificSample</code> - an optional single sample who's genotypes to load from the supplied files</dd>
<dd><code>ignoreReadGroups</code> - aggregate data into one fingerprint per file, instead of splitting by RG</dd>
</dl>
</li>
</ul>
<a name="checkFingerprintsFromPaths-java.util.List-java.util.List-java.lang.String-java.lang.String-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>checkFingerprintsFromPaths</h4>
<pre>public&nbsp;java.util.List&lt;<a href="../../picard/fingerprint/FingerprintResults.html" title="class in picard.fingerprint">FingerprintResults</a>&gt;&nbsp;checkFingerprintsFromPaths(java.util.List&lt;java.nio.file.Path&gt;&nbsp;observedGenotypeFiles,
                                                                     java.util.List&lt;java.nio.file.Path&gt;&nbsp;expectedGenotypeFiles,
                                                                     java.lang.String&nbsp;observedSample,
                                                                     java.lang.String&nbsp;expectedSample)</pre>
<div class="block">Top level method to take a set of one or more observed genotype (VCF) files and one or more expected genotype (VCF) files and compare
 one or more sample in the observed genotype file with one or more in the expected file and generate results for each set.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>observedGenotypeFiles</code> - The list of genotype files containing observed calls, from which to pull fingerprint genotypes</dd>
<dd><code>expectedGenotypeFiles</code> - The list of genotype files containing expected calls, from which to pull fingerprint genotypes</dd>
<dd><code>observedSample</code> - an optional single sample whose genotypes to load from the observed genotype file (if null, use all)</dd>
<dd><code>expectedSample</code> - an optional single sample whose genotypes to load from the expected genotype file (if null, use all)</dd>
</dl>
</li>
</ul>
<a name="calculateMatchResults-picard.fingerprint.Fingerprint-picard.fingerprint.Fingerprint-double-double-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>calculateMatchResults</h4>
<pre>public static&nbsp;<a href="../../picard/fingerprint/MatchResults.html" title="class in picard.fingerprint">MatchResults</a>&nbsp;calculateMatchResults(<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&nbsp;observedFp,
                                                 <a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&nbsp;expectedFp,
                                                 double&nbsp;minPExpected,
                                                 double&nbsp;pLoH)</pre>
</li>
</ul>
<a name="calculateMatchResults-picard.fingerprint.Fingerprint-picard.fingerprint.Fingerprint-double-double-boolean-boolean-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>calculateMatchResults</h4>
<pre>public static&nbsp;<a href="../../picard/fingerprint/MatchResults.html" title="class in picard.fingerprint">MatchResults</a>&nbsp;calculateMatchResults(<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&nbsp;observedFp,
                                                 <a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&nbsp;expectedFp,
                                                 double&nbsp;minPExpected,
                                                 double&nbsp;pLoH,
                                                 boolean&nbsp;calculateLocusInfo,
                                                 boolean&nbsp;calculateTumorAwareLod)</pre>
<div class="block">Compares two fingerprints and calculates a MatchResults object which contains detailed
 information about the match (or mismatch) between fingerprints including the LOD score
 for whether or not the two are likely from the same sample.
 <p>
 If comparing sequencing data to genotype data then the sequencing data should be passed
 as the observedFp and the genotype data as the expectedFp in order to get the best output.
 <p>
 In the cases where the most likely genotypes from the two fingerprints do not match the
 lExpectedSample is Max(actualpExpectedSample, minPExpected).</div>
</li>
</ul>
<a name="calculateMatchResults-picard.fingerprint.Fingerprint-picard.fingerprint.Fingerprint-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>calculateMatchResults</h4>
<pre>public static&nbsp;<a href="../../picard/fingerprint/MatchResults.html" title="class in picard.fingerprint">MatchResults</a>&nbsp;calculateMatchResults(<a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&nbsp;observedFp,
                                                 <a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a>&nbsp;expectedFp)</pre>
<div class="block">Compares two fingerprints and calculates a MatchResults object which contains detailed
 information about the match (or mismatch) between fingerprints including the LOD score
 for whether or not the two are likely from the same sample.
 <p>
 If comparing sequencing data to genotype data then the sequencing data should be passed
 as the observedFp and the genotype data as the expectedFp in order to get the best output.</div>
</li>
</ul>
<a name="setReferenceFasta-java.io.File-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>setReferenceFasta</h4>
<pre>public&nbsp;void&nbsp;setReferenceFasta(java.io.File&nbsp;referenceFasta)</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/fingerprint/FingerprintIdDetails.html" title="class in picard.fingerprint"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/fingerprint/FingerprintChecker.html" target="_top">Frames</a></li>
<li><a href="FingerprintChecker.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
