<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_201) on Fri Oct 04 10:35:33 EDT 2019 -->
<title>HaplotypeProbabilitiesFromSequence (picard 2.21.1-SNAPSHOT API)</title>
<meta name="date" content="2019-10-04">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="HaplotypeProbabilitiesFromSequence (picard 2.21.1-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":10,"i10":10,"i11":10,"i12":10};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromGenotypeLikelihoods.html" title="class in picard.fingerprint"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/fingerprint/HaplotypeProbabilityOfNormalGivenTumor.html" title="class in picard.fingerprint"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/fingerprint/HaplotypeProbabilitiesFromSequence.html" target="_top">Frames</a></li>
<li><a href="HaplotypeProbabilitiesFromSequence.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.classes.inherited.from.class.picard.fingerprint.HaplotypeProbabilities">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">picard.fingerprint</div>
<h2 title="Class HaplotypeProbabilitiesFromSequence" class="title">Class HaplotypeProbabilitiesFromSequence</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li><a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">picard.fingerprint.HaplotypeProbabilities</a></li>
<li>
<ul class="inheritance">
<li>picard.fingerprint.HaplotypeProbabilitiesFromSequence</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>Direct Known Subclasses:</dt>
<dd><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromContaminatorSequence.html" title="class in picard.fingerprint">HaplotypeProbabilitiesFromContaminatorSequence</a></dd>
</dl>
<hr>
<br>
<pre>public class <span class="typeNameLabel">HaplotypeProbabilitiesFromSequence</span>
extends <a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">HaplotypeProbabilities</a></pre>
<div class="block">Represents the probability of the underlying haplotype given the data. By convention the
 alleles stored for each SNP are in phase.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="nested.classes.inherited.from.class.picard.fingerprint.HaplotypeProbabilities">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;picard.fingerprint.<a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">HaplotypeProbabilities</a></h3>
<code><a href="../../picard/fingerprint/HaplotypeProbabilities.Genotype.html" title="enum in picard.fingerprint">HaplotypeProbabilities.Genotype</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html#obsAllele1">obsAllele1</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html#obsAllele2">obsAllele2</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html#obsAlleleOther">obsAlleleOther</a></span></code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html#HaplotypeProbabilitiesFromSequence-picard.fingerprint.HaplotypeBlock-">HaplotypeProbabilitiesFromSequence</a></span>(<a href="../../picard/fingerprint/HaplotypeBlock.html" title="class in picard.fingerprint">HaplotypeBlock</a>&nbsp;haplotypeBlock)</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html#addToProbs-picard.fingerprint.Snp-byte-byte-">addToProbs</a></span>(<a href="../../picard/fingerprint/Snp.html" title="class in picard.fingerprint">Snp</a>&nbsp;snp,
          byte&nbsp;base,
          byte&nbsp;qual)</code>
<div class="block">Adds a base observation with the observed quality to the evidence for this haplotype
 based on the fact that the SNP is part of the haplotype.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html#getFractionUnexpectedAlleleObs--">getFractionUnexpectedAlleleObs</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>double[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html#getLikelihoods--">getLikelihoods</a></span>()</code>
<div class="block">getter for likelihoods;</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html#getLodMostProbableGenotype--">getLodMostProbableGenotype</a></span>()</code>
<div class="block">Overridden to calculate the LOD from the loglikelihoods instead of the probabilities
 because it will allow for more accurate calculation before overflowing.</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>double[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html#getLogLikelihoods--">getLogLikelihoods</a></span>()</code>
<div class="block">Since this class uses loglikelihoods natively, we override and return the native variable</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html#getObsAllele1--">getObsAllele1</a></span>()</code>
<div class="block">Returns the number of bases/reads that support the first allele.</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html#getObsAllele2--">getObsAllele2</a></span>()</code>
<div class="block">Returns the number of bases/reads that support the second allele.</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>double[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html#getPosteriorProbabilities--">getPosteriorProbabilities</a></span>()</code>
<div class="block">getter for posteriors</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code><a href="../../picard/fingerprint/Snp.html" title="class in picard.fingerprint">Snp</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html#getRepresentativeSnp--">getRepresentativeSnp</a></span>()</code>
<div class="block">Simple returns the SNP from the haplotype that has the lowest genome coordinate.</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html#getTotalObs--">getTotalObs</a></span>()</code>
<div class="block">Gets the total number of observations presented at this locus.</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html#hasEvidence--">hasEvidence</a></span>()</code>
<div class="block">Returns true if evidence has been added, false if the probabilities are just the priors.</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html#merge-picard.fingerprint.HaplotypeProbabilities-">merge</a></span>(<a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">HaplotypeProbabilities</a>&nbsp;other)</code>
<div class="block">Merges information from another haplotype probabilities object for the same haplotype into
 this object.</div>
</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html#setLogLikelihoods-double:A-">setLogLikelihoods</a></span>(double[]&nbsp;ll)</code>&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.picard.fingerprint.HaplotypeProbabilities">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;picard.fingerprint.<a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">HaplotypeProbabilities</a></h3>
<code><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getHaplotype--">getHaplotype</a>, <a href="../../picard/fingerprint/HaplotypeProbabilities.html#getMostLikelyGenotype-picard.fingerprint.Snp-">getMostLikelyGenotype</a>, <a href="../../picard/fingerprint/HaplotypeProbabilities.html#getMostLikelyHaplotype--">getMostLikelyHaplotype</a>, <a href="../../picard/fingerprint/HaplotypeProbabilities.html#getPriorProbablities--">getPriorProbablities</a>, <a href="../../picard/fingerprint/HaplotypeProbabilities.html#scaledEvidenceProbabilityUsingGenotypeFrequencies-double:A-">scaledEvidenceProbabilityUsingGenotypeFrequencies</a>, <a href="../../picard/fingerprint/HaplotypeProbabilities.html#shiftedLogEvidenceProbability--">shiftedLogEvidenceProbability</a>, <a href="../../picard/fingerprint/HaplotypeProbabilities.html#shiftedLogEvidenceProbabilityGivenOtherEvidence-picard.fingerprint.HaplotypeProbabilities-">shiftedLogEvidenceProbabilityGivenOtherEvidence</a>, <a href="../../picard/fingerprint/HaplotypeProbabilities.html#shiftedLogEvidenceProbabilityUsingGenotypeFrequencies-double:A-">shiftedLogEvidenceProbabilityUsingGenotypeFrequencies</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="obsAllele1">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>obsAllele1</h4>
<pre>protected&nbsp;int obsAllele1</pre>
</li>
</ul>
<a name="obsAllele2">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>obsAllele2</h4>
<pre>protected&nbsp;int obsAllele2</pre>
</li>
</ul>
<a name="obsAlleleOther">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>obsAlleleOther</h4>
<pre>protected&nbsp;int obsAlleleOther</pre>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="HaplotypeProbabilitiesFromSequence-picard.fingerprint.HaplotypeBlock-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>HaplotypeProbabilitiesFromSequence</h4>
<pre>public&nbsp;HaplotypeProbabilitiesFromSequence(<a href="../../picard/fingerprint/HaplotypeBlock.html" title="class in picard.fingerprint">HaplotypeBlock</a>&nbsp;haplotypeBlock)</pre>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="hasEvidence--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>hasEvidence</h4>
<pre>public&nbsp;boolean&nbsp;hasEvidence()</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from class:&nbsp;<code><a href="../../picard/fingerprint/HaplotypeProbabilities.html#hasEvidence--">HaplotypeProbabilities</a></code></span></div>
<div class="block">Returns true if evidence has been added, false if the probabilities are just the priors.</div>
</li>
</ul>
<a name="addToProbs-picard.fingerprint.Snp-byte-byte-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addToProbs</h4>
<pre>public&nbsp;void&nbsp;addToProbs(<a href="../../picard/fingerprint/Snp.html" title="class in picard.fingerprint">Snp</a>&nbsp;snp,
                       byte&nbsp;base,
                       byte&nbsp;qual)</pre>
<div class="block">Adds a base observation with the observed quality to the evidence for this haplotype
 based on the fact that the SNP is part of the haplotype.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>snp</code> - The snp in the HaplotypeBlock to which evidence is being added</dd>
<dd><code>base</code> - the base observed</dd>
<dd><code>qual</code> - the quality of the observed base</dd>
</dl>
</li>
</ul>
<a name="merge-picard.fingerprint.HaplotypeProbabilities-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>merge</h4>
<pre>public&nbsp;void&nbsp;merge(<a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">HaplotypeProbabilities</a>&nbsp;other)</pre>
<div class="block">Merges information from another haplotype probabilities object for the same haplotype into
 this object. Useful for when probabilities need to  be merged to levels higher than the
 read group, e.g. the sample or individual.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>other</code> - Another haplotype probabilities object to merge in</dd>
</dl>
</li>
</ul>
<a name="getObsAllele1--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getObsAllele1</h4>
<pre>public&nbsp;int&nbsp;getObsAllele1()</pre>
<div class="block">Returns the number of bases/reads that support the first allele.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getObsAllele1--">getObsAllele1</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">HaplotypeProbabilities</a></code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>int</dd>
</dl>
</li>
</ul>
<a name="getObsAllele2--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getObsAllele2</h4>
<pre>public&nbsp;int&nbsp;getObsAllele2()</pre>
<div class="block">Returns the number of bases/reads that support the second allele.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getObsAllele2--">getObsAllele2</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">HaplotypeProbabilities</a></code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>int</dd>
</dl>
</li>
</ul>
<a name="getTotalObs--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTotalObs</h4>
<pre>public&nbsp;int&nbsp;getTotalObs()</pre>
<div class="block">Gets the total number of observations presented at this locus.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getTotalObs--">getTotalObs</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">HaplotypeProbabilities</a></code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>int</dd>
</dl>
</li>
</ul>
<a name="getFractionUnexpectedAlleleObs--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFractionUnexpectedAlleleObs</h4>
<pre>public&nbsp;double&nbsp;getFractionUnexpectedAlleleObs()</pre>
</li>
</ul>
<a name="getRepresentativeSnp--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getRepresentativeSnp</h4>
<pre>public&nbsp;<a href="../../picard/fingerprint/Snp.html" title="class in picard.fingerprint">Snp</a>&nbsp;getRepresentativeSnp()</pre>
<div class="block">Simple returns the SNP from the haplotype that has the lowest genome coordinate.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getRepresentativeSnp--">getRepresentativeSnp</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">HaplotypeProbabilities</a></code></dd>
</dl>
</li>
</ul>
<a name="getPosteriorProbabilities--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getPosteriorProbabilities</h4>
<pre>public&nbsp;double[]&nbsp;getPosteriorProbabilities()</pre>
<div class="block">getter for posteriors</div>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getPosteriorProbabilities--">getPosteriorProbabilities</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">HaplotypeProbabilities</a></code></dd>
</dl>
</li>
</ul>
<a name="getLikelihoods--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLikelihoods</h4>
<pre>public&nbsp;double[]&nbsp;getLikelihoods()</pre>
<div class="block">getter for likelihoods;</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getLikelihoods--">getLikelihoods</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">HaplotypeProbabilities</a></code></dd>
</dl>
</li>
</ul>
<a name="getLogLikelihoods--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLogLikelihoods</h4>
<pre>public&nbsp;double[]&nbsp;getLogLikelihoods()</pre>
<div class="block">Since this class uses loglikelihoods natively, we override and return the native variable</div>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getLogLikelihoods--">getLogLikelihoods</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">HaplotypeProbabilities</a></code></dd>
</dl>
</li>
</ul>
<a name="setLogLikelihoods-double:A-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setLogLikelihoods</h4>
<pre>public&nbsp;void&nbsp;setLogLikelihoods(double[]&nbsp;ll)</pre>
</li>
</ul>
<a name="getLodMostProbableGenotype--">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getLodMostProbableGenotype</h4>
<pre>public&nbsp;double&nbsp;getLodMostProbableGenotype()</pre>
<div class="block">Overridden to calculate the LOD from the loglikelihoods instead of the probabilities
 because it will allow for more accurate calculation before overflowing.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getLodMostProbableGenotype--">getLodMostProbableGenotype</a></code>&nbsp;in class&nbsp;<code><a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">HaplotypeProbabilities</a></code></dd>
</dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromGenotypeLikelihoods.html" title="class in picard.fingerprint"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/fingerprint/HaplotypeProbabilityOfNormalGivenTumor.html" title="class in picard.fingerprint"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/fingerprint/HaplotypeProbabilitiesFromSequence.html" target="_top">Frames</a></li>
<li><a href="HaplotypeProbabilitiesFromSequence.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.classes.inherited.from.class.picard.fingerprint.HaplotypeProbabilities">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
