<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_201) on Fri Oct 04 10:35:33 EDT 2019 -->
<title>HaplotypeProbabilities (picard 2.21.1-SNAPSHOT API)</title>
<meta name="date" content="2019-10-04">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="HaplotypeProbabilities (picard 2.21.1-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":10,"i1":6,"i2":10,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":10,"i10":6,"i11":10,"i12":10,"i13":6,"i14":10,"i15":10,"i16":10,"i17":10};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],4:["t3","Abstract Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/fingerprint/HaplotypeMap.html" title="class in picard.fingerprint"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/fingerprint/HaplotypeProbabilities.Genotype.html" title="enum in picard.fingerprint"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/fingerprint/HaplotypeProbabilities.html" target="_top">Frames</a></li>
<li><a href="HaplotypeProbabilities.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">picard.fingerprint</div>
<h2 title="Class HaplotypeProbabilities" class="title">Class HaplotypeProbabilities</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>picard.fingerprint.HaplotypeProbabilities</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>Direct Known Subclasses:</dt>
<dd><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromGenotype.html" title="class in picard.fingerprint">HaplotypeProbabilitiesFromGenotype</a>, <a href="../../picard/fingerprint/HaplotypeProbabilitiesFromGenotypeLikelihoods.html" title="class in picard.fingerprint">HaplotypeProbabilitiesFromGenotypeLikelihoods</a>, <a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html" title="class in picard.fingerprint">HaplotypeProbabilitiesFromSequence</a>, <a href="../../picard/fingerprint/HaplotypeProbabilityOfNormalGivenTumor.html" title="class in picard.fingerprint">HaplotypeProbabilityOfNormalGivenTumor</a></dd>
</dl>
<hr>
<br>
<pre>public abstract class <span class="typeNameLabel">HaplotypeProbabilities</span>
extends java.lang.Object</pre>
<div class="block">Abstract class for storing and calculating various likelihoods and probabilities
 for haplotype alleles given evidence.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.Genotype.html" title="enum in picard.fingerprint">HaplotypeProbabilities.Genotype</a></span></code>
<div class="block">Log10(P(evidence| haplotype)) for the 3 different possible haplotypes
 {aa, ab, bb}</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier</th>
<th class="colLast" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected </code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#HaplotypeProbabilities-picard.fingerprint.HaplotypeBlock-">HaplotypeProbabilities</a></span>(<a href="../../picard/fingerprint/HaplotypeBlock.html" title="class in picard.fingerprint">HaplotypeBlock</a>&nbsp;haplotypeBlock)</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t3" class="tableTab"><span><a href="javascript:show(4);">Abstract Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code><a href="../../picard/fingerprint/HaplotypeBlock.html" title="class in picard.fingerprint">HaplotypeBlock</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getHaplotype--">getHaplotype</a></span>()</code>
<div class="block">Returns the haplotype for which the probabilities apply.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>abstract double[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getLikelihoods--">getLikelihoods</a></span>()</code>
<div class="block">Returns the likelihoods, in order, of the AA, Aa and aa haplotypes given the evidence</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getLodMostProbableGenotype--">getLodMostProbableGenotype</a></span>()</code>
<div class="block">Returns the LOD score between the most probable haplotype and the second most probable.</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>double[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getLogLikelihoods--">getLogLikelihoods</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code><a href="../../picard/fingerprint/DiploidGenotype.html" title="enum in picard.fingerprint">DiploidGenotype</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getMostLikelyGenotype-picard.fingerprint.Snp-">getMostLikelyGenotype</a></span>(<a href="../../picard/fingerprint/Snp.html" title="class in picard.fingerprint">Snp</a>&nbsp;snp)</code>
<div class="block">Gets the genotype for this Snp given the most likely haplotype.</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code><a href="../../picard/fingerprint/DiploidHaplotype.html" title="enum in picard.fingerprint">DiploidHaplotype</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getMostLikelyHaplotype--">getMostLikelyHaplotype</a></span>()</code>
<div class="block">Gets the most likely haplotype given the probabilities.</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getObsAllele1--">getObsAllele1</a></span>()</code>
<div class="block">Returns the number of observations of alleles supporting the first/major haplotype allele.</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getObsAllele2--">getObsAllele2</a></span>()</code>
<div class="block">Returns the number of observations of alleles supporting the second/minor haplotype allele.</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>double[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getPosteriorProbabilities--">getPosteriorProbabilities</a></span>()</code>
<div class="block">Returns the posterior probabilities using the population frequency as a prior.</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>double[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getPriorProbablities--">getPriorProbablities</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>abstract <a href="../../picard/fingerprint/Snp.html" title="class in picard.fingerprint">Snp</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getRepresentativeSnp--">getRepresentativeSnp</a></span>()</code>
<div class="block">Returns a representative SNP for this haplotype.</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#getTotalObs--">getTotalObs</a></span>()</code>
<div class="block">Returns the total number of observations of any allele.</div>
</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#hasEvidence--">hasEvidence</a></span>()</code>
<div class="block">Returns true if evidence has been added, false if the probabilities are just the priors.</div>
</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>abstract void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#merge-picard.fingerprint.HaplotypeProbabilities-">merge</a></span>(<a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">HaplotypeProbabilities</a>&nbsp;other)</code>
<div class="block">Merges in the likelihood information from the supplied haplotype probabilities object.</div>
</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#scaledEvidenceProbabilityUsingGenotypeFrequencies-double:A-">scaledEvidenceProbabilityUsingGenotypeFrequencies</a></span>(double[]&nbsp;genotypeFrequencies)</code>
<div class="block">This function returns the scaled probability of the evidence collected
 given a vector of priors on the haplotype using the internal likelihood, which may be
 scaled by an unknown factor.</div>
</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#shiftedLogEvidenceProbability--">shiftedLogEvidenceProbability</a></span>()</code>
<div class="block">Returns log (p(evidence)) + c assuming that the prior on haplotypes is given by
 the internal haplotypeFrequencies</div>
</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#shiftedLogEvidenceProbabilityGivenOtherEvidence-picard.fingerprint.HaplotypeProbabilities-">shiftedLogEvidenceProbabilityGivenOtherEvidence</a></span>(<a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">HaplotypeProbabilities</a>&nbsp;otherHp)</code>
<div class="block">returns the log-probability the evidence, using as priors the posteriors of another object</div>
</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/fingerprint/HaplotypeProbabilities.html#shiftedLogEvidenceProbabilityUsingGenotypeFrequencies-double:A-">shiftedLogEvidenceProbabilityUsingGenotypeFrequencies</a></span>(double[]&nbsp;genotypeFrequencies)</code>&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="HaplotypeProbabilities-picard.fingerprint.HaplotypeBlock-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>HaplotypeProbabilities</h4>
<pre>protected&nbsp;HaplotypeProbabilities(<a href="../../picard/fingerprint/HaplotypeBlock.html" title="class in picard.fingerprint">HaplotypeBlock</a>&nbsp;haplotypeBlock)</pre>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getHaplotype--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getHaplotype</h4>
<pre>public&nbsp;<a href="../../picard/fingerprint/HaplotypeBlock.html" title="class in picard.fingerprint">HaplotypeBlock</a>&nbsp;getHaplotype()</pre>
<div class="block">Returns the haplotype for which the probabilities apply.</div>
</li>
</ul>
<a name="getPriorProbablities--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getPriorProbablities</h4>
<pre>public&nbsp;double[]&nbsp;getPriorProbablities()</pre>
</li>
</ul>
<a name="getPosteriorProbabilities--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getPosteriorProbabilities</h4>
<pre>public&nbsp;double[]&nbsp;getPosteriorProbabilities()</pre>
<div class="block">Returns the posterior probabilities using the population frequency as a prior.</div>
</li>
</ul>
<a name="getLikelihoods--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLikelihoods</h4>
<pre>public abstract&nbsp;double[]&nbsp;getLikelihoods()</pre>
<div class="block">Returns the likelihoods, in order, of the AA, Aa and aa haplotypes given the evidence
 <p>
 Mathematically this is P(evidence | haplotype) where haplotype={AA,Aa,aa}.</div>
</li>
</ul>
<a name="getLogLikelihoods--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLogLikelihoods</h4>
<pre>public&nbsp;double[]&nbsp;getLogLikelihoods()</pre>
</li>
</ul>
<a name="getRepresentativeSnp--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getRepresentativeSnp</h4>
<pre>public abstract&nbsp;<a href="../../picard/fingerprint/Snp.html" title="class in picard.fingerprint">Snp</a>&nbsp;getRepresentativeSnp()</pre>
<div class="block">Returns a representative SNP for this haplotype. Different subclasses may implement this in
 different ways, but should do so in a deterministic/repeatable fashion.</div>
</li>
</ul>
<a name="getObsAllele1--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getObsAllele1</h4>
<pre>public&nbsp;int&nbsp;getObsAllele1()</pre>
<div class="block">Returns the number of observations of alleles supporting the first/major haplotype allele.
 Strictly this doesn't make sense for all subclasses, but it's nice to have it part of the API so
 a default implementation is provided here.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>int</dd>
</dl>
</li>
</ul>
<a name="getObsAllele2--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getObsAllele2</h4>
<pre>public&nbsp;int&nbsp;getObsAllele2()</pre>
<div class="block">Returns the number of observations of alleles supporting the second/minor haplotype allele.
 Strictly this doesn't make sense for all subclasses, but it's nice to have it part of the API so
 a default implementation is provided here.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>int</dd>
</dl>
</li>
</ul>
<a name="getTotalObs--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTotalObs</h4>
<pre>public&nbsp;int&nbsp;getTotalObs()</pre>
<div class="block">Returns the total number of observations of any allele.
 Strictly this doesn't make sense for all subclasses, but it's nice to have it part of the API so
 a default implementation is provided here.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>int</dd>
</dl>
</li>
</ul>
<a name="hasEvidence--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>hasEvidence</h4>
<pre>public&nbsp;boolean&nbsp;hasEvidence()</pre>
<div class="block">Returns true if evidence has been added, false if the probabilities are just the priors.</div>
</li>
</ul>
<a name="merge-picard.fingerprint.HaplotypeProbabilities-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>merge</h4>
<pre>public abstract&nbsp;void&nbsp;merge(<a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">HaplotypeProbabilities</a>&nbsp;other)</pre>
<div class="block">Merges in the likelihood information from the supplied haplotype probabilities object.</div>
</li>
</ul>
<a name="getMostLikelyHaplotype--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMostLikelyHaplotype</h4>
<pre>public&nbsp;<a href="../../picard/fingerprint/DiploidHaplotype.html" title="enum in picard.fingerprint">DiploidHaplotype</a>&nbsp;getMostLikelyHaplotype()</pre>
<div class="block">Gets the most likely haplotype given the probabilities.</div>
</li>
</ul>
<a name="getMostLikelyGenotype-picard.fingerprint.Snp-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMostLikelyGenotype</h4>
<pre>public&nbsp;<a href="../../picard/fingerprint/DiploidGenotype.html" title="enum in picard.fingerprint">DiploidGenotype</a>&nbsp;getMostLikelyGenotype(<a href="../../picard/fingerprint/Snp.html" title="class in picard.fingerprint">Snp</a>&nbsp;snp)</pre>
<div class="block">Gets the genotype for this Snp given the most likely haplotype.</div>
</li>
</ul>
<a name="scaledEvidenceProbabilityUsingGenotypeFrequencies-double:A-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>scaledEvidenceProbabilityUsingGenotypeFrequencies</h4>
<pre>public&nbsp;double&nbsp;scaledEvidenceProbabilityUsingGenotypeFrequencies(double[]&nbsp;genotypeFrequencies)</pre>
<div class="block">This function returns the scaled probability of the evidence collected
 given a vector of priors on the haplotype using the internal likelihood, which may be
 scaled by an unknown factor. This factor causes the result to be scaled, hence the name.
 <p>
 Mathematically:
 <p>
 P(Evidence| P(h_i)=F_i) = \sum_i P(Evidence | h_i) P(h_i)
 = \sum_i P(Evidence | h_i) F_i
 = c * \sum_i Likelihood_i * F_i
 <p>
 Here, h_i are the three possible haplotypes, F_i are the given priors, and Likelihood_i
 are the stored likelihoods which are scaled from the actually likelihoods by an unknown
 factor, c. Note that the calculation ignores the internal haplotype probabilities (i.e. priors)</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genotypeFrequencies</code> - vector of (possibly scaled) probabilities of the three haplotypes</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>P(evidence | P_h)) / c</dd>
</dl>
</li>
</ul>
<a name="shiftedLogEvidenceProbabilityUsingGenotypeFrequencies-double:A-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>shiftedLogEvidenceProbabilityUsingGenotypeFrequencies</h4>
<pre>public&nbsp;double&nbsp;shiftedLogEvidenceProbabilityUsingGenotypeFrequencies(double[]&nbsp;genotypeFrequencies)</pre>
</li>
</ul>
<a name="shiftedLogEvidenceProbabilityGivenOtherEvidence-picard.fingerprint.HaplotypeProbabilities-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>shiftedLogEvidenceProbabilityGivenOtherEvidence</h4>
<pre>public&nbsp;double&nbsp;shiftedLogEvidenceProbabilityGivenOtherEvidence(<a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">HaplotypeProbabilities</a>&nbsp;otherHp)</pre>
<div class="block">returns the log-probability the evidence, using as priors the posteriors of another object</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>otherHp</code> - an additional HaplotypeProbabilities object representing the same underlying HaplotypeBlock</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>log10(P(evidence| P(h_i)=P(h_i|otherHp) ) + c where c is an unknown constant</dd>
</dl>
</li>
</ul>
<a name="shiftedLogEvidenceProbability--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>shiftedLogEvidenceProbability</h4>
<pre>public&nbsp;double&nbsp;shiftedLogEvidenceProbability()</pre>
<div class="block">Returns log (p(evidence)) + c assuming that the prior on haplotypes is given by
 the internal haplotypeFrequencies</div>
</li>
</ul>
<a name="getLodMostProbableGenotype--">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getLodMostProbableGenotype</h4>
<pre>public&nbsp;double&nbsp;getLodMostProbableGenotype()</pre>
<div class="block">Returns the LOD score between the most probable haplotype and the second most probable.</div>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/fingerprint/HaplotypeMap.html" title="class in picard.fingerprint"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/fingerprint/HaplotypeProbabilities.Genotype.html" title="enum in picard.fingerprint"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/fingerprint/HaplotypeProbabilities.html" target="_top">Frames</a></li>
<li><a href="HaplotypeProbabilities.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
