<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_201) on Fri Oct 04 10:35:35 EDT 2019 -->
<title>picard.fingerprint (picard 2.21.1-SNAPSHOT API)</title>
<meta name="date" content="2019-10-04">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="picard.fingerprint (picard 2.21.1-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li class="navBarCell1Rev">Package</li>
<li>Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/filter/package-summary.html">Prev&nbsp;Package</a></li>
<li><a href="../../picard/illumina/package-summary.html">Next&nbsp;Package</a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/fingerprint/package-summary.html" target="_top">Frames</a></li>
<li><a href="package-summary.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h1 title="Package" class="title">Package&nbsp;picard.fingerprint</h1>
</div>
<div class="contentContainer">
<ul class="blockList">
<li class="blockList">
<table class="typeSummary" border="0" cellpadding="3" cellspacing="0" summary="Class Summary table, listing classes, and an explanation">
<caption><span>Class Summary</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/fingerprint/CalculateFingerprintMetrics.html" title="class in picard.fingerprint">CalculateFingerprintMetrics</a></td>
<td class="colLast">
<div class="block">Calculates various metrics on a sample fingerprint, indicating whether the fingerprint satisfies the assumptions we have.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/fingerprint/CheckFingerprint.html" title="class in picard.fingerprint">CheckFingerprint</a></td>
<td class="colLast">
<div class="block">Checks the sample identity of the sequence/genotype data in the provided file (SAM/BAM or VCF)
 against a set of known genotypes in the supplied genotype file (in VCF format).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/fingerprint/ClusterCrosscheckMetrics.html" title="class in picard.fingerprint">ClusterCrosscheckMetrics</a></td>
<td class="colLast">
<div class="block">Summary</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/fingerprint/ClusteredCrosscheckMetric.html" title="class in picard.fingerprint">ClusteredCrosscheckMetric</a></td>
<td class="colLast">
<div class="block">A metric class to hold the result of <a href="../../picard/fingerprint/ClusterCrosscheckMetrics.html" title="class in picard.fingerprint"><code>ClusterCrosscheckMetrics</code></a> fingerprints.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/fingerprint/CrosscheckFingerprints.html" title="class in picard.fingerprint">CrosscheckFingerprints</a></td>
<td class="colLast">
<div class="block">Checks that all data in the set of input files appear to come from the same
 individual.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/fingerprint/CrosscheckMetric.html" title="class in picard.fingerprint">CrosscheckMetric</a></td>
<td class="colLast">
<div class="block">A class to hold the result of crosschecking fingerprints.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/fingerprint/CrosscheckReadGroupFingerprints.html" title="class in picard.fingerprint">CrosscheckReadGroupFingerprints</a></td>
<td class="colLast">Deprecated
<div class="block"><span class="deprecationComment">6/6/2017 Use <a href="../../picard/fingerprint/CrosscheckFingerprints.html" title="class in picard.fingerprint"><code>CrosscheckFingerprints</code></a> instead.</span></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/fingerprint/Fingerprint.html" title="class in picard.fingerprint">Fingerprint</a></td>
<td class="colLast">
<div class="block">class to represent a genetic fingerprint as a set of HaplotypeProbabilities
 objects that give the relative probabilities of each of the possible haplotypes
 at a locus.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/fingerprint/FingerprintChecker.html" title="class in picard.fingerprint">FingerprintChecker</a></td>
<td class="colLast">
<div class="block">Major class that coordinates the activities involved in comparing genetic fingerprint
 data whether the source is from a genotyping platform or derived from sequence data.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/fingerprint/FingerprintIdDetails.html" title="class in picard.fingerprint">FingerprintIdDetails</a></td>
<td class="colLast">
<div class="block">class to hold the details of a element of fingerprinting PU tag</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/fingerprint/FingerprintMetrics.html" title="class in picard.fingerprint">FingerprintMetrics</a></td>
<td class="colLast">
<div class="block">Class for holding metrics on a single fingerprint.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/fingerprint/FingerprintResults.html" title="class in picard.fingerprint">FingerprintResults</a></td>
<td class="colLast">
<div class="block">Class that is used to represent the results of comparing a read group within a SAM file, or a sample
 within a VCF against one or more set of fingerprint genotypes.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/fingerprint/FingerprintUtils.html" title="class in picard.fingerprint">FingerprintUtils</a></td>
<td class="colLast">
<div class="block">A set of utilities used in the fingerprinting environment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/fingerprint/FingerprintUtils.VariantContextSet.html" title="class in picard.fingerprint">FingerprintUtils.VariantContextSet</a></td>
<td class="colLast">
<div class="block">A class that holds VariantContexts sorted by genomic position</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/fingerprint/HaplotypeBlock.html" title="class in picard.fingerprint">HaplotypeBlock</a></td>
<td class="colLast">
<div class="block">Represents information about a group of SNPs that form a haplotype in perfect LD
 with one another.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/fingerprint/HaplotypeMap.html" title="class in picard.fingerprint">HaplotypeMap</a></td>
<td class="colLast">
<div class="block">A collection of metadata about Haplotype Blocks including multiple in memory "indices" of the data
 to make it easy to query the correct HaplotypeBlock or Snp by snp names, positions etc.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/fingerprint/HaplotypeProbabilities.html" title="class in picard.fingerprint">HaplotypeProbabilities</a></td>
<td class="colLast">
<div class="block">Abstract class for storing and calculating various likelihoods and probabilities
 for haplotype alleles given evidence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromContaminatorSequence.html" title="class in picard.fingerprint">HaplotypeProbabilitiesFromContaminatorSequence</a></td>
<td class="colLast">
<div class="block">Represents the probability of the underlying haplotype of the contaminating sample given the data.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromGenotype.html" title="class in picard.fingerprint">HaplotypeProbabilitiesFromGenotype</a></td>
<td class="colLast">
<div class="block">Represents a set of HaplotypeProbabilities that were derived from a single SNP
 genotype at a point in time.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromGenotypeLikelihoods.html" title="class in picard.fingerprint">HaplotypeProbabilitiesFromGenotypeLikelihoods</a></td>
<td class="colLast">
<div class="block">Represents the likelihood of the HaplotypeBlock given the GenotypeLikelihoods (GL field from a VCF, which is actually a log10-likelihood)
 for each of the SNPs in that block.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/fingerprint/HaplotypeProbabilitiesFromSequence.html" title="class in picard.fingerprint">HaplotypeProbabilitiesFromSequence</a></td>
<td class="colLast">
<div class="block">Represents the probability of the underlying haplotype given the data.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/fingerprint/HaplotypeProbabilityOfNormalGivenTumor.html" title="class in picard.fingerprint">HaplotypeProbabilityOfNormalGivenTumor</a></td>
<td class="colLast">
<div class="block">A wrapper class for any HaplotypeProbabilities instance that will assume that the given evidence is that of a tumor sample and
 provide an hp for the normal sample that tumor came from.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/fingerprint/IdentifyContaminant.html" title="class in picard.fingerprint">IdentifyContaminant</a></td>
<td class="colLast">
<div class="block">Program to create a fingerprint for the <b>contaminating</b> sample when the level of contamination is both known and
 uniform in the genome.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/fingerprint/MatchResults.html" title="class in picard.fingerprint">MatchResults</a></td>
<td class="colLast">
<div class="block">Represents the results of a fingerprint comparison between one dataset and a specific
 fingerprint file.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/fingerprint/Snp.html" title="class in picard.fingerprint">Snp</a></td>
<td class="colLast">
<div class="block">Class to represent a SNP in context of a haplotype block that is used in fingerprinting.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<table class="typeSummary" border="0" cellpadding="3" cellspacing="0" summary="Enum Summary table, listing enums, and an explanation">
<caption><span>Enum Summary</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Enum</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/fingerprint/CrosscheckMetric.DataType.html" title="enum in picard.fingerprint">CrosscheckMetric.DataType</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/fingerprint/CrosscheckMetric.FingerprintResult.html" title="enum in picard.fingerprint">CrosscheckMetric.FingerprintResult</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/fingerprint/DiploidGenotype.html" title="enum in picard.fingerprint">DiploidGenotype</a></td>
<td class="colLast">
<div class="block">A genotype produced by one of the concrete implementations of AbstractAlleleCaller.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="../../picard/fingerprint/DiploidHaplotype.html" title="enum in picard.fingerprint">DiploidHaplotype</a></td>
<td class="colLast">
<div class="block">Simple enum to represent the three possible combinations of major/major, major/minor
 and minor/minor haplotypes for a diploid individual.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="../../picard/fingerprint/HaplotypeProbabilities.Genotype.html" title="enum in picard.fingerprint">HaplotypeProbabilities.Genotype</a></td>
<td class="colLast">
<div class="block">Log10(P(evidence| haplotype)) for the 3 different possible haplotypes
 {aa, ab, bb}</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li class="navBarCell1Rev">Package</li>
<li>Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/filter/package-summary.html">Prev&nbsp;Package</a></li>
<li><a href="../../picard/illumina/package-summary.html">Next&nbsp;Package</a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/fingerprint/package-summary.html" target="_top">Frames</a></li>
<li><a href="package-summary.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
