<?php

    include '../../../common/include/common.php';
    include_once '../../config.php';
    $module = modules::GATK;
    printHeader($module, "Picard | Tool Documentation Index", "Guide");
?>

<link type='text/css' rel='stylesheet' href='picardDoc.css'>

<div class='row-fluid' id="top">

	

	<?php $group = 'Read Data Manipulation'; ?>

	<section class="span4">
		<aside class="well">
			<a href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Docs Index</h4></a>
		</aside>
		<aside class="well">
			<h2>Categories</h2>
        <style>
            #sidenav .accordion-body a {
                color : gray;
            }

            .accordion-body li {
                list-style : none;
            }
        </style>
        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#BaseCalling">Base Calling</a>
							<div id="BaseCalling"
								<?php echo ($group == 'Base Calling')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_illumina_CheckIlluminaDirectory.html">CheckIlluminaDirectory</a>
											</li>
											<li>
												<a href="picard_illumina_CollectIlluminaBasecallingMetrics.html">CollectIlluminaBasecallingMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_CollectIlluminaLaneMetrics.html">CollectIlluminaLaneMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_ExtractIlluminaBarcodes.html">ExtractIlluminaBarcodes</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallsToFastq.html">IlluminaBasecallsToFastq</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallsToSam.html">IlluminaBasecallsToSam</a>
											</li>
											<li>
												<a href="picard_illumina_MarkIlluminaAdapters.html">MarkIlluminaAdapters</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControl">Diagnostics and Quality Control</a>
							<div id="DiagnosticsandQualityControl"
								<?php echo ($group == 'Diagnostics and Quality Control')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_AccumulateVariantCallingMetrics.html">AccumulateVariantCallingMetrics</a>
											</li>
											<li>
												<a href="picard_sam_BamIndexStats.html">BamIndexStats</a>
											</li>
											<li>
												<a href="picard_fingerprint_CalculateFingerprintMetrics.html">CalculateFingerprintMetrics</a>
											</li>
											<li>
												<a href="picard_sam_CalculateReadGroupChecksum.html">CalculateReadGroupChecksum</a>
											</li>
											<li>
												<a href="picard_fingerprint_CheckFingerprint.html">CheckFingerprint</a>
											</li>
											<li>
												<a href="picard_sam_CheckTerminatorBlock.html">CheckTerminatorBlock</a>
											</li>
											<li>
												<a href="picard_fingerprint_ClusterCrosscheckMetrics.html">ClusterCrosscheckMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectAlignmentSummaryMetrics.html">CollectAlignmentSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectBaseDistributionByCycle.html">CollectBaseDistributionByCycle</a>
											</li>
											<li>
												<a href="picard_analysis_CollectGcBiasMetrics.html">CollectGcBiasMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_quality_CollectHiSeqXPfFailMetrics.html">CollectHiSeqXPfFailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_CollectHsMetrics.html">CollectHsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_replicates_CollectIndependentReplicateMetrics.html">CollectIndependentReplicateMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectInsertSizeMetrics.html">CollectInsertSizeMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectJumpingLibraryMetrics.html">CollectJumpingLibraryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectMultipleMetrics.html">CollectMultipleMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectOxoGMetrics.html">CollectOxoGMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectQualityYieldMetrics.html">CollectQualityYieldMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRawWgsMetrics.html">CollectRawWgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRnaSeqMetrics.html">CollectRnaSeqMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRrbsMetrics.html">CollectRrbsMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_CollectSamErrorMetrics.html">CollectSamErrorMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_CollectSequencingArtifactMetrics.html">CollectSequencingArtifactMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_CollectTargetedPcrMetrics.html">CollectTargetedPcrMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_CollectVariantCallingMetrics.html">CollectVariantCallingMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetrics.html">CollectWgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetricsWithNonZeroCoverage.html">CollectWgsMetricsWithNonZeroCoverage</a>
											</li>
											<li>
												<a href="picard_analysis_CompareMetrics.html">CompareMetrics</a>
											</li>
											<li>
												<a href="picard_sam_CompareSAMs.html">CompareSAMs</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_ConvertSequencingArtifactToOxoG.html">ConvertSequencingArtifactToOxoG</a>
											</li>
											<li>
												<a href="picard_fingerprint_CrosscheckFingerprints.html">CrosscheckFingerprints</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_EstimateLibraryComplexity.html">EstimateLibraryComplexity</a>
											</li>
											<li>
												<a href="picard_analysis_MeanQualityByCycle.html">MeanQualityByCycle</a>
											</li>
											<li>
												<a href="picard_analysis_QualityScoreDistribution.html">QualityScoreDistribution</a>
											</li>
											<li>
												<a href="picard_sam_ValidateSamFile.html">ValidateSamFile</a>
											</li>
											<li>
												<a href="picard_sam_ViewSam.html">ViewSam</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#GenotypingArraysManipulation">Genotyping Arrays Manipulation</a>
							<div id="GenotypingArraysManipulation"
								<?php echo ($group == 'Genotyping Arrays Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_arrays_CreateVerifyIDIntensityContaminationMetricsFile.html">CreateVerifyIDIntensityContaminationMetricsFile</a>
											</li>
											<li>
												<a href="picard_arrays_GtcToVcf.html">GtcToVcf</a>
											</li>
											<li>
												<a href="picard_arrays_VcfToAdpc.html">VcfToAdpc</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#IntervalsManipulation">Intervals Manipulation</a>
							<div id="IntervalsManipulation"
								<?php echo ($group == 'Intervals Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_BedToIntervalList.html">BedToIntervalList</a>
											</li>
											<li>
												<a href="picard_util_IntervalListToBed.html">IntervalListToBed</a>
											</li>
											<li>
												<a href="picard_util_IntervalListTools.html">IntervalListTools</a>
											</li>
											<li>
												<a href="picard_util_LiftOverIntervalList.html">LiftOverIntervalList</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Other">Other</a>
							<div id="Other"
								<?php echo ($group == 'Other')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_FifoBuffer.html">FifoBuffer</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadDataManipulation">Read Data Manipulation</a>
							<div id="ReadDataManipulation"
								<?php echo ($group == 'Read Data Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_sam_AddCommentsToBam.html">AddCommentsToBam</a>
											</li>
											<li>
												<a href="picard_sam_AddOATag.html">AddOATag</a>
											</li>
											<li>
												<a href="picard_sam_AddOrReplaceReadGroups.html">AddOrReplaceReadGroups</a>
											</li>
											<li>
												<a href="picard_fastq_BamToBfq.html">BamToBfq</a>
											</li>
											<li>
												<a href="picard_sam_BuildBamIndex.html">BuildBamIndex</a>
											</li>
											<li>
												<a href="picard_sam_CleanSam.html">CleanSam</a>
											</li>
											<li>
												<a href="picard_sam_DownsampleSam.html">DownsampleSam</a>
											</li>
											<li>
												<a href="picard_sam_FastqToSam.html">FastqToSam</a>
											</li>
											<li>
												<a href="picard_sam_FilterSamReads.html">FilterSamReads</a>
											</li>
											<li>
												<a href="picard_sam_FixMateInformation.html">FixMateInformation</a>
											</li>
											<li>
												<a href="picard_sam_GatherBamFiles.html">GatherBamFiles</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_MarkDuplicates.html">MarkDuplicates</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_MarkDuplicatesWithMateCigar.html">MarkDuplicatesWithMateCigar</a>
											</li>
											<li>
												<a href="picard_sam_MergeBamAlignment.html">MergeBamAlignment</a>
											</li>
											<li>
												<a href="picard_sam_MergeSamFiles.html">MergeSamFiles</a>
											</li>
											<li>
												<a href="picard_sam_PositionBasedDownsampleSam.html">PositionBasedDownsampleSam</a>
											</li>
											<li>
												<a href="picard_sam_ReorderSam.html">ReorderSam</a>
											</li>
											<li>
												<a href="picard_sam_ReplaceSamHeader.html">ReplaceSamHeader</a>
											</li>
											<li>
												<a href="picard_sam_RevertOriginalBaseQualitiesAndAddMateCigar.html">RevertOriginalBaseQualitiesAndAddMateCigar</a>
											</li>
											<li>
												<a href="picard_sam_RevertSam.html">RevertSam</a>
											</li>
											<li>
												<a href="picard_sam_SamFormatConverter.html">SamFormatConverter</a>
											</li>
											<li>
												<a href="picard_sam_SamToFastq.html">SamToFastq</a>
											</li>
											<li>
												<a href="picard_sam_SetNmAndUqTags.html">SetNmAndUqTags</a>
											</li>
											<li>
												<a href="picard_sam_SetNmMdAndUqTags.html">SetNmMdAndUqTags</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_SimpleMarkDuplicatesWithMateCigar.html">SimpleMarkDuplicatesWithMateCigar</a>
											</li>
											<li>
												<a href="picard_sam_SortSam.html">SortSam</a>
											</li>
											<li>
												<a href="picard_sam_SplitSamByLibrary.html">SplitSamByLibrary</a>
											</li>
											<li>
												<a href="picard_sam_SplitSamByNumberOfReads.html">SplitSamByNumberOfReads</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_UmiAwareMarkDuplicatesWithMateCigar.html">UmiAwareMarkDuplicatesWithMateCigar</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Reference">Reference</a>
							<div id="Reference"
								<?php echo ($group == 'Reference')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_BaitDesigner.html">BaitDesigner</a>
											</li>
											<li>
												<a href="picard_sam_CreateSequenceDictionary.html">CreateSequenceDictionary</a>
											</li>
											<li>
												<a href="picard_reference_ExtractSequences.html">ExtractSequences</a>
											</li>
											<li>
												<a href="picard_reference_NonNFastaSize.html">NonNFastaSize</a>
											</li>
											<li>
												<a href="picard_reference_NormalizeFasta.html">NormalizeFasta</a>
											</li>
											<li>
												<a href="picard_util_ScatterIntervalsByNs.html">ScatterIntervalsByNs</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationandRefinement">Variant Evaluation and Refinement</a>
							<div id="VariantEvaluationandRefinement"
								<?php echo ($group == 'Variant Evaluation and Refinement')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_MendelianViolations_FindMendelianViolations.html">FindMendelianViolations</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordance.html">GenotypeConcordance</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantFiltering">Variant Filtering</a>
							<div id="VariantFiltering"
								<?php echo ($group == 'Variant Filtering')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_filter_FilterVcf.html">FilterVcf</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantManipulation">Variant Manipulation</a>
							<div id="VariantManipulation"
								<?php echo ($group == 'Variant Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_FixVcfHeader.html">FixVcfHeader</a>
											</li>
											<li>
												<a href="picard_vcf_GatherVcfs.html">GatherVcfs</a>
											</li>
											<li>
												<a href="picard_vcf_LiftoverVcf.html">LiftoverVcf</a>
											</li>
											<li>
												<a href="picard_vcf_MakeSitesOnlyVcf.html">MakeSitesOnlyVcf</a>
											</li>
											<li>
												<a href="picard_vcf_MergeVcfs.html">MergeVcfs</a>
											</li>
											<li>
												<a href="picard_vcf_RenameSampleInVcf.html">RenameSampleInVcf</a>
											</li>
											<li>
												<a href="picard_vcf_SortVcf.html">SortVcf</a>
											</li>
											<li>
												<a href="picard_vcf_SplitVcfs.html">SplitVcfs</a>
											</li>
											<li>
												<a href="picard_vcf_UpdateVcfSequenceDictionary.html">UpdateVcfSequenceDictionary</a>
											</li>
											<li>
												<a href="picard_vcf_VcfFormatConverter.html">VcfFormatConverter</a>
											</li>
											<li>
												<a href="picard_vcf_VcfToIntervalList.html">VcfToIntervalList</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
        		<hr>
        </ul>
		</aside>
		<?php getForumPosts( 'PositionBasedDownsampleSam' ) ?>

	</section>

	<div class="span8">

		<h1>PositionBasedDownsampleSam</h1>

		<p class="lead">Downsample a SAM or BAM file to retain a subset of the reads based on the reads location in each tile in the flowcell.</p>

			<h3>Category
				<small> Read Data Manipulation</small>
			</h3>
		<hr>
		<h2>Overview</h2>
		<h3>Summary</h3>
 Class to downsample a SAM/BAM file based on the position of the read in a flowcell. As with DownsampleSam, all the
 reads with the same queryname are either kept or dropped as a unit.
 <br/>

 <h3>Details</h3>
 The downsampling is <b>not</b> random (and there is no random seed). It is deterministically determined by the position
 of each read within its tile. Specifically, it draws an ellipse that covers a #FRACTION of the total tile's
 area and of all the edges of the tile. It uses this area to determine whether to keep or drop the record. Since reads
 with the same name have the same position (mates, secondary and supplemental alignments), the decision will be the
 same for all of them. The main concern of this downsampling method is that due to "optical duplicates" downsampling
 randomly can create a result that has a different optical duplicate rate, and therefore a different estimated library
 size (when running MarkDuplicates). This method keeps (physically) close read together, so that (except
 for reads near the boundary of the circle) optical duplicates are kept or dropped as a group.

 By default the program expects the read names to have 5 or 7 fields separated by colons (:), and it takes the last two
 to indicate the x and y coordinates of the reads within the tile whence it was sequenced. See
 ReadNameParser#DEFAULT_READ_NAME_REGEX for more detail. The program traverses the #INPUT twice: first
 to find out the size of each of the tiles, and next to perform the downsampling.

 Downsampling invalidates the duplicate flag because duplicate reads before downsampling may not all remain duplicated
 after downsampling. Thus, the default setting also removes the duplicate information.

 <h3>Example</h3>
 <pre>
 java -jar picard.jar PositionBasedDownsampleSam \
       I=input.bam \
       O=downsampled.bam \
       FRACTION=0.1
 </pre>
 <h3>Caveats</h3>
 <ol>
 <li>
 This method is <b>technology and read-name dependent</b>. If the read-names do not have coordinate information
 embedded in them, or if your BAM contains reads from multiple technologies (flowcell versions, sequencing machines).
 this will not work properly. It has been designed to work with Illumina technology and reads-names. Consider
 modifying #READ_NAME_REGEX in other cases.
 </li>
 <li>
 The code has been designed to simulate, as accurately as possible, sequencing less, <b>not</b> for getting an exact
 downsampled fraction (Use DownsampleSam for that.) In particular, since the reads may be distributed non-evenly
 within the lanes/tiles, the resulting downsampling percentage will not be accurately determined by the input argument
 #FRACTION.
 </li>
 <li>
 Consider running MarkDuplicates after downsampling in order to "expose" the duplicates whose representative has
 been downsampled away.
 </li>
 <li>
 The downsampling assumes a uniform distribution of reads in the flowcell.
 Input already downsampled with PositionBasedDownsampleSam violates this assumption.
 To guard against such input,
 PositionBasedDownsampleSam always places a PG record in the header of its output,
 and aborts whenever it finds such a PG record in its input.
 </li>
 </ol>


				<h3>PositionBasedDownsampleSam specific arguments</h3>
				<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
				<table class="table table-striped table-bordered table-condensed">
					<thead>
					<tr>
						<th>Argument name(s)</th>
						<th>Default value</th>
						<th>Summary</th>
					</tr>
					</thead>
					<tbody>
			<tr>
				<th colspan="4" id="row-divider">Required Arguments</th>
			</tr>
				<tr>
					<td><a href="#--FRACTION">--FRACTION</a><br />
								&nbsp;<em>-F</em>
					</td>
					<!--<td>Double</td> -->
					<td>null</td>
					<td>The (approximate) fraction of reads to be kept, between 0 and 1.</td>
				</tr>
				<tr>
					<td><a href="#--INPUT">--INPUT</a><br />
								&nbsp;<em>-I</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>The input SAM or BAM file to downsample.</td>
				</tr>
				<tr>
					<td><a href="#--OUTPUT">--OUTPUT</a><br />
								&nbsp;<em>-O</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>The output, downsampled, SAM or BAM file.</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Tool Arguments</th>
			</tr>
				<tr>
					<td><a href="#--ALLOW_MULTIPLE_DOWNSAMPLING_DESPITE_WARNINGS">--ALLOW_MULTIPLE_DOWNSAMPLING_DESPITE_WARNINGS</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Allow downsampling again despite this being a bad idea with possibly unexpected results.</td>
				</tr>
				<tr>
					<td><a href="#--READ_NAME_REGEX">--READ_NAME_REGEX</a><br />
					</td>
					<!--<td>String</td> -->
					<td><optimized capture of last three ':' separated fields as numeric values></td>
					<td>Use these regular expressions to parse read names in the input SAM file. Read names are parsed to extract three variables: tile/region, x coordinate and y coordinate. The x and y coordinates are used to determine the downsample decision. Set this option to null to disable optical duplicate detection, e.g. for RNA-seq The regular expression should contain three capture groups for the three variables, in order. It must match the entire read name. Note that if the default regex is specified, a regex match is not actually done, but instead the read name is split on colons (:). For 5 element names, the 3rd, 4th and 5th elements are assumed to be tile, x and y values. For 7 element names (CASAVA 1.8), the 5th, 6th, and 7th elements are assumed to be tile, x and y values.</td>
				</tr>
				<tr>
					<td><a href="#--REMOVE_DUPLICATE_INFORMATION">--REMOVE_DUPLICATE_INFORMATION</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>true</td>
					<td>Determines whether the duplicate tag should be reset since the downsampling requires re-marking duplicates.</td>
				</tr>
				<tr>
					<td><a href="#--STOP_AFTER">--STOP_AFTER</a><br />
					</td>
					<!--<td>Long</td> -->
					<td>null</td>
					<td>Stop after processing N reads, mainly for debugging.</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Common Arguments</th>
			</tr>
				<tr>
					<td><a href="#--COMPRESSION_LEVEL">--COMPRESSION_LEVEL</a><br />
					</td>
					<!--<td>int</td> -->
					<td>5</td>
					<td>Compression level for all compressed files created (e.g. BAM and VCF).</td>
				</tr>
				<tr>
					<td><a href="#--CREATE_INDEX">--CREATE_INDEX</a><br />
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Whether to create a BAM index when writing a coordinate-sorted BAM file.</td>
				</tr>
				<tr>
					<td><a href="#--CREATE_MD5_FILE">--CREATE_MD5_FILE</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Whether to create an MD5 digest for any BAM or FASTQ files created.  </td>
				</tr>
				<tr>
					<td><a href="#--GA4GH_CLIENT_SECRETS">--GA4GH_CLIENT_SECRETS</a><br />
					</td>
					<!--<td>String</td> -->
					<td>client_secrets.json</td>
					<td>Google Genomics API client_secrets.json file path.</td>
				</tr>
				<tr>
					<td><a href="#--MAX_RECORDS_IN_RAM">--MAX_RECORDS_IN_RAM</a><br />
					</td>
					<!--<td>Integer</td> -->
					<td>500000</td>
					<td>When writing files that need to be sorted, this will specify the number of records stored in RAM before spilling to disk. Increasing this number reduces the number of file handles needed to sort the file, and increases the amount of RAM needed.</td>
				</tr>
				<tr>
					<td><a href="#--QUIET">--QUIET</a><br />
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Whether to suppress job-summary info on System.err.</td>
				</tr>
				<tr>
					<td><a href="#--REFERENCE_SEQUENCE">--REFERENCE_SEQUENCE</a><br />
								&nbsp;<em>-R</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>Reference sequence file.</td>
				</tr>
				<tr>
					<td><a href="#--TMP_DIR">--TMP_DIR</a><br />
					</td>
					<!--<td>List[File]</td> -->
					<td>[]</td>
					<td>One or more directories with space available to be used by this program for temporary storage of working files</td>
				</tr>
				<tr>
					<td><a href="#--USE_JDK_DEFLATER">--USE_JDK_DEFLATER</a><br />
								&nbsp;<em>-use_jdk_deflater</em>
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Use the JDK Deflater instead of the Intel Deflater for writing compressed output</td>
				</tr>
				<tr>
					<td><a href="#--USE_JDK_INFLATER">--USE_JDK_INFLATER</a><br />
								&nbsp;<em>-use_jdk_inflater</em>
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Use the JDK Inflater instead of the Intel Inflater for reading compressed input</td>
				</tr>
				<tr>
					<td><a href="#--VALIDATION_STRINGENCY">--VALIDATION_STRINGENCY</a><br />
					</td>
					<!--<td>ValidationStringency</td> -->
					<td>STRICT</td>
					<td>Validation stringency for all SAM files read by this program.  Setting stringency to SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.</td>
				</tr>
				<tr>
					<td><a href="#--VERBOSITY">--VERBOSITY</a><br />
					</td>
					<!--<td>LogLevel</td> -->
					<td>INFO</td>
					<td>Control verbosity of logging.</td>
				</tr>
					</tbody>
				</table>

					<h3>Argument details</h3>
					<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--ALLOW_MULTIPLE_DOWNSAMPLING_DESPITE_WARNINGS">--ALLOW_MULTIPLE_DOWNSAMPLING_DESPITE_WARNINGS </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Allow downsampling again despite this being a bad idea with possibly unexpected results.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--COMPRESSION_LEVEL">--COMPRESSION_LEVEL </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Compression level for all compressed files created (e.g. BAM and VCF).</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">5</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--CREATE_INDEX">--CREATE_INDEX </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to create a BAM index when writing a coordinate-sorted BAM file.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--CREATE_MD5_FILE">--CREATE_MD5_FILE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to create an MD5 digest for any BAM or FASTQ files created.  </b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--FRACTION">--FRACTION </a>
			 / <small>-F</small>
		</h3>
		<p class="args">
			<b>The (approximate) fraction of reads to be kept, between 0 and 1.</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">Double</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--GA4GH_CLIENT_SECRETS">--GA4GH_CLIENT_SECRETS </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Google Genomics API client_secrets.json file path.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">client_secrets.json</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--INPUT">--INPUT </a>
			 / <small>-I</small>
		</h3>
		<p class="args">
			<b>The input SAM or BAM file to downsample.</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MAX_RECORDS_IN_RAM">--MAX_RECORDS_IN_RAM </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>When writing files that need to be sorted, this will specify the number of records stored in RAM before spilling to disk. Increasing this number reduces the number of file handles needed to sort the file, and increases the amount of RAM needed.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Integer</span>
			&nbsp;<span class="label">500000</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--OUTPUT">--OUTPUT </a>
			 / <small>-O</small>
		</h3>
		<p class="args">
			<b>The output, downsampled, SAM or BAM file.</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--QUIET">--QUIET </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to suppress job-summary info on System.err.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--READ_NAME_REGEX">--READ_NAME_REGEX </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Use these regular expressions to parse read names in the input SAM file. Read names are parsed to extract three variables: tile/region, x coordinate and y coordinate. The x and y coordinates are used to determine the downsample decision. Set this option to null to disable optical duplicate detection, e.g. for RNA-seq The regular expression should contain three capture groups for the three variables, in order. It must match the entire read name. Note that if the default regex is specified, a regex match is not actually done, but instead the read name is split on colons (:). For 5 element names, the 3rd, 4th and 5th elements are assumed to be tile, x and y values. For 7 element names (CASAVA 1.8), the 5th, 6th, and 7th elements are assumed to be tile, x and y values.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label"><optimized capture of last three ':' separated fields as numeric values></span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--REFERENCE_SEQUENCE">--REFERENCE_SEQUENCE </a>
			 / <small>-R</small>
		</h3>
		<p class="args">
			<b>Reference sequence file.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--REMOVE_DUPLICATE_INFORMATION">--REMOVE_DUPLICATE_INFORMATION </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Determines whether the duplicate tag should be reset since the downsampling requires re-marking duplicates.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">true</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--STOP_AFTER">--STOP_AFTER </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Stop after processing N reads, mainly for debugging.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Long</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--TMP_DIR">--TMP_DIR </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>One or more directories with space available to be used by this program for temporary storage of working files</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[File]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--USE_JDK_DEFLATER">--USE_JDK_DEFLATER </a>
			 / <small>-use_jdk_deflater</small>
		</h3>
		<p class="args">
			<b>Use the JDK Deflater instead of the Intel Deflater for writing compressed output</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--USE_JDK_INFLATER">--USE_JDK_INFLATER </a>
			 / <small>-use_jdk_inflater</small>
		</h3>
		<p class="args">
			<b>Use the JDK Inflater instead of the Intel Inflater for reading compressed input</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--VALIDATION_STRINGENCY">--VALIDATION_STRINGENCY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Validation stringency for all SAM files read by this program.  Setting stringency to SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.</b><br />
			
		</p>
			<p>
				The --VALIDATION_STRINGENCY argument is an enumerated type (ValidationStringency), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">STRICT</dt>
					<dd class="enum"></dd>
					<dt class="enum">LENIENT</dt>
					<dd class="enum"></dd>
					<dt class="enum">SILENT</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">ValidationStringency</span>
			&nbsp;<span class="label">STRICT</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--VERBOSITY">--VERBOSITY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Control verbosity of logging.</b><br />
			
		</p>
			<p>
				The --VERBOSITY argument is an enumerated type (LogLevel), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">ERROR</dt>
					<dd class="enum"></dd>
					<dt class="enum">WARNING</dt>
					<dd class="enum"></dd>
					<dt class="enum">INFO</dt>
					<dd class="enum"></dd>
					<dt class="enum">DEBUG</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">LogLevel</span>
			&nbsp;<span class="label">INFO</span>
		</p>

        <hr>
        <p><a href='#top'><i class='fa fa-chevron-up'></i> Return to top</a></p>
        <hr>
        <p class="see-also">See also 
        	<a href="https://www.broadinstitute.org/picard/guide/">Picard Documentation Index</a> |
        	<a href="index.html">Tool Docs Index</a> |
        	<a href="http://gatkforums.broadinstitute.org/">Support Forum</a>
        </p>

        <p class="version">Picard version 2.21.1-SNAPSHOT built at 04-35-2019 10:35:30.
        </p>

	</div>

	<?php printFooter($module); ?>