<?php

    include '../../../common/include/common.php';
    include_once '../../config.php';
    $module = modules::GATK;
    printHeader($module, "Picard | Tool Documentation Index", "Guide");
?>

<link type='text/css' rel='stylesheet' href='picardDoc.css'>

<div class='row-fluid' id="top">

	

	<?php $group = 'Base Calling'; ?>

	<section class="span4">
		<aside class="well">
			<a href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Docs Index</h4></a>
		</aside>
		<aside class="well">
			<h2>Categories</h2>
        <style>
            #sidenav .accordion-body a {
                color : gray;
            }

            .accordion-body li {
                list-style : none;
            }
        </style>
        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#BaseCalling">Base Calling</a>
							<div id="BaseCalling"
								<?php echo ($group == 'Base Calling')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_illumina_CheckIlluminaDirectory.html">CheckIlluminaDirectory</a>
											</li>
											<li>
												<a href="picard_illumina_CollectIlluminaBasecallingMetrics.html">CollectIlluminaBasecallingMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_CollectIlluminaLaneMetrics.html">CollectIlluminaLaneMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_ExtractIlluminaBarcodes.html">ExtractIlluminaBarcodes</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallsToFastq.html">IlluminaBasecallsToFastq</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallsToSam.html">IlluminaBasecallsToSam</a>
											</li>
											<li>
												<a href="picard_illumina_MarkIlluminaAdapters.html">MarkIlluminaAdapters</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControl">Diagnostics and Quality Control</a>
							<div id="DiagnosticsandQualityControl"
								<?php echo ($group == 'Diagnostics and Quality Control')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_AccumulateVariantCallingMetrics.html">AccumulateVariantCallingMetrics</a>
											</li>
											<li>
												<a href="picard_sam_BamIndexStats.html">BamIndexStats</a>
											</li>
											<li>
												<a href="picard_fingerprint_CalculateFingerprintMetrics.html">CalculateFingerprintMetrics</a>
											</li>
											<li>
												<a href="picard_sam_CalculateReadGroupChecksum.html">CalculateReadGroupChecksum</a>
											</li>
											<li>
												<a href="picard_fingerprint_CheckFingerprint.html">CheckFingerprint</a>
											</li>
											<li>
												<a href="picard_sam_CheckTerminatorBlock.html">CheckTerminatorBlock</a>
											</li>
											<li>
												<a href="picard_fingerprint_ClusterCrosscheckMetrics.html">ClusterCrosscheckMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectAlignmentSummaryMetrics.html">CollectAlignmentSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectBaseDistributionByCycle.html">CollectBaseDistributionByCycle</a>
											</li>
											<li>
												<a href="picard_analysis_CollectGcBiasMetrics.html">CollectGcBiasMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_quality_CollectHiSeqXPfFailMetrics.html">CollectHiSeqXPfFailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_CollectHsMetrics.html">CollectHsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_replicates_CollectIndependentReplicateMetrics.html">CollectIndependentReplicateMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectInsertSizeMetrics.html">CollectInsertSizeMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectJumpingLibraryMetrics.html">CollectJumpingLibraryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectMultipleMetrics.html">CollectMultipleMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectOxoGMetrics.html">CollectOxoGMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectQualityYieldMetrics.html">CollectQualityYieldMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRawWgsMetrics.html">CollectRawWgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRnaSeqMetrics.html">CollectRnaSeqMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRrbsMetrics.html">CollectRrbsMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_CollectSamErrorMetrics.html">CollectSamErrorMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_CollectSequencingArtifactMetrics.html">CollectSequencingArtifactMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_CollectTargetedPcrMetrics.html">CollectTargetedPcrMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_CollectVariantCallingMetrics.html">CollectVariantCallingMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetrics.html">CollectWgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetricsWithNonZeroCoverage.html">CollectWgsMetricsWithNonZeroCoverage</a>
											</li>
											<li>
												<a href="picard_analysis_CompareMetrics.html">CompareMetrics</a>
											</li>
											<li>
												<a href="picard_sam_CompareSAMs.html">CompareSAMs</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_ConvertSequencingArtifactToOxoG.html">ConvertSequencingArtifactToOxoG</a>
											</li>
											<li>
												<a href="picard_fingerprint_CrosscheckFingerprints.html">CrosscheckFingerprints</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_EstimateLibraryComplexity.html">EstimateLibraryComplexity</a>
											</li>
											<li>
												<a href="picard_analysis_MeanQualityByCycle.html">MeanQualityByCycle</a>
											</li>
											<li>
												<a href="picard_analysis_QualityScoreDistribution.html">QualityScoreDistribution</a>
											</li>
											<li>
												<a href="picard_sam_ValidateSamFile.html">ValidateSamFile</a>
											</li>
											<li>
												<a href="picard_sam_ViewSam.html">ViewSam</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#GenotypingArraysManipulation">Genotyping Arrays Manipulation</a>
							<div id="GenotypingArraysManipulation"
								<?php echo ($group == 'Genotyping Arrays Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_arrays_CreateVerifyIDIntensityContaminationMetricsFile.html">CreateVerifyIDIntensityContaminationMetricsFile</a>
											</li>
											<li>
												<a href="picard_arrays_GtcToVcf.html">GtcToVcf</a>
											</li>
											<li>
												<a href="picard_arrays_VcfToAdpc.html">VcfToAdpc</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#IntervalsManipulation">Intervals Manipulation</a>
							<div id="IntervalsManipulation"
								<?php echo ($group == 'Intervals Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_BedToIntervalList.html">BedToIntervalList</a>
											</li>
											<li>
												<a href="picard_util_IntervalListToBed.html">IntervalListToBed</a>
											</li>
											<li>
												<a href="picard_util_IntervalListTools.html">IntervalListTools</a>
											</li>
											<li>
												<a href="picard_util_LiftOverIntervalList.html">LiftOverIntervalList</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Other">Other</a>
							<div id="Other"
								<?php echo ($group == 'Other')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_FifoBuffer.html">FifoBuffer</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadDataManipulation">Read Data Manipulation</a>
							<div id="ReadDataManipulation"
								<?php echo ($group == 'Read Data Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_sam_AddCommentsToBam.html">AddCommentsToBam</a>
											</li>
											<li>
												<a href="picard_sam_AddOATag.html">AddOATag</a>
											</li>
											<li>
												<a href="picard_sam_AddOrReplaceReadGroups.html">AddOrReplaceReadGroups</a>
											</li>
											<li>
												<a href="picard_fastq_BamToBfq.html">BamToBfq</a>
											</li>
											<li>
												<a href="picard_sam_BuildBamIndex.html">BuildBamIndex</a>
											</li>
											<li>
												<a href="picard_sam_CleanSam.html">CleanSam</a>
											</li>
											<li>
												<a href="picard_sam_DownsampleSam.html">DownsampleSam</a>
											</li>
											<li>
												<a href="picard_sam_FastqToSam.html">FastqToSam</a>
											</li>
											<li>
												<a href="picard_sam_FilterSamReads.html">FilterSamReads</a>
											</li>
											<li>
												<a href="picard_sam_FixMateInformation.html">FixMateInformation</a>
											</li>
											<li>
												<a href="picard_sam_GatherBamFiles.html">GatherBamFiles</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_MarkDuplicates.html">MarkDuplicates</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_MarkDuplicatesWithMateCigar.html">MarkDuplicatesWithMateCigar</a>
											</li>
											<li>
												<a href="picard_sam_MergeBamAlignment.html">MergeBamAlignment</a>
											</li>
											<li>
												<a href="picard_sam_MergeSamFiles.html">MergeSamFiles</a>
											</li>
											<li>
												<a href="picard_sam_PositionBasedDownsampleSam.html">PositionBasedDownsampleSam</a>
											</li>
											<li>
												<a href="picard_sam_ReorderSam.html">ReorderSam</a>
											</li>
											<li>
												<a href="picard_sam_ReplaceSamHeader.html">ReplaceSamHeader</a>
											</li>
											<li>
												<a href="picard_sam_RevertOriginalBaseQualitiesAndAddMateCigar.html">RevertOriginalBaseQualitiesAndAddMateCigar</a>
											</li>
											<li>
												<a href="picard_sam_RevertSam.html">RevertSam</a>
											</li>
											<li>
												<a href="picard_sam_SamFormatConverter.html">SamFormatConverter</a>
											</li>
											<li>
												<a href="picard_sam_SamToFastq.html">SamToFastq</a>
											</li>
											<li>
												<a href="picard_sam_SetNmAndUqTags.html">SetNmAndUqTags</a>
											</li>
											<li>
												<a href="picard_sam_SetNmMdAndUqTags.html">SetNmMdAndUqTags</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_SimpleMarkDuplicatesWithMateCigar.html">SimpleMarkDuplicatesWithMateCigar</a>
											</li>
											<li>
												<a href="picard_sam_SortSam.html">SortSam</a>
											</li>
											<li>
												<a href="picard_sam_SplitSamByLibrary.html">SplitSamByLibrary</a>
											</li>
											<li>
												<a href="picard_sam_SplitSamByNumberOfReads.html">SplitSamByNumberOfReads</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_UmiAwareMarkDuplicatesWithMateCigar.html">UmiAwareMarkDuplicatesWithMateCigar</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Reference">Reference</a>
							<div id="Reference"
								<?php echo ($group == 'Reference')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_BaitDesigner.html">BaitDesigner</a>
											</li>
											<li>
												<a href="picard_sam_CreateSequenceDictionary.html">CreateSequenceDictionary</a>
											</li>
											<li>
												<a href="picard_reference_ExtractSequences.html">ExtractSequences</a>
											</li>
											<li>
												<a href="picard_reference_NonNFastaSize.html">NonNFastaSize</a>
											</li>
											<li>
												<a href="picard_reference_NormalizeFasta.html">NormalizeFasta</a>
											</li>
											<li>
												<a href="picard_util_ScatterIntervalsByNs.html">ScatterIntervalsByNs</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationandRefinement">Variant Evaluation and Refinement</a>
							<div id="VariantEvaluationandRefinement"
								<?php echo ($group == 'Variant Evaluation and Refinement')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_MendelianViolations_FindMendelianViolations.html">FindMendelianViolations</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordance.html">GenotypeConcordance</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantFiltering">Variant Filtering</a>
							<div id="VariantFiltering"
								<?php echo ($group == 'Variant Filtering')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_filter_FilterVcf.html">FilterVcf</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantManipulation">Variant Manipulation</a>
							<div id="VariantManipulation"
								<?php echo ($group == 'Variant Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_FixVcfHeader.html">FixVcfHeader</a>
											</li>
											<li>
												<a href="picard_vcf_GatherVcfs.html">GatherVcfs</a>
											</li>
											<li>
												<a href="picard_vcf_LiftoverVcf.html">LiftoverVcf</a>
											</li>
											<li>
												<a href="picard_vcf_MakeSitesOnlyVcf.html">MakeSitesOnlyVcf</a>
											</li>
											<li>
												<a href="picard_vcf_MergeVcfs.html">MergeVcfs</a>
											</li>
											<li>
												<a href="picard_vcf_RenameSampleInVcf.html">RenameSampleInVcf</a>
											</li>
											<li>
												<a href="picard_vcf_SortVcf.html">SortVcf</a>
											</li>
											<li>
												<a href="picard_vcf_SplitVcfs.html">SplitVcfs</a>
											</li>
											<li>
												<a href="picard_vcf_UpdateVcfSequenceDictionary.html">UpdateVcfSequenceDictionary</a>
											</li>
											<li>
												<a href="picard_vcf_VcfFormatConverter.html">VcfFormatConverter</a>
											</li>
											<li>
												<a href="picard_vcf_VcfToIntervalList.html">VcfToIntervalList</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
        		<hr>
        </ul>
		</aside>
		<?php getForumPosts( 'ExtractIlluminaBarcodes' ) ?>

	</section>

	<div class="span8">

		<h1>ExtractIlluminaBarcodes</h1>

		<p class="lead">Tool determines the barcode for each read in an Illumina lane.  </p>

			<h3>Category
				<small> Base Calling</small>
			</h3>
		<hr>
		<h2>Overview</h2>
		Determine the barcode for each read in an Illumina lane.
 For each tile, a file is written to the basecalls directory of the form s_<lane>_<tile>_barcode.txt.
 An output file contains a line for each read in the tile, aligned with the regular basecall output
 The output file contains the following tab-separated columns:
 - read subsequence at barcode position
 - Y or N indicating if there was a barcode match
 - matched barcode sequence (empty if read did not match one of the barcodes).  If there is no match
 but we're close to the threshold of calling it a match we output the barcode that would have been
 matched but in lower case
 - distance to best matching barcode, "mismatches" (*)
 - distance to second-best matching barcode, "mismatchesToSecondBest" (*)

 NOTE (*): Due to an optimization the reported mismatches & mismatchesToSecondBest values may be inaccurate as long as
 the conclusion (match vs. no-match) isn't affected. For example, reported mismatches and
 mismatchesToSecondBest may be smaller than their true value if mismatches is truly larger than MAX_MISMATCHES.
 Also, mismatchesToSecondBest might be smaller than its true value if its true value is greater than
 mismatches + MIN_MISMATCH_DELTA.


				<h3>ExtractIlluminaBarcodes specific arguments</h3>
				<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
				<table class="table table-striped table-bordered table-condensed">
					<thead>
					<tr>
						<th>Argument name(s)</th>
						<th>Default value</th>
						<th>Summary</th>
					</tr>
					</thead>
					<tbody>
			<tr>
				<th colspan="4" id="row-divider">Required Arguments</th>
			</tr>
				<tr>
					<td><a href="#--BARCODE">--BARCODE</a><br />
					</td>
					<!--<td>List[String]</td> -->
					<td>[]</td>
					<td>Barcode sequence.  These must be unique, and all the same length.  This cannot be used with reads that have more than one barcode; use BARCODE_FILE in that case. </td>
				</tr>
				<tr>
					<td><a href="#--BARCODE_FILE">--BARCODE_FILE</a><br />
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>Tab-delimited file of barcode sequences, barcode name and, optionally, library name.  Barcodes must be unique and all the same length.  Column headers must be 'barcode_sequence' (or 'barcode_sequence_1'), 'barcode_sequence_2' (optional), 'barcode_name', and 'library_name'.</td>
				</tr>
				<tr>
					<td><a href="#--BASECALLS_DIR">--BASECALLS_DIR</a><br />
								&nbsp;<em>-B</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>The Illumina basecalls directory. </td>
				</tr>
				<tr>
					<td><a href="#--LANE">--LANE</a><br />
								&nbsp;<em>-L</em>
					</td>
					<!--<td>Integer</td> -->
					<td>null</td>
					<td>Lane number. </td>
				</tr>
				<tr>
					<td><a href="#--METRICS_FILE">--METRICS_FILE</a><br />
								&nbsp;<em>-M</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>Per-barcode and per-lane metrics written to this file.</td>
				</tr>
				<tr>
					<td><a href="#--READ_STRUCTURE">--READ_STRUCTURE</a><br />
								&nbsp;<em>-RS</em>
					</td>
					<!--<td>String</td> -->
					<td>null</td>
					<td>A description of the logical structure of clusters in an Illumina Run, i.e. a description of the structure IlluminaBasecallsToSam assumes the  data to be in. It should consist of integer/character pairs describing the number of cycles and the type of those cycles (B for Sample Barcode, M for molecular barcode, T for Template, and S for skip).  E.g. If the input data consists of 80 base clusters and we provide a read structure of "28T8M8B8S28T" then the sequence may be split up into four reads:
* read one with 28 cycles (bases) of template
* read two with 8 cycles (bases) of molecular barcode (ex. unique molecular barcode)
* read three with 8 cycles (bases) of sample barcode
* 8 cycles (bases) skipped.
* read four with 28 cycles (bases) of template
The skipped cycles would NOT be included in an output SAM/BAM file or in read groups therein.</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Tool Arguments</th>
			</tr>
				<tr>
					<td><a href="#--COMPRESS_OUTPUTS">--COMPRESS_OUTPUTS</a><br />
								&nbsp;<em>-GZIP</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Compress output s_l_t_barcode.txt files using gzip and append a .gz extension to the file names.</td>
				</tr>
				<tr>
					<td><a href="#--DISTANCE_MODE">--DISTANCE_MODE</a><br />
					</td>
					<!--<td>DistanceMetric</td> -->
					<td>HAMMING</td>
					<td>The distance metric that should be used to compare the barcode-reads and the provided barcodes for finding the best and second-best assignments.</td>
				</tr>
				<tr>
					<td><a href="#--MAX_MISMATCHES">--MAX_MISMATCHES</a><br />
					</td>
					<!--<td>int</td> -->
					<td>1</td>
					<td>Maximum mismatches for a barcode to be considered a match.</td>
				</tr>
				<tr>
					<td><a href="#--MAX_NO_CALLS">--MAX_NO_CALLS</a><br />
					</td>
					<!--<td>int</td> -->
					<td>2</td>
					<td>Maximum allowable number of no-calls in a barcode read before it is considered unmatchable.</td>
				</tr>
				<tr>
					<td><a href="#--MIN_MISMATCH_DELTA">--MIN_MISMATCH_DELTA</a><br />
					</td>
					<!--<td>int</td> -->
					<td>1</td>
					<td>Minimum difference between number of mismatches in the best and second best barcodes for a barcode to be considered a match.</td>
				</tr>
				<tr>
					<td><a href="#--MINIMUM_BASE_QUALITY">--MINIMUM_BASE_QUALITY</a><br />
								&nbsp;<em>-Q</em>
					</td>
					<!--<td>int</td> -->
					<td>0</td>
					<td>Minimum base quality. Any barcode bases falling below this quality will be considered a mismatch even if the bases match.</td>
				</tr>
				<tr>
					<td><a href="#--MINIMUM_QUALITY">--MINIMUM_QUALITY</a><br />
					</td>
					<!--<td>int</td> -->
					<td>2</td>
					<td>The minimum quality (after transforming 0s to 1s) expected from reads.  If qualities are lower than this value, an error is thrown.The default of 2 is what the Illumina's spec describes as the minimum, but in practice the value has been observed lower.</td>
				</tr>
				<tr>
					<td><a href="#--NUM_PROCESSORS">--NUM_PROCESSORS</a><br />
					</td>
					<!--<td>int</td> -->
					<td>1</td>
					<td>Run this many PerTileBarcodeExtractors in parallel.  If NUM_PROCESSORS = 0, number of cores is automatically set to the number of cores available on the machine. If NUM_PROCESSORS < 0 then the number of cores used will be the number available on the machine less NUM_PROCESSORS.</td>
				</tr>
				<tr>
					<td><a href="#--OUTPUT_DIR">--OUTPUT_DIR</a><br />
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>Where to write _barcode.txt files.  By default, these are written to BASECALLS_DIR.</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Common Arguments</th>
			</tr>
				<tr>
					<td><a href="#--COMPRESSION_LEVEL">--COMPRESSION_LEVEL</a><br />
					</td>
					<!--<td>int</td> -->
					<td>5</td>
					<td>Compression level for all compressed files created (e.g. BAM and VCF).</td>
				</tr>
				<tr>
					<td><a href="#--CREATE_INDEX">--CREATE_INDEX</a><br />
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Whether to create a BAM index when writing a coordinate-sorted BAM file.</td>
				</tr>
				<tr>
					<td><a href="#--CREATE_MD5_FILE">--CREATE_MD5_FILE</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Whether to create an MD5 digest for any BAM or FASTQ files created.  </td>
				</tr>
				<tr>
					<td><a href="#--GA4GH_CLIENT_SECRETS">--GA4GH_CLIENT_SECRETS</a><br />
					</td>
					<!--<td>String</td> -->
					<td>client_secrets.json</td>
					<td>Google Genomics API client_secrets.json file path.</td>
				</tr>
				<tr>
					<td><a href="#--MAX_RECORDS_IN_RAM">--MAX_RECORDS_IN_RAM</a><br />
					</td>
					<!--<td>Integer</td> -->
					<td>500000</td>
					<td>When writing files that need to be sorted, this will specify the number of records stored in RAM before spilling to disk. Increasing this number reduces the number of file handles needed to sort the file, and increases the amount of RAM needed.</td>
				</tr>
				<tr>
					<td><a href="#--QUIET">--QUIET</a><br />
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Whether to suppress job-summary info on System.err.</td>
				</tr>
				<tr>
					<td><a href="#--REFERENCE_SEQUENCE">--REFERENCE_SEQUENCE</a><br />
								&nbsp;<em>-R</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>Reference sequence file.</td>
				</tr>
				<tr>
					<td><a href="#--TMP_DIR">--TMP_DIR</a><br />
					</td>
					<!--<td>List[File]</td> -->
					<td>[]</td>
					<td>One or more directories with space available to be used by this program for temporary storage of working files</td>
				</tr>
				<tr>
					<td><a href="#--USE_JDK_DEFLATER">--USE_JDK_DEFLATER</a><br />
								&nbsp;<em>-use_jdk_deflater</em>
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Use the JDK Deflater instead of the Intel Deflater for writing compressed output</td>
				</tr>
				<tr>
					<td><a href="#--USE_JDK_INFLATER">--USE_JDK_INFLATER</a><br />
								&nbsp;<em>-use_jdk_inflater</em>
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Use the JDK Inflater instead of the Intel Inflater for reading compressed input</td>
				</tr>
				<tr>
					<td><a href="#--VALIDATION_STRINGENCY">--VALIDATION_STRINGENCY</a><br />
					</td>
					<!--<td>ValidationStringency</td> -->
					<td>STRICT</td>
					<td>Validation stringency for all SAM files read by this program.  Setting stringency to SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.</td>
				</tr>
				<tr>
					<td><a href="#--VERBOSITY">--VERBOSITY</a><br />
					</td>
					<!--<td>LogLevel</td> -->
					<td>INFO</td>
					<td>Control verbosity of logging.</td>
				</tr>
					</tbody>
				</table>

					<h3>Argument details</h3>
					<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--BARCODE">--BARCODE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Barcode sequence.  These must be unique, and all the same length.  This cannot be used with reads that have more than one barcode; use BARCODE_FILE in that case. </b><br />
			
		</p>
			<p><b>Exclusion:</b> This argument cannot be used at the same time as <code>BARCODE_FILE</code>.</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">List[String]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--BARCODE_FILE">--BARCODE_FILE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Tab-delimited file of barcode sequences, barcode name and, optionally, library name.  Barcodes must be unique and all the same length.  Column headers must be 'barcode_sequence' (or 'barcode_sequence_1'), 'barcode_sequence_2' (optional), 'barcode_name', and 'library_name'.</b><br />
			
		</p>
			<p><b>Exclusion:</b> This argument cannot be used at the same time as <code>BARCODE</code>.</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--BASECALLS_DIR">--BASECALLS_DIR </a>
			 / <small>-B</small>
		</h3>
		<p class="args">
			<b>The Illumina basecalls directory. </b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--COMPRESS_OUTPUTS">--COMPRESS_OUTPUTS </a>
			 / <small>-GZIP</small>
		</h3>
		<p class="args">
			<b>Compress output s_l_t_barcode.txt files using gzip and append a .gz extension to the file names.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--COMPRESSION_LEVEL">--COMPRESSION_LEVEL </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Compression level for all compressed files created (e.g. BAM and VCF).</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">5</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--CREATE_INDEX">--CREATE_INDEX </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to create a BAM index when writing a coordinate-sorted BAM file.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--CREATE_MD5_FILE">--CREATE_MD5_FILE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to create an MD5 digest for any BAM or FASTQ files created.  </b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--DISTANCE_MODE">--DISTANCE_MODE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>The distance metric that should be used to compare the barcode-reads and the provided barcodes for finding the best and second-best assignments.</b><br />
			
		</p>
			<p>
				The --DISTANCE_MODE argument is an enumerated type (DistanceMetric), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">HAMMING</dt>
					<dd class="enum"></dd>
					<dt class="enum">LENIENT_HAMMING</dt>
					<dd class="enum"></dd>
					<dt class="enum">FREE</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">DistanceMetric</span>
			&nbsp;<span class="label">HAMMING</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--GA4GH_CLIENT_SECRETS">--GA4GH_CLIENT_SECRETS </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Google Genomics API client_secrets.json file path.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">client_secrets.json</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--LANE">--LANE </a>
			 / <small>-L</small>
		</h3>
		<p class="args">
			<b>Lane number. </b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">Integer</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MAX_MISMATCHES">--MAX_MISMATCHES </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Maximum mismatches for a barcode to be considered a match.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">1</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MAX_NO_CALLS">--MAX_NO_CALLS </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Maximum allowable number of no-calls in a barcode read before it is considered unmatchable.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">2</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MAX_RECORDS_IN_RAM">--MAX_RECORDS_IN_RAM </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>When writing files that need to be sorted, this will specify the number of records stored in RAM before spilling to disk. Increasing this number reduces the number of file handles needed to sort the file, and increases the amount of RAM needed.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Integer</span>
			&nbsp;<span class="label">500000</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--METRICS_FILE">--METRICS_FILE </a>
			 / <small>-M</small>
		</h3>
		<p class="args">
			<b>Per-barcode and per-lane metrics written to this file.</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MIN_MISMATCH_DELTA">--MIN_MISMATCH_DELTA </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Minimum difference between number of mismatches in the best and second best barcodes for a barcode to be considered a match.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">1</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MINIMUM_BASE_QUALITY">--MINIMUM_BASE_QUALITY </a>
			 / <small>-Q</small>
		</h3>
		<p class="args">
			<b>Minimum base quality. Any barcode bases falling below this quality will be considered a mismatch even if the bases match.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">0</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MINIMUM_QUALITY">--MINIMUM_QUALITY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>The minimum quality (after transforming 0s to 1s) expected from reads.  If qualities are lower than this value, an error is thrown.The default of 2 is what the Illumina's spec describes as the minimum, but in practice the value has been observed lower.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">2</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--NUM_PROCESSORS">--NUM_PROCESSORS </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Run this many PerTileBarcodeExtractors in parallel.  If NUM_PROCESSORS = 0, number of cores is automatically set to the number of cores available on the machine. If NUM_PROCESSORS < 0 then the number of cores used will be the number available on the machine less NUM_PROCESSORS.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">1</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--OUTPUT_DIR">--OUTPUT_DIR </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Where to write _barcode.txt files.  By default, these are written to BASECALLS_DIR.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--QUIET">--QUIET </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to suppress job-summary info on System.err.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--READ_STRUCTURE">--READ_STRUCTURE </a>
			 / <small>-RS</small>
		</h3>
		<p class="args">
			<b>A description of the logical structure of clusters in an Illumina Run, i.e. a description of the structure IlluminaBasecallsToSam assumes the  data to be in. It should consist of integer/character pairs describing the number of cycles and the type of those cycles (B for Sample Barcode, M for molecular barcode, T for Template, and S for skip).  E.g. If the input data consists of 80 base clusters and we provide a read structure of "28T8M8B8S28T" then the sequence may be split up into four reads:
* read one with 28 cycles (bases) of template
* read two with 8 cycles (bases) of molecular barcode (ex. unique molecular barcode)
* read three with 8 cycles (bases) of sample barcode
* 8 cycles (bases) skipped.
* read four with 28 cycles (bases) of template
The skipped cycles would NOT be included in an output SAM/BAM file or in read groups therein.</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--REFERENCE_SEQUENCE">--REFERENCE_SEQUENCE </a>
			 / <small>-R</small>
		</h3>
		<p class="args">
			<b>Reference sequence file.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--TMP_DIR">--TMP_DIR </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>One or more directories with space available to be used by this program for temporary storage of working files</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[File]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--USE_JDK_DEFLATER">--USE_JDK_DEFLATER </a>
			 / <small>-use_jdk_deflater</small>
		</h3>
		<p class="args">
			<b>Use the JDK Deflater instead of the Intel Deflater for writing compressed output</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--USE_JDK_INFLATER">--USE_JDK_INFLATER </a>
			 / <small>-use_jdk_inflater</small>
		</h3>
		<p class="args">
			<b>Use the JDK Inflater instead of the Intel Inflater for reading compressed input</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--VALIDATION_STRINGENCY">--VALIDATION_STRINGENCY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Validation stringency for all SAM files read by this program.  Setting stringency to SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.</b><br />
			
		</p>
			<p>
				The --VALIDATION_STRINGENCY argument is an enumerated type (ValidationStringency), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">STRICT</dt>
					<dd class="enum"></dd>
					<dt class="enum">LENIENT</dt>
					<dd class="enum"></dd>
					<dt class="enum">SILENT</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">ValidationStringency</span>
			&nbsp;<span class="label">STRICT</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--VERBOSITY">--VERBOSITY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Control verbosity of logging.</b><br />
			
		</p>
			<p>
				The --VERBOSITY argument is an enumerated type (LogLevel), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">ERROR</dt>
					<dd class="enum"></dd>
					<dt class="enum">WARNING</dt>
					<dd class="enum"></dd>
					<dt class="enum">INFO</dt>
					<dd class="enum"></dd>
					<dt class="enum">DEBUG</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">LogLevel</span>
			&nbsp;<span class="label">INFO</span>
		</p>

        <hr>
        <p><a href='#top'><i class='fa fa-chevron-up'></i> Return to top</a></p>
        <hr>
        <p class="see-also">See also 
        	<a href="https://www.broadinstitute.org/picard/guide/">Picard Documentation Index</a> |
        	<a href="index.html">Tool Docs Index</a> |
        	<a href="http://gatkforums.broadinstitute.org/">Support Forum</a>
        </p>

        <p class="version">Picard version 2.21.1-SNAPSHOT built at 04-35-2019 10:35:30.
        </p>

	</div>

	<?php printFooter($module); ?>