<?php

    include '../../../common/include/common.php';
    include_once '../../config.php';
    $module = modules::GATK;
    printHeader($module, "Picard | Tool Documentation Index", "Guide");
?>

<link type='text/css' rel='stylesheet' href='picardDoc.css'>

<div class='row-fluid' id="top">

	

	<?php $group = 'Variant Manipulation'; ?>

	<section class="span4">
		<aside class="well">
			<a href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Docs Index</h4></a>
		</aside>
		<aside class="well">
			<h2>Categories</h2>
        <style>
            #sidenav .accordion-body a {
                color : gray;
            }

            .accordion-body li {
                list-style : none;
            }
        </style>
        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#BaseCalling">Base Calling</a>
							<div id="BaseCalling"
								<?php echo ($group == 'Base Calling')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_illumina_CheckIlluminaDirectory.html">CheckIlluminaDirectory</a>
											</li>
											<li>
												<a href="picard_illumina_CollectIlluminaBasecallingMetrics.html">CollectIlluminaBasecallingMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_CollectIlluminaLaneMetrics.html">CollectIlluminaLaneMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_ExtractIlluminaBarcodes.html">ExtractIlluminaBarcodes</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallsToFastq.html">IlluminaBasecallsToFastq</a>
											</li>
											<li>
												<a href="picard_illumina_IlluminaBasecallsToSam.html">IlluminaBasecallsToSam</a>
											</li>
											<li>
												<a href="picard_illumina_MarkIlluminaAdapters.html">MarkIlluminaAdapters</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControl">Diagnostics and Quality Control</a>
							<div id="DiagnosticsandQualityControl"
								<?php echo ($group == 'Diagnostics and Quality Control')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_AccumulateVariantCallingMetrics.html">AccumulateVariantCallingMetrics</a>
											</li>
											<li>
												<a href="picard_sam_BamIndexStats.html">BamIndexStats</a>
											</li>
											<li>
												<a href="picard_fingerprint_CalculateFingerprintMetrics.html">CalculateFingerprintMetrics</a>
											</li>
											<li>
												<a href="picard_sam_CalculateReadGroupChecksum.html">CalculateReadGroupChecksum</a>
											</li>
											<li>
												<a href="picard_fingerprint_CheckFingerprint.html">CheckFingerprint</a>
											</li>
											<li>
												<a href="picard_sam_CheckTerminatorBlock.html">CheckTerminatorBlock</a>
											</li>
											<li>
												<a href="picard_fingerprint_ClusterCrosscheckMetrics.html">ClusterCrosscheckMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectAlignmentSummaryMetrics.html">CollectAlignmentSummaryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectBaseDistributionByCycle.html">CollectBaseDistributionByCycle</a>
											</li>
											<li>
												<a href="picard_analysis_CollectGcBiasMetrics.html">CollectGcBiasMetrics</a>
											</li>
											<li>
												<a href="picard_illumina_quality_CollectHiSeqXPfFailMetrics.html">CollectHiSeqXPfFailMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_CollectHsMetrics.html">CollectHsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_replicates_CollectIndependentReplicateMetrics.html">CollectIndependentReplicateMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectInsertSizeMetrics.html">CollectInsertSizeMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectJumpingLibraryMetrics.html">CollectJumpingLibraryMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectMultipleMetrics.html">CollectMultipleMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectOxoGMetrics.html">CollectOxoGMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectQualityYieldMetrics.html">CollectQualityYieldMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRawWgsMetrics.html">CollectRawWgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRnaSeqMetrics.html">CollectRnaSeqMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectRrbsMetrics.html">CollectRrbsMetrics</a>
											</li>
											<li>
												<a href="picard_sam_SamErrorMetric_CollectSamErrorMetrics.html">CollectSamErrorMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_CollectSequencingArtifactMetrics.html">CollectSequencingArtifactMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_directed_CollectTargetedPcrMetrics.html">CollectTargetedPcrMetrics</a>
											</li>
											<li>
												<a href="picard_vcf_CollectVariantCallingMetrics.html">CollectVariantCallingMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetrics.html">CollectWgsMetrics</a>
											</li>
											<li>
												<a href="picard_analysis_CollectWgsMetricsWithNonZeroCoverage.html">CollectWgsMetricsWithNonZeroCoverage</a>
											</li>
											<li>
												<a href="picard_analysis_CompareMetrics.html">CompareMetrics</a>
											</li>
											<li>
												<a href="picard_sam_CompareSAMs.html">CompareSAMs</a>
											</li>
											<li>
												<a href="picard_analysis_artifacts_ConvertSequencingArtifactToOxoG.html">ConvertSequencingArtifactToOxoG</a>
											</li>
											<li>
												<a href="picard_fingerprint_CrosscheckFingerprints.html">CrosscheckFingerprints</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_EstimateLibraryComplexity.html">EstimateLibraryComplexity</a>
											</li>
											<li>
												<a href="picard_analysis_MeanQualityByCycle.html">MeanQualityByCycle</a>
											</li>
											<li>
												<a href="picard_analysis_QualityScoreDistribution.html">QualityScoreDistribution</a>
											</li>
											<li>
												<a href="picard_sam_ValidateSamFile.html">ValidateSamFile</a>
											</li>
											<li>
												<a href="picard_sam_ViewSam.html">ViewSam</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#GenotypingArraysManipulation">Genotyping Arrays Manipulation</a>
							<div id="GenotypingArraysManipulation"
								<?php echo ($group == 'Genotyping Arrays Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_arrays_CreateVerifyIDIntensityContaminationMetricsFile.html">CreateVerifyIDIntensityContaminationMetricsFile</a>
											</li>
											<li>
												<a href="picard_arrays_GtcToVcf.html">GtcToVcf</a>
											</li>
											<li>
												<a href="picard_arrays_VcfToAdpc.html">VcfToAdpc</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#IntervalsManipulation">Intervals Manipulation</a>
							<div id="IntervalsManipulation"
								<?php echo ($group == 'Intervals Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_BedToIntervalList.html">BedToIntervalList</a>
											</li>
											<li>
												<a href="picard_util_IntervalListToBed.html">IntervalListToBed</a>
											</li>
											<li>
												<a href="picard_util_IntervalListTools.html">IntervalListTools</a>
											</li>
											<li>
												<a href="picard_util_LiftOverIntervalList.html">LiftOverIntervalList</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Other">Other</a>
							<div id="Other"
								<?php echo ($group == 'Other')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_FifoBuffer.html">FifoBuffer</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadDataManipulation">Read Data Manipulation</a>
							<div id="ReadDataManipulation"
								<?php echo ($group == 'Read Data Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_sam_AddCommentsToBam.html">AddCommentsToBam</a>
											</li>
											<li>
												<a href="picard_sam_AddOATag.html">AddOATag</a>
											</li>
											<li>
												<a href="picard_sam_AddOrReplaceReadGroups.html">AddOrReplaceReadGroups</a>
											</li>
											<li>
												<a href="picard_fastq_BamToBfq.html">BamToBfq</a>
											</li>
											<li>
												<a href="picard_sam_BuildBamIndex.html">BuildBamIndex</a>
											</li>
											<li>
												<a href="picard_sam_CleanSam.html">CleanSam</a>
											</li>
											<li>
												<a href="picard_sam_DownsampleSam.html">DownsampleSam</a>
											</li>
											<li>
												<a href="picard_sam_FastqToSam.html">FastqToSam</a>
											</li>
											<li>
												<a href="picard_sam_FilterSamReads.html">FilterSamReads</a>
											</li>
											<li>
												<a href="picard_sam_FixMateInformation.html">FixMateInformation</a>
											</li>
											<li>
												<a href="picard_sam_GatherBamFiles.html">GatherBamFiles</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_MarkDuplicates.html">MarkDuplicates</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_MarkDuplicatesWithMateCigar.html">MarkDuplicatesWithMateCigar</a>
											</li>
											<li>
												<a href="picard_sam_MergeBamAlignment.html">MergeBamAlignment</a>
											</li>
											<li>
												<a href="picard_sam_MergeSamFiles.html">MergeSamFiles</a>
											</li>
											<li>
												<a href="picard_sam_PositionBasedDownsampleSam.html">PositionBasedDownsampleSam</a>
											</li>
											<li>
												<a href="picard_sam_ReorderSam.html">ReorderSam</a>
											</li>
											<li>
												<a href="picard_sam_ReplaceSamHeader.html">ReplaceSamHeader</a>
											</li>
											<li>
												<a href="picard_sam_RevertOriginalBaseQualitiesAndAddMateCigar.html">RevertOriginalBaseQualitiesAndAddMateCigar</a>
											</li>
											<li>
												<a href="picard_sam_RevertSam.html">RevertSam</a>
											</li>
											<li>
												<a href="picard_sam_SamFormatConverter.html">SamFormatConverter</a>
											</li>
											<li>
												<a href="picard_sam_SamToFastq.html">SamToFastq</a>
											</li>
											<li>
												<a href="picard_sam_SetNmAndUqTags.html">SetNmAndUqTags</a>
											</li>
											<li>
												<a href="picard_sam_SetNmMdAndUqTags.html">SetNmMdAndUqTags</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_SimpleMarkDuplicatesWithMateCigar.html">SimpleMarkDuplicatesWithMateCigar</a>
											</li>
											<li>
												<a href="picard_sam_SortSam.html">SortSam</a>
											</li>
											<li>
												<a href="picard_sam_SplitSamByLibrary.html">SplitSamByLibrary</a>
											</li>
											<li>
												<a href="picard_sam_SplitSamByNumberOfReads.html">SplitSamByNumberOfReads</a>
											</li>
											<li>
												<a href="picard_sam_markduplicates_UmiAwareMarkDuplicatesWithMateCigar.html">UmiAwareMarkDuplicatesWithMateCigar</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Reference">Reference</a>
							<div id="Reference"
								<?php echo ($group == 'Reference')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_util_BaitDesigner.html">BaitDesigner</a>
											</li>
											<li>
												<a href="picard_sam_CreateSequenceDictionary.html">CreateSequenceDictionary</a>
											</li>
											<li>
												<a href="picard_reference_ExtractSequences.html">ExtractSequences</a>
											</li>
											<li>
												<a href="picard_reference_NonNFastaSize.html">NonNFastaSize</a>
											</li>
											<li>
												<a href="picard_reference_NormalizeFasta.html">NormalizeFasta</a>
											</li>
											<li>
												<a href="picard_util_ScatterIntervalsByNs.html">ScatterIntervalsByNs</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationandRefinement">Variant Evaluation and Refinement</a>
							<div id="VariantEvaluationandRefinement"
								<?php echo ($group == 'Variant Evaluation and Refinement')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_MendelianViolations_FindMendelianViolations.html">FindMendelianViolations</a>
											</li>
											<li>
												<a href="picard_vcf_GenotypeConcordance.html">GenotypeConcordance</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantFiltering">Variant Filtering</a>
							<div id="VariantFiltering"
								<?php echo ($group == 'Variant Filtering')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_filter_FilterVcf.html">FilterVcf</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantManipulation">Variant Manipulation</a>
							<div id="VariantManipulation"
								<?php echo ($group == 'Variant Manipulation')? 'class="accordion-body collapse in"'.chr(62) : 'class="accordion-body collapse"'.chr(62);?>
								<ul>
											<li>
												<a href="picard_vcf_FixVcfHeader.html">FixVcfHeader</a>
											</li>
											<li>
												<a href="picard_vcf_GatherVcfs.html">GatherVcfs</a>
											</li>
											<li>
												<a href="picard_vcf_LiftoverVcf.html">LiftoverVcf</a>
											</li>
											<li>
												<a href="picard_vcf_MakeSitesOnlyVcf.html">MakeSitesOnlyVcf</a>
											</li>
											<li>
												<a href="picard_vcf_MergeVcfs.html">MergeVcfs</a>
											</li>
											<li>
												<a href="picard_vcf_RenameSampleInVcf.html">RenameSampleInVcf</a>
											</li>
											<li>
												<a href="picard_vcf_SortVcf.html">SortVcf</a>
											</li>
											<li>
												<a href="picard_vcf_SplitVcfs.html">SplitVcfs</a>
											</li>
											<li>
												<a href="picard_vcf_UpdateVcfSequenceDictionary.html">UpdateVcfSequenceDictionary</a>
											</li>
											<li>
												<a href="picard_vcf_VcfFormatConverter.html">VcfFormatConverter</a>
											</li>
											<li>
												<a href="picard_vcf_VcfToIntervalList.html">VcfToIntervalList</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
        		<hr>
        </ul>
		</aside>
		<?php getForumPosts( 'VcfToIntervalList' ) ?>

	</section>

	<div class="span8">

		<h1>VcfToIntervalList</h1>

		<p class="lead">Converts a VCF or BCF file to a Picard Interval List</p>

			<h3>Category
				<small> Variant Manipulation</small>
			</h3>
		<hr>
		<h2>Overview</h2>
		Converts a VCF or BCF file to a Picard Interval List.

 <p>This tool creates a Picard Interval List from a VCF or BCF. It is important
 that the file extension is included as the file format is determined by the file
 extension. Variants that were filtered can be included in the output interval list by
 setting INCLUDE_FILTERED to true. </p>

 <h3>Inputs</h3>
 <ul>
     <li> A BCF or VCF input file  </li>
     <li> Boolean if variants that were filtered should be included in the output interval list </li>
 </ul>

 <h3>Output</h3>
 <ul>
 A Picard Interval List
 </ul>

 <p>
 <h4>Usage example:</h4>
 <pre>
     java -jar picard.jar VcfToIntervalList \
          I=input_variants.vcf \
          O=output.interval_list
 </pre>
 </p>


				<h3>VcfToIntervalList specific arguments</h3>
				<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
				<table class="table table-striped table-bordered table-condensed">
					<thead>
					<tr>
						<th>Argument name(s)</th>
						<th>Default value</th>
						<th>Summary</th>
					</tr>
					</thead>
					<tbody>
			<tr>
				<th colspan="4" id="row-divider">Required Arguments</th>
			</tr>
				<tr>
					<td><a href="#--INPUT">--INPUT</a><br />
								&nbsp;<em>-I</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>The BCF or VCF input file. The file format is determined by file extension.</td>
				</tr>
				<tr>
					<td><a href="#--OUTPUT">--OUTPUT</a><br />
								&nbsp;<em>-O</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>The output Picard Interval List.</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Tool Arguments</th>
			</tr>
				<tr>
					<td><a href="#--INCLUDE_FILTERED">--INCLUDE_FILTERED</a><br />
								&nbsp;<em>-IF</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Include variants that were filtered in the output interval list.</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Common Arguments</th>
			</tr>
				<tr>
					<td><a href="#--COMPRESSION_LEVEL">--COMPRESSION_LEVEL</a><br />
					</td>
					<!--<td>int</td> -->
					<td>5</td>
					<td>Compression level for all compressed files created (e.g. BAM and VCF).</td>
				</tr>
				<tr>
					<td><a href="#--CREATE_INDEX">--CREATE_INDEX</a><br />
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Whether to create a BAM index when writing a coordinate-sorted BAM file.</td>
				</tr>
				<tr>
					<td><a href="#--CREATE_MD5_FILE">--CREATE_MD5_FILE</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Whether to create an MD5 digest for any BAM or FASTQ files created.  </td>
				</tr>
				<tr>
					<td><a href="#--GA4GH_CLIENT_SECRETS">--GA4GH_CLIENT_SECRETS</a><br />
					</td>
					<!--<td>String</td> -->
					<td>client_secrets.json</td>
					<td>Google Genomics API client_secrets.json file path.</td>
				</tr>
				<tr>
					<td><a href="#--MAX_RECORDS_IN_RAM">--MAX_RECORDS_IN_RAM</a><br />
					</td>
					<!--<td>Integer</td> -->
					<td>500000</td>
					<td>When writing files that need to be sorted, this will specify the number of records stored in RAM before spilling to disk. Increasing this number reduces the number of file handles needed to sort the file, and increases the amount of RAM needed.</td>
				</tr>
				<tr>
					<td><a href="#--QUIET">--QUIET</a><br />
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Whether to suppress job-summary info on System.err.</td>
				</tr>
				<tr>
					<td><a href="#--REFERENCE_SEQUENCE">--REFERENCE_SEQUENCE</a><br />
								&nbsp;<em>-R</em>
					</td>
					<!--<td>File</td> -->
					<td>null</td>
					<td>Reference sequence file.</td>
				</tr>
				<tr>
					<td><a href="#--TMP_DIR">--TMP_DIR</a><br />
					</td>
					<!--<td>List[File]</td> -->
					<td>[]</td>
					<td>One or more directories with space available to be used by this program for temporary storage of working files</td>
				</tr>
				<tr>
					<td><a href="#--USE_JDK_DEFLATER">--USE_JDK_DEFLATER</a><br />
								&nbsp;<em>-use_jdk_deflater</em>
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Use the JDK Deflater instead of the Intel Deflater for writing compressed output</td>
				</tr>
				<tr>
					<td><a href="#--USE_JDK_INFLATER">--USE_JDK_INFLATER</a><br />
								&nbsp;<em>-use_jdk_inflater</em>
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Use the JDK Inflater instead of the Intel Inflater for reading compressed input</td>
				</tr>
				<tr>
					<td><a href="#--VALIDATION_STRINGENCY">--VALIDATION_STRINGENCY</a><br />
					</td>
					<!--<td>ValidationStringency</td> -->
					<td>STRICT</td>
					<td>Validation stringency for all SAM files read by this program.  Setting stringency to SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.</td>
				</tr>
				<tr>
					<td><a href="#--VERBOSITY">--VERBOSITY</a><br />
					</td>
					<!--<td>LogLevel</td> -->
					<td>INFO</td>
					<td>Control verbosity of logging.</td>
				</tr>
					</tbody>
				</table>

					<h3>Argument details</h3>
					<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--COMPRESSION_LEVEL">--COMPRESSION_LEVEL </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Compression level for all compressed files created (e.g. BAM and VCF).</b><br />
			
		</p>
		<p>
		<span class="label label-info ">int</span>
			&nbsp;<span class="label">5</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--CREATE_INDEX">--CREATE_INDEX </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to create a BAM index when writing a coordinate-sorted BAM file.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--CREATE_MD5_FILE">--CREATE_MD5_FILE </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to create an MD5 digest for any BAM or FASTQ files created.  </b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--GA4GH_CLIENT_SECRETS">--GA4GH_CLIENT_SECRETS </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Google Genomics API client_secrets.json file path.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">String</span>
			&nbsp;<span class="label">client_secrets.json</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--INCLUDE_FILTERED">--INCLUDE_FILTERED </a>
			 / <small>-IF</small>
		</h3>
		<p class="args">
			<b>Include variants that were filtered in the output interval list.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--INPUT">--INPUT </a>
			 / <small>-I</small>
		</h3>
		<p class="args">
			<b>The BCF or VCF input file. The file format is determined by file extension.</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MAX_RECORDS_IN_RAM">--MAX_RECORDS_IN_RAM </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>When writing files that need to be sorted, this will specify the number of records stored in RAM before spilling to disk. Increasing this number reduces the number of file handles needed to sort the file, and increases the amount of RAM needed.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Integer</span>
			&nbsp;<span class="label">500000</span>
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--OUTPUT">--OUTPUT </a>
			 / <small>-O</small>
		</h3>
		<p class="args">
			<b>The output Picard Interval List.</b><br />
			
		</p>
		<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--QUIET">--QUIET </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Whether to suppress job-summary info on System.err.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--REFERENCE_SEQUENCE">--REFERENCE_SEQUENCE </a>
			 / <small>-R</small>
		</h3>
		<p class="args">
			<b>Reference sequence file.</b><br />
			
		</p>
		<p>
		<span class="label label-info ">File</span>
			&nbsp;<span class="label">null</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--TMP_DIR">--TMP_DIR </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>One or more directories with space available to be used by this program for temporary storage of working files</b><br />
			
		</p>
		<p>
		<span class="label label-info ">List[File]</span>
			&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--USE_JDK_DEFLATER">--USE_JDK_DEFLATER </a>
			 / <small>-use_jdk_deflater</small>
		</h3>
		<p class="args">
			<b>Use the JDK Deflater instead of the Intel Deflater for writing compressed output</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--USE_JDK_INFLATER">--USE_JDK_INFLATER </a>
			 / <small>-use_jdk_inflater</small>
		</h3>
		<p class="args">
			<b>Use the JDK Inflater instead of the Intel Inflater for reading compressed input</b><br />
			
		</p>
		<p>
		<span class="label label-info ">Boolean</span>
			&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--VALIDATION_STRINGENCY">--VALIDATION_STRINGENCY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Validation stringency for all SAM files read by this program.  Setting stringency to SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.</b><br />
			
		</p>
			<p>
				The --VALIDATION_STRINGENCY argument is an enumerated type (ValidationStringency), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">STRICT</dt>
					<dd class="enum"></dd>
					<dt class="enum">LENIENT</dt>
					<dd class="enum"></dd>
					<dt class="enum">SILENT</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">ValidationStringency</span>
			&nbsp;<span class="label">STRICT</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--VERBOSITY">--VERBOSITY </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Control verbosity of logging.</b><br />
			
		</p>
			<p>
				The --VERBOSITY argument is an enumerated type (LogLevel), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">ERROR</dt>
					<dd class="enum"></dd>
					<dt class="enum">WARNING</dt>
					<dd class="enum"></dd>
					<dt class="enum">INFO</dt>
					<dd class="enum"></dd>
					<dt class="enum">DEBUG</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
		<span class="label label-info ">LogLevel</span>
			&nbsp;<span class="label">INFO</span>
		</p>

        <hr>
        <p><a href='#top'><i class='fa fa-chevron-up'></i> Return to top</a></p>
        <hr>
        <p class="see-also">See also 
        	<a href="https://www.broadinstitute.org/picard/guide/">Picard Documentation Index</a> |
        	<a href="index.html">Tool Docs Index</a> |
        	<a href="http://gatkforums.broadinstitute.org/">Support Forum</a>
        </p>

        <p class="version">Picard version 2.21.1-SNAPSHOT built at 04-35-2019 10:35:30.
        </p>

	</div>

	<?php printFooter($module); ?>