<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_201) on Fri Oct 04 10:35:33 EDT 2019 -->
<title>ReadBaseStratification (picard 2.21.1-SNAPSHOT API)</title>
<meta name="date" content="2019-10-04">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="ReadBaseStratification (picard 2.21.1-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":9,"i1":9,"i2":9,"i3":9};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-all.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../picard/sam/SamErrorMetric/OverlappingReadsErrorCalculator.html" title="class in picard.sam.SamErrorMetric"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.BinnedReadCycleStratifier.html" title="class in picard.sam.SamErrorMetric"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?picard/sam/SamErrorMetric/ReadBaseStratification.html" target="_top">Frames</a></li>
<li><a href="ReadBaseStratification.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">picard.sam.SamErrorMetric</div>
<h2 title="Class ReadBaseStratification" class="title">Class ReadBaseStratification</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>picard.sam.SamErrorMetric.ReadBaseStratification</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public class <span class="typeNameLabel">ReadBaseStratification</span>
extends java.lang.Object</pre>
<div class="block">Classes, methods, and enums that deal with the stratification of read bases and reference information.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.BinnedReadCycleStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.BinnedReadCycleStratifier</a></span></code>
<div class="block">Stratifies into quintiles of read cycle.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.CigarOperatorsInReadStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.CigarOperatorsInReadStratifier</a></span></code>
<div class="block">Stratifies according to the number of matching cigar operators (from CIGAR string) that the read has.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.CollectionStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.CollectionStratifier</a></span></code>
<div class="block">A CollectionStratifier is a stratifier that uses a collection of stratifiers to inform the stratification.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.Consensus.html" title="enum in picard.sam.SamErrorMetric">ReadBaseStratification.Consensus</a></span></code>
<div class="block">Types of consensus reads as determined by the number of duplicates used from
 first and second strands.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.ConsensusStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.ConsensusStratifier</a></span></code>
<div class="block">Stratify by tags used during duplex and single index consensus calling.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.CycleBin.html" title="enum in picard.sam.SamErrorMetric">ReadBaseStratification.CycleBin</a></span></code>
<div class="block">An enum designed to hold a binned version of any probability-like number (between 0 and 1)
 in quintiles</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.FlowCellTileStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.FlowCellTileStratifier</a></span></code>
<div class="block">Stratifies base into their read's tile which is parsed from the read-name.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.GCContentStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.GCContentStratifier</a></span></code>
<div class="block">A stratifier that uses GC (of the read) to stratify.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.IndelLengthStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.IndelLengthStratifier</a></span></code>
<div class="block">Stratifies according to the length of an insertion or deletion.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.IndelsInReadStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.IndelsInReadStratifier</a></span></code>
<div class="block">Stratifies according to the number of indel bases (from CIGAR string) that the read has.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.LongShortHomopolymer.html" title="enum in picard.sam.SamErrorMetric">ReadBaseStratification.LongShortHomopolymer</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.LongShortHomopolymerStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.LongShortHomopolymerStratifier</a></span></code>
<div class="block">Stratify bases according to the type of Homopolymer that they belong to (repeating element, final reference base and
 whether the length is "long" or not).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.MismatchesInReadStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.MismatchesInReadStratifier</a></span></code>
<div class="block">Stratifies according to the overall mismatches (from <code>SAMTag.NM</code>) that the read has against the reference, NOT
 including the current base.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.NsInReadStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.NsInReadStratifier</a></span></code>
<div class="block">Stratify by the number of Ns found in the read.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.PairOrientation.html" title="enum in picard.sam.SamErrorMetric">ReadBaseStratification.PairOrientation</a></span></code>
<div class="block">An enum for holding a reads read-pair's Orientation (i.e.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.PairStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.PairStratifier</a>&lt;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.PairStratifier.html" title="type parameter in ReadBaseStratification.PairStratifier">T</a> extends java.lang.Comparable&lt;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.PairStratifier.html" title="type parameter in ReadBaseStratification.PairStratifier">T</a>&gt;,<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.PairStratifier.html" title="type parameter in ReadBaseStratification.PairStratifier">R</a> extends java.lang.Comparable&lt;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.PairStratifier.html" title="type parameter in ReadBaseStratification.PairStratifier">R</a>&gt;&gt;</span></code>
<div class="block">A PairStratifier is a stratifier that uses two other stratifiers to inform the stratification.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.ProperPaired.html" title="enum in picard.sam.SamErrorMetric">ReadBaseStratification.ProperPaired</a></span></code>
<div class="block">An enum to hold information about the "properness" of a read pair</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.ReadDirection.html" title="enum in picard.sam.SamErrorMetric">ReadBaseStratification.ReadDirection</a></span></code>
<div class="block">An enum for holding the direction for a read (positive strand or negative strand</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.ReadOrdinality.html" title="enum in picard.sam.SamErrorMetric">ReadBaseStratification.ReadOrdinality</a></span></code>
<div class="block">An enum to hold the ordinality of a read</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static interface&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="type parameter in ReadBaseStratification.RecordAndOffsetStratifier">T</a> extends java.lang.Comparable&lt;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="type parameter in ReadBaseStratification.RecordAndOffsetStratifier">T</a>&gt;&gt;</span></code>
<div class="block">The main interface for a stratifier.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Integer&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#baseCycleStratifier">baseCycleStratifier</a></span></code>
<div class="block">Get the one-based cycle number of the base, taking the direction of the read into account</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Byte&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#baseQualityStratifier">baseQualityStratifier</a></span></code>
<div class="block">Stratifies into the base-quality of the base under consideration</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static htsjdk.samtools.util.Lazy&lt;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.PairStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.PairStratifier</a>&lt;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.LongShortHomopolymer.html" title="enum in picard.sam.SamErrorMetric">ReadBaseStratification.LongShortHomopolymer</a>,<a href="../../../picard/sam/util/Pair.html" title="class in picard.sam.util">Pair</a>&lt;java.lang.Character,java.lang.Character&gt;&gt;&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#binnedHomopolymerStratifier">binnedHomopolymerStratifier</a></span></code>
<div class="block">Stratifies a base onto the make up (repeating base and following reference base) and length-scale of the homopolymer from whence it came.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.BinnedReadCycleStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.BinnedReadCycleStratifier</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#binnedReadCycleStratifier">binnedReadCycleStratifier</a></span></code>
<div class="block">Stratifies into quintiles of read cycle.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.ConsensusStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.ConsensusStratifier</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#consensusStratifier">consensusStratifier</a></span></code>
<div class="block">Stratify by tags used during duplex and single index consensus calling.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Character&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#currentReadBaseStratifier">currentReadBaseStratifier</a></span></code>
<div class="block">Stratifies bases into the current (uppercase) base as it was read from the sequencer (i.e.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.CigarOperatorsInReadStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.CigarOperatorsInReadStratifier</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#deletionsInReadStratifier">deletionsInReadStratifier</a></span></code>
<div class="block">Stratify by Deletions in the read cigars.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.FlowCellTileStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.FlowCellTileStratifier</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#flowCellTileStratifier">flowCellTileStratifier</a></span></code>
<div class="block">Stratifies base into their read's tile which is parsed from the read-name.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.GCContentStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.GCContentStratifier</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#gcContentStratifier">gcContentStratifier</a></span></code>
<div class="block">A stratifier that uses GC (of the read) to stratify.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Integer&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#homoPolymerLengthStratifier">homoPolymerLengthStratifier</a></span></code>
<div class="block">Stratifies a base onto the length of the homopolymer preceding it (in the read direction).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.PairStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.PairStratifier</a>&lt;java.lang.Integer,<a href="../../../picard/sam/util/Pair.html" title="class in picard.sam.util">Pair</a>&lt;java.lang.Character,java.lang.Character&gt;&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#homopolymerStratifier">homopolymerStratifier</a></span></code>
<div class="block">Stratifies a base onto the make up (repeating base and following reference base) and length of the homopolymer from whence it came.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.IndelLengthStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.IndelLengthStratifier</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#indelLengthStratifier">indelLengthStratifier</a></span></code>
<div class="block">Stratifies into the number of bases in an insertion</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.IndelsInReadStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.IndelsInReadStratifier</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#indelsInReadStratifier">indelsInReadStratifier</a></span></code>
<div class="block">Stratify by Indels in the read cigars.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.CigarOperatorsInReadStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.CigarOperatorsInReadStratifier</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#insertionsInReadStratifier">insertionsInReadStratifier</a></span></code>
<div class="block">Stratify by Insertions in the read cigars.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Integer&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#insertLengthStratifier">insertLengthStratifier</a></span></code>
<div class="block">Stratifies into the read-pairs estimated insert-length, as long as it isn't larger than 10x the length of the read</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Integer&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#mappingQualityStratifier">mappingQualityStratifier</a></span></code>
<div class="block">Stratifies into the mapping-quality of the read under consideration</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.MismatchesInReadStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.MismatchesInReadStratifier</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#mismatchesInReadStratifier">mismatchesInReadStratifier</a></span></code>
<div class="block">Stratifies according to the overall mismatches (from NM) that the read has against the reference, NOT
 including the current base.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Character&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#nextReadBaseStratifier">nextReadBaseStratifier</a></span></code>
<div class="block">Stratifies bases into the following (uppercase) base as it was read from the sequencer (i.e.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static picard.sam.SamErrorMetric.ReadBaseStratification.RecordStratifier&lt;java.lang.String&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#nonStratifier">nonStratifier</a></span></code>
<div class="block">A constant stratifier which places all the reads into a single stratum.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#NOT_ALIGNED_ERROR">NOT_ALIGNED_ERROR</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.NsInReadStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.NsInReadStratifier</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#nsInReadStratifier">nsInReadStratifier</a></span></code>
<div class="block">Stratify by the number of Ns found in the read.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.String&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#oneBasePaddedContextStratifier">oneBasePaddedContextStratifier</a></span></code>
<div class="block">Stratifies into the one base context of the base, which includes one read-base on each side and the reference base from where the current base is
 the bases will be reverse-complemented so that the bases are in the original order they were read from the sequencer</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.PairStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.PairStratifier</a>&lt;java.lang.Character,java.lang.Character&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#postDiNucleotideStratifier">postDiNucleotideStratifier</a></span></code>
<div class="block">Stratifies a base onto the reference base that it covers and the following base, possibly reverse complemented if the read
 has been reversed by the aligner.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.PairStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.PairStratifier</a>&lt;java.lang.Character,java.lang.Character&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#preDiNucleotideStratifier">preDiNucleotideStratifier</a></span></code>
<div class="block">Stratifies a base onto the reference base that it covers and the preceding base, possibly reverse complemented if the read
 has been reversed by the aligner.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Character&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#previousReadBaseStratifier">previousReadBaseStratifier</a></span></code>
<div class="block">Stratifies bases into the previous (uppercase) base as it was read from the sequencer (i.e.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.ReadDirection.html" title="enum in picard.sam.SamErrorMetric">ReadBaseStratification.ReadDirection</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#readDirectionStratifier">readDirectionStratifier</a></span></code>
<div class="block">Stratifies bases into their read's Direction (i.e.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static picard.sam.SamErrorMetric.ReadBaseStratification.RecordStratifier&lt;java.lang.String&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#readgroupStratifier">readgroupStratifier</a></span></code>
<div class="block">Stratifies to the readgroup id of the read.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.ReadOrdinality.html" title="enum in picard.sam.SamErrorMetric">ReadBaseStratification.ReadOrdinality</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#readOrdinalityStratifier">readOrdinalityStratifier</a></span></code>
<div class="block">Stratifies bases into their read's Ordinality (i.e.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.PairOrientation.html" title="enum in picard.sam.SamErrorMetric">ReadBaseStratification.PairOrientation</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#readOrientationStratifier">readOrientationStratifier</a></span></code>
<div class="block">Stratifies bases into their read-pair's Orientation (i.e.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.ProperPaired.html" title="enum in picard.sam.SamErrorMetric">ReadBaseStratification.ProperPaired</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#readPairednessStratifier">readPairednessStratifier</a></span></code>
<div class="block">Stratifies bases into their read's Proper-pairedness</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Character&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#referenceBaseStratifier">referenceBaseStratifier</a></span></code>
<div class="block">Stratifies a base onto the reference base that it covers, possibly reverse complemented if the read
 has been reversed by the aligner.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Integer&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#softClipsLengthStratifier">softClipsLengthStratifier</a></span></code>
<div class="block">Stratifies into the number of soft-clipped bases that the read has in its alignment, or <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#NOT_ALIGNED_ERROR">-1</a> if not aligned.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.String&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#twoBasePaddedContextStratifier">twoBasePaddedContextStratifier</a></span></code>
<div class="block">Stratifies into the two base context of the base, which includes two read-bases on each side and the reference base from where the current base is
 the bases will be reverse-complemented so that the bases are in the original order they were read from the sequencer</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#ReadBaseStratification--">ReadBaseStratification</a></span>()</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>static htsjdk.samtools.CigarElement</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#getIndelElement-htsjdk.samtools.util.SamLocusIterator.RecordAndOffset-">getIndelElement</a></span>(htsjdk.samtools.util.SamLocusIterator.RecordAndOffset&nbsp;recordAndOffset)</code>&nbsp;</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>static &lt;T extends java.lang.Comparable&lt;T&gt;,S extends java.lang.Comparable&lt;S&gt;&gt;<br><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.PairStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.PairStratifier</a>&lt;T,S&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#PairStratifierFactory-picard.sam.SamErrorMetric.ReadBaseStratification.RecordAndOffsetStratifier-picard.sam.SamErrorMetric.ReadBaseStratification.RecordAndOffsetStratifier-java.lang.String-">PairStratifierFactory</a></span>(<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;T&gt;&nbsp;leftStratifier,
                     <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;S&gt;&nbsp;rightStratifier,
                     java.lang.String&nbsp;suffix)</code>
<div class="block">A factory for generating "pair" stratifier instances from two stratifiers and a string</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#setGcCacheSize-int-">setGcCacheSize</a></span>(int&nbsp;gcCacheSize)</code>&nbsp;</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#setLongHomopolymer-int-">setLongHomopolymer</a></span>(int&nbsp;longHomopolymer)</code>
<div class="block">defaults to 6</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="currentReadBaseStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>currentReadBaseStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Character&gt; currentReadBaseStratifier</pre>
<div class="block">Stratifies bases into the current (uppercase) base as it was read from the sequencer (i.e. complemented if needed)</div>
</li>
</ul>
<a name="previousReadBaseStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>previousReadBaseStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Character&gt; previousReadBaseStratifier</pre>
<div class="block">Stratifies bases into the previous (uppercase) base as it was read from the sequencer (i.e. complemented if needed)</div>
</li>
</ul>
<a name="nextReadBaseStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>nextReadBaseStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Character&gt; nextReadBaseStratifier</pre>
<div class="block">Stratifies bases into the following (uppercase) base as it was read from the sequencer (i.e. complemented if needed)</div>
</li>
</ul>
<a name="referenceBaseStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>referenceBaseStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Character&gt; referenceBaseStratifier</pre>
<div class="block">Stratifies a base onto the reference base that it covers, possibly reverse complemented if the read
 has been reversed by the aligner.</div>
</li>
</ul>
<a name="postDiNucleotideStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>postDiNucleotideStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.PairStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.PairStratifier</a>&lt;java.lang.Character,java.lang.Character&gt; postDiNucleotideStratifier</pre>
<div class="block">Stratifies a base onto the reference base that it covers and the following base, possibly reverse complemented if the read
 has been reversed by the aligner.</div>
</li>
</ul>
<a name="preDiNucleotideStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>preDiNucleotideStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.PairStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.PairStratifier</a>&lt;java.lang.Character,java.lang.Character&gt; preDiNucleotideStratifier</pre>
<div class="block">Stratifies a base onto the reference base that it covers and the preceding base, possibly reverse complemented if the read
 has been reversed by the aligner.</div>
</li>
</ul>
<a name="homoPolymerLengthStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>homoPolymerLengthStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Integer&gt; homoPolymerLengthStratifier</pre>
<div class="block">Stratifies a base onto the length of the homopolymer preceding it (in the read direction).
 Read direction and only the preceding bases are taken into account (i.e. ignoring reference and current base).</div>
</li>
</ul>
<a name="homopolymerStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>homopolymerStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.PairStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.PairStratifier</a>&lt;java.lang.Integer,<a href="../../../picard/sam/util/Pair.html" title="class in picard.sam.util">Pair</a>&lt;java.lang.Character,java.lang.Character&gt;&gt; homopolymerStratifier</pre>
<div class="block">Stratifies a base onto the make up (repeating base and following reference base) and length of the homopolymer from whence it came.
 Read direction and only the preceding bases are taken into account.</div>
</li>
</ul>
<a name="binnedHomopolymerStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>binnedHomopolymerStratifier</h4>
<pre>public static final&nbsp;htsjdk.samtools.util.Lazy&lt;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.PairStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.PairStratifier</a>&lt;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.LongShortHomopolymer.html" title="enum in picard.sam.SamErrorMetric">ReadBaseStratification.LongShortHomopolymer</a>,<a href="../../../picard/sam/util/Pair.html" title="class in picard.sam.util">Pair</a>&lt;java.lang.Character,java.lang.Character&gt;&gt;&gt; binnedHomopolymerStratifier</pre>
<div class="block">Stratifies a base onto the make up (repeating base and following reference base) and length-scale of the homopolymer from whence it came.
 Read direction and only the preceding bases are taken into account.
 <p>
 This is done with a Lazy wrapper since it requires access to <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#LONG_HOMOPOLYMER"><code>LONG_HOMOPOLYMER</code></a> which can be set with <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#setLongHomopolymer-int-"><code>setLongHomopolymer(int)</code></a>.</div>
</li>
</ul>
<a name="oneBasePaddedContextStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>oneBasePaddedContextStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.String&gt; oneBasePaddedContextStratifier</pre>
<div class="block">Stratifies into the one base context of the base, which includes one read-base on each side and the reference base from where the current base is
 the bases will be reverse-complemented so that the bases are in the original order they were read from the sequencer</div>
</li>
</ul>
<a name="twoBasePaddedContextStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>twoBasePaddedContextStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.String&gt; twoBasePaddedContextStratifier</pre>
<div class="block">Stratifies into the two base context of the base, which includes two read-bases on each side and the reference base from where the current base is
 the bases will be reverse-complemented so that the bases are in the original order they were read from the sequencer</div>
</li>
</ul>
<a name="nonStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>nonStratifier</h4>
<pre>public static final&nbsp;picard.sam.SamErrorMetric.ReadBaseStratification.RecordStratifier&lt;java.lang.String&gt; nonStratifier</pre>
<div class="block">A constant stratifier which places all the reads into a single stratum.</div>
</li>
</ul>
<a name="gcContentStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>gcContentStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.GCContentStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.GCContentStratifier</a> gcContentStratifier</pre>
<div class="block">A stratifier that uses GC (of the read) to stratify.
 Since the reads are expected to be seen over and over again, the GC is cached
 <p>
 Stratification happens into (integer) percents.</div>
</li>
</ul>
<a name="flowCellTileStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>flowCellTileStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.FlowCellTileStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.FlowCellTileStratifier</a> flowCellTileStratifier</pre>
<div class="block">Stratifies base into their read's tile which is parsed from the read-name.</div>
</li>
</ul>
<a name="readgroupStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readgroupStratifier</h4>
<pre>public static final&nbsp;picard.sam.SamErrorMetric.ReadBaseStratification.RecordStratifier&lt;java.lang.String&gt; readgroupStratifier</pre>
<div class="block">Stratifies to the readgroup id of the read.</div>
</li>
</ul>
<a name="readOrdinalityStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readOrdinalityStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.ReadOrdinality.html" title="enum in picard.sam.SamErrorMetric">ReadBaseStratification.ReadOrdinality</a>&gt; readOrdinalityStratifier</pre>
<div class="block">Stratifies bases into their read's Ordinality (i.e. First or Second)</div>
</li>
</ul>
<a name="readPairednessStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readPairednessStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.ProperPaired.html" title="enum in picard.sam.SamErrorMetric">ReadBaseStratification.ProperPaired</a>&gt; readPairednessStratifier</pre>
<div class="block">Stratifies bases into their read's Proper-pairedness</div>
</li>
</ul>
<a name="readDirectionStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readDirectionStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.ReadDirection.html" title="enum in picard.sam.SamErrorMetric">ReadBaseStratification.ReadDirection</a>&gt; readDirectionStratifier</pre>
<div class="block">Stratifies bases into their read's Direction (i.e. forward or reverse)</div>
</li>
</ul>
<a name="readOrientationStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readOrientationStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.PairOrientation.html" title="enum in picard.sam.SamErrorMetric">ReadBaseStratification.PairOrientation</a>&gt; readOrientationStratifier</pre>
<div class="block">Stratifies bases into their read-pair's Orientation (i.e. F1R2, F2R1, F1F2 or R1R2)</div>
</li>
</ul>
<a name="binnedReadCycleStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>binnedReadCycleStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.BinnedReadCycleStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.BinnedReadCycleStratifier</a> binnedReadCycleStratifier</pre>
<div class="block">Stratifies into quintiles of read cycle.</div>
</li>
</ul>
<a name="baseCycleStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>baseCycleStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Integer&gt; baseCycleStratifier</pre>
<div class="block">Get the one-based cycle number of the base, taking the direction of the read into account</div>
</li>
</ul>
<a name="insertLengthStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>insertLengthStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Integer&gt; insertLengthStratifier</pre>
<div class="block">Stratifies into the read-pairs estimated insert-length, as long as it isn't larger than 10x the length of the read</div>
</li>
</ul>
<a name="softClipsLengthStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>softClipsLengthStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Integer&gt; softClipsLengthStratifier</pre>
<div class="block">Stratifies into the number of soft-clipped bases that the read has in its alignment, or <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.html#NOT_ALIGNED_ERROR">-1</a> if not aligned.</div>
</li>
</ul>
<a name="baseQualityStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>baseQualityStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Byte&gt; baseQualityStratifier</pre>
<div class="block">Stratifies into the base-quality of the base under consideration</div>
</li>
</ul>
<a name="mappingQualityStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>mappingQualityStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;java.lang.Integer&gt; mappingQualityStratifier</pre>
<div class="block">Stratifies into the mapping-quality of the read under consideration</div>
</li>
</ul>
<a name="mismatchesInReadStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>mismatchesInReadStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.MismatchesInReadStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.MismatchesInReadStratifier</a> mismatchesInReadStratifier</pre>
<div class="block">Stratifies according to the overall mismatches (from NM) that the read has against the reference, NOT
 including the current base.</div>
</li>
</ul>
<a name="consensusStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>consensusStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.ConsensusStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.ConsensusStratifier</a> consensusStratifier</pre>
<div class="block">Stratify by tags used during duplex and single index consensus calling.</div>
</li>
</ul>
<a name="nsInReadStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>nsInReadStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.NsInReadStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.NsInReadStratifier</a> nsInReadStratifier</pre>
<div class="block">Stratify by the number of Ns found in the read.
 This is particularly useful for data that has been consensus-called (a process that can add 'N' bases when there is no consensus)</div>
</li>
</ul>
<a name="insertionsInReadStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>insertionsInReadStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.CigarOperatorsInReadStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.CigarOperatorsInReadStratifier</a> insertionsInReadStratifier</pre>
<div class="block">Stratify by Insertions in the read cigars.</div>
</li>
</ul>
<a name="deletionsInReadStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>deletionsInReadStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.CigarOperatorsInReadStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.CigarOperatorsInReadStratifier</a> deletionsInReadStratifier</pre>
<div class="block">Stratify by Deletions in the read cigars.</div>
</li>
</ul>
<a name="indelsInReadStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>indelsInReadStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.IndelsInReadStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.IndelsInReadStratifier</a> indelsInReadStratifier</pre>
<div class="block">Stratify by Indels in the read cigars.</div>
</li>
</ul>
<a name="indelLengthStratifier">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>indelLengthStratifier</h4>
<pre>public static final&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.IndelLengthStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.IndelLengthStratifier</a> indelLengthStratifier</pre>
<div class="block">Stratifies into the number of bases in an insertion</div>
</li>
</ul>
<a name="NOT_ALIGNED_ERROR">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>NOT_ALIGNED_ERROR</h4>
<pre>public static final&nbsp;int NOT_ALIGNED_ERROR</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../constant-values.html#picard.sam.SamErrorMetric.ReadBaseStratification.NOT_ALIGNED_ERROR">Constant Field Values</a></dd>
</dl>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="ReadBaseStratification--">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>ReadBaseStratification</h4>
<pre>public&nbsp;ReadBaseStratification()</pre>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="setGcCacheSize-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setGcCacheSize</h4>
<pre>public static&nbsp;void&nbsp;setGcCacheSize(int&nbsp;gcCacheSize)</pre>
</li>
</ul>
<a name="setLongHomopolymer-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setLongHomopolymer</h4>
<pre>public static&nbsp;void&nbsp;setLongHomopolymer(int&nbsp;longHomopolymer)</pre>
<div class="block">defaults to 6</div>
</li>
</ul>
<a name="PairStratifierFactory-picard.sam.SamErrorMetric.ReadBaseStratification.RecordAndOffsetStratifier-picard.sam.SamErrorMetric.ReadBaseStratification.RecordAndOffsetStratifier-java.lang.String-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PairStratifierFactory</h4>
<pre>public static&nbsp;&lt;T extends java.lang.Comparable&lt;T&gt;,S extends java.lang.Comparable&lt;S&gt;&gt;&nbsp;<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.PairStratifier.html" title="class in picard.sam.SamErrorMetric">ReadBaseStratification.PairStratifier</a>&lt;T,S&gt;&nbsp;PairStratifierFactory(<a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;T&gt;&nbsp;leftStratifier,
                                                                                                                                                     <a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.RecordAndOffsetStratifier.html" title="interface in picard.sam.SamErrorMetric">ReadBaseStratification.RecordAndOffsetStratifier</a>&lt;S&gt;&nbsp;rightStratifier,
                                                                                                                                                     java.lang.String&nbsp;suffix)</pre>
<div class="block">A factory for generating "pair" stratifier instances from two stratifiers and a string</div>
<dl>
<dt><span class="paramLabel">Type Parameters:</span></dt>
<dd><code>T</code> - the type of the left Stratifier</dd>
<dd><code>S</code> - the type of the right Stratifier</dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>leftStratifier</code> - a <code>RecordAndOffsetStratifier<T></code> to use</dd>
<dd><code>rightStratifier</code> - a <code>RecordAndOffsetStratifier<S></code> to use</dd>
<dd><code>suffix</code> - the suffix to use for the new stratifier</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>an instance of <code>PairStratifier></code> that will stratify according to both <code>leftStratifier</code>
 and <code>rightStratifier</code></dd>
</dl>
</li>
</ul>
<a name="getIndelElement-htsjdk.samtools.util.SamLocusIterator.RecordAndOffset-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getIndelElement</h4>
<pre>public static&nbsp;htsjdk.samtools.CigarElement&nbsp;getIndelElement(htsjdk.samtools.util.SamLocusIterator.RecordAndOffset&nbsp;recordAndOffset)</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-all.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../picard/sam/SamErrorMetric/OverlappingReadsErrorCalculator.html" title="class in picard.sam.SamErrorMetric"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../../picard/sam/SamErrorMetric/ReadBaseStratification.BinnedReadCycleStratifier.html" title="class in picard.sam.SamErrorMetric"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?picard/sam/SamErrorMetric/ReadBaseStratification.html" target="_top">Frames</a></li>
<li><a href="ReadBaseStratification.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
